diff -ru vim-9.0.1128.orig/src/feature.h vim-9.0.1128/src/feature.h
--- vim-9.0.1128.orig/src/feature.h	2023-01-01 21:31:30.000000000 +0100
+++ vim-9.0.1128/src/feature.h	2023-01-02 11:59:16.584845483 +0100
@@ -270,6 +270,7 @@
 /*
  * +reltime		reltime() function
  */
+#ifndef __midipix__
 #if defined(FEAT_NORMAL) \
 	&& defined(FEAT_EVAL) \
 	&& ((defined(HAVE_GETTIMEOFDAY) && defined(HAVE_SYS_TIME_H) \
@@ -277,6 +278,7 @@
 		|| defined(MSWIN))
 # define FEAT_RELTIME
 #endif
+#endif
 
 /*
  * +timers		timer_start()
