diff -ru git-2.35.2.orig/setup.c git-2.35.2/setup.c
--- git-2.35.2.orig/setup.c	2022-09-25 15:35:21.270533842 +0200
+++ git-2.35.2/setup.c	2022-09-25 15:33:14.246537819 +0200
@@ -1116,6 +1116,10 @@
 
 static int ensure_valid_ownership(const char *path)
 {
+#ifdef __midipix__
+	(void)path;
+	return 1;
+#else
 	struct safe_directory_data data = { .path = path };
 
 	if (is_path_owned_by_current_user(path))
@@ -1124,6 +1128,7 @@
 	read_very_early_config(safe_directory_cb, &data);
 
 	return data.is_safe;
+#endif
 }
 
 enum discovery_result {
