Only in alsa-lib-1.1.8/include: alsa
diff -ru alsa-lib-1.1.8.orig/include/alsa-symbols.h alsa-lib-1.1.8/include/alsa-symbols.h
--- alsa-lib-1.1.8.orig/include/alsa-symbols.h	2019-01-07 13:55:38.000000000 +0100
+++ alsa-lib-1.1.8/include/alsa-symbols.h	2019-03-24 06:05:21.993902732 +0100
@@ -27,7 +27,12 @@
 #endif
 
 #define INTERNAL_CONCAT2_2(Pre, Post) Pre##Post
+#ifdef __midipix__
+/* no .symver support */
+#define INTERNAL(Name) INTERNAL_CONCAT2_2(, Name)
+#else
 #define INTERNAL(Name) INTERNAL_CONCAT2_2(__, Name)
+#endif
 
 # define symbol_version(real, name, version) \
 	__asm__ (".symver " ASM_NAME(#real) "," ASM_NAME(#name) "@" #version)
@@ -45,6 +50,8 @@
 #define use_default_symbol_version(real, name, version) \
         __asm__ (".weak " ASM_NAME(#name)); \
         __asm__ (ASM_NAME(#name) " = " ASM_NAME(#real))
+#elif defined(__midipix__)
+#define use_default_symbol_version(real, name, version) /* nothing */
 #else
 #define use_default_symbol_version(real, name, version) \
 	__asm__ (".weak " ASM_NAME(#name)); \
diff -ru alsa-lib-1.1.8.orig/include/local.h alsa-lib-1.1.8/include/local.h
--- alsa-lib-1.1.8.orig/include/local.h	2019-01-07 13:55:38.000000000 +0100
+++ alsa-lib-1.1.8/include/local.h	2019-03-23 19:06:39.073662943 +0100
@@ -254,8 +254,10 @@
 /*
  */
 #define HAVE_GNU_LD
+#ifndef __midipix__
 #define HAVE_ELF
 #define HAVE_ASM_PREVIOUS_DIRECTIVE
+#endif
 
 /* Stolen from libc-symbols.h in GNU glibc */
 
