--- apk-tools-2.6.6/src/apk.c.orig	2016-02-09 15:57:53.000000000 +0100
+++ apk-tools-2.6.6/src/apk.c	2016-03-09 10:24:39.763540600 +0100
@@ -425,12 +425,15 @@
 
 static void apk_applet_register_builtin(void)
 {
+#if 0
 	extern apk_init_func_t __start_initapplets[], __stop_initapplets[];
 	apk_init_func_t *p;
 
 	list_init(&apk_applet_list);
 	for (p = __start_initapplets; p < __stop_initapplets; p++)
 		(*p)();
+#endif
+	list_init(&apk_applet_list);
 }
 
 int main(int argc, char **argv)
--- apk-tools-2.6.6/Make.rules.orig	2016-02-09 15:57:53.000000000 +0100
+++ apk-tools-2.6.6/Make.rules	2016-03-09 10:44:12.457459300 +0100
@@ -70,7 +70,7 @@
 INSTALLDIR	:= $(INSTALL) -d
 
 CFLAGS		?= -g -O2
-CFLAGS_ALL	:= -Werror -Wall -Wstrict-prototypes -D_GNU_SOURCE -std=gnu99 -fPIC
+CFLAGS_ALL	:= -Wall -Wstrict-prototypes -D_GNU_SOURCE -std=gnu99 -fPIC
 CFLAGS_ALL	+= $(CFLAGS)
 
 LDFLAGS		?= -g
--- apk-tools-2.6.6/src/Makefile.orig	2016-02-09 15:57:53.000000000 +0100
+++ apk-tools-2.6.6/src/Makefile	2016-11-16 13:49:33.370051100 +0100
@@ -6,7 +6,7 @@
 ifneq ($(LUAAPK),)
 LUA_VERSION		?= 5.2
 LUA_PC			?= lua$(LUA_VERSION)
-LUA_LIBDIR		?= /usr/lib/lua/$(LUA_VERSION)
+LUA_LIBDIR		?= $(PREFIX)/lib/lua/$(LUA_VERSION)
 LIBAPK			:= YesPlease
 shlibs-y		+= apk.so
 apk.so-objs		:= lua-apk.o
@@ -66,7 +66,7 @@
 LDFLAGS_apk-test	+= -L$(obj)
 
 CFLAGS_ALL		+= $(shell $(PKG_CONFIG) --cflags $(PKGDEPS))
-LIBS			:=  /usr/lib/libfetch.a \
+LIBS			:=  $(PREFIX)/lib/libfetch.a \
 			   -Wl,--as-needed \
 				$(shell $(PKG_CONFIG) --libs $(PKGDEPS)) \
 			   -Wl,--no-as-needed
