--- bdwgc/configure.ac.orig	2022-01-02 10:17:27.670680925 +0100
+++ bdwgc/configure.ac	2022-01-02 10:19:10.029685443 +0100
@@ -206,7 +206,7 @@
       *-*-aix* | *-*-android* | *-*-cygwin* | *-*-darwin* | *-*-dragonfly* | \
       *-*-freebsd* | *-*-haiku* | *-*-hpux11* | *-*-irix* | \
       *-*-kfreebsd*-gnu | *-*-gnu* | *-*-*linux* | *-*-msys* | *-*-nacl* | \
-      *-*-netbsd* | *-*-openbsd* | *-*-osf* | *-*-solaris*)
+      *-*-netbsd* | *-*-openbsd* | *-*-osf* | *-*-solaris* | *-*-midipix*)
         AC_DEFINE(GC_THREADS)
         AC_DEFINE([_REENTRANT], [1],
                   [Required define if using POSIX threads.])
@@ -252,6 +252,10 @@
         THREADDLLIBS=""
         win32_threads=true
         ;;
+      *-*-midipix*)
+        # libpthread is part of libc
+        THREADDLLIBS=""
+        ;;
       *-*-mingw*)
         AC_DEFINE(GC_WIN32_PTHREADS)
         # Using pthreads-win32 (or other non-Cygwin pthreads) library.
