--- bdwgc/configure.ac.orig	2017-06-15 16:27:37.964531977 +0000
+++ bdwgc/configure.ac	2017-06-15 18:17:47.369988647 +0000
@@ -208,6 +208,10 @@
         THREADDLLIBS="-lpthread -lrt"
         # HPUX needs REENTRANT for the _r calls.
         ;;
+     *-*-midipix*)
+        AC_DEFINE(GC_THREADS)
+        AC_DEFINE(_REENTRANT)
+        ;;
       *-*-openbsd*)
         AM_CFLAGS="$AM_CFLAGS -pthread"
         THREADDLLIBS=-pthread
