diff --git a/include/private/gcconfig.h b/include/private/gcconfig.h
index b342883..174dd88 100644
--- a/include/private/gcconfig.h
+++ b/include/private/gcconfig.h
@@ -109,6 +109,21 @@ EXTERN_C_BEGIN
 #   define LINUX
 # endif
 
+# if defined(__midipix__)
+#    define MIDIPIX
+
+#    include <fcntl.h>
+#    include <sys/select.h>
+
+#    include <psxtypes/psxtypes.h>
+#    include <pemagine/pemagine.h>
+#    include <ntapi/nt_thread.h>
+
+     int __gc_data_start_mark__[1];
+     int __gc_data_end_mark__[1];
+
+# endif
+
 /* And one for NetBSD: */
 # if defined(__NetBSD__)
 #    define NETBSD
@@ -350,6 +359,16 @@ EXTERN_C_BEGIN
 #    define X86_64
 #    define mach_type_known
 # endif
+# if defined(MIDIPIX)
+#    if defined(__i386__)
+#      define I386
+#      define mach_type_known
+#    endif
+#    if defined(__x86_64__)
+#      define X86_64
+#      define mach_type_known
+#    endif
+# endif
 # if defined(LINUX) && (defined(i386) || defined(__i386__))
 #    define I386
 #    define mach_type_known
@@ -1542,6 +1561,13 @@ EXTERN_C_BEGIN
           EXTERN_C_BEGIN
 #       endif
 #   endif
+#   ifdef MIDIPIX
+#      define OS_TYPE "MIDIPIX"
+#      define DATASTART   ((ptr_t)__gc_data_start_mark__)
+#      define DATAEND     ((ptr_t)__gc_data_end_mark__)
+#      define STACKBOTTOM (ptr_t)(((nt_tib *)pe_get_teb_address())->stack_base)
+#      define USE_MMAP_ANON
+#   endif
 #   ifdef CYGWIN32
 #       define OS_TYPE "CYGWIN32"
 #       define WOW64_THREAD_CONTEXT_WORKAROUND
@@ -2706,6 +2732,13 @@ EXTERN_C_BEGIN
           EXTERN_C_BEGIN
 #       endif
 #   endif
+#   ifdef MIDIPIX
+#      define OS_TYPE "MIDIPIX"
+#      define DATASTART   ((ptr_t)__gc_data_start_mark__)
+#      define DATAEND     ((ptr_t)__gc_data_end_mark__)
+#      define STACKBOTTOM (ptr_t)(((nt_tib *)pe_get_teb_address())->stack_base)
+#      define USE_MMAP_ANON
+#   endif
 #   ifdef DARWIN
 #     define OS_TYPE "DARWIN"
 #     define DARWIN_DONT_PARSE_STACK 1
@@ -3111,7 +3144,7 @@ EXTERN_C_BEGIN
 #if defined(SVR4) || defined(LINUX) || defined(IRIX5) || defined(HPUX) \
     || defined(OPENBSD) || defined(NETBSD) || defined(FREEBSD) \
     || defined(DGUX) || defined(BSD) || defined(HAIKU) || defined(HURD) \
-    || defined(AIX) || defined(DARWIN) || defined(OSF1)
+    || defined(AIX) || defined(DARWIN) || defined(OSF1) || defined(MIDIPIX)
 # define UNIX_LIKE      /* Basic Unix-like system calls work.   */
 #endif
 
