diff -ru bind-9.11.2.orig/bin/check/Makefile.in bind-9.11.2/bin/check/Makefile.in
--- bind-9.11.2.orig/bin/check/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/check/Makefile.in	2017-09-13 16:09:24.589543862 +0200
@@ -20,7 +20,7 @@
 CDEFINES = 	@CRYPTO@ -DNAMED_CONFFILE=\"${sysconfdir}/named.conf\"
 CWARNINGS =
 
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 ISCCFGLIBS =	../../lib/isccfg/libisccfg.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
@@ -31,8 +31,8 @@
 ISCDEPLIBS =	../../lib/isc/libisc.@A@
 BIND9DEPLIBS =	../../lib/bind9/libbind9.@A@
 
-LIBS =		${ISCLIBS} @LIBS@
-NOSYMLIBS =	${ISCNOSYMLIBS} @LIBS@
+LIBS =		${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
+NOSYMLIBS =	${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 SUBDIRS =
 
diff -ru bind-9.11.2.orig/bin/confgen/Makefile.in bind-9.11.2/bin/confgen/Makefile.in
--- bind-9.11.2.orig/bin/confgen/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/confgen/Makefile.in	2017-09-13 16:10:26.857177684 +0200
@@ -24,7 +24,7 @@
 ISCCCLIBS =	../../lib/isccc/libisccc.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 BIND9LIBS =	../../lib/bind9/libbind9.@A@
 
 ISCCFGDEPLIBS =	../../lib/isccfg/libisccfg.@A@
@@ -36,9 +36,9 @@
 RNDCLIBS =	${ISCCFGLIBS} ${ISCCCLIBS} ${BIND9LIBS} ${DNSLIBS} ${ISCLIBS} @LIBS@
 RNDCDEPLIBS =	${ISCCFGDEPLIBS} ${ISCCCDEPLIBS} ${BIND9DEPLIBS} ${DNSDEPLIBS} ${ISCDEPLIBS}
 
-LIBS =		${DNSLIBS} ${ISCLIBS} @LIBS@
+LIBS =		${DNSLIBS} ${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
-NOSYMLIBS =	${DNSLIBS} ${ISCNOSYMLIBS} @LIBS@
+NOSYMLIBS =	${DNSLIBS} ${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 CONFDEPLIBS =	${DNSDEPLIBS} ${ISCDEPLIBS}
 
diff -ru bind-9.11.2.orig/bin/delv/Makefile.in bind-9.11.2/bin/delv/Makefile.in
--- bind-9.11.2.orig/bin/delv/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/delv/Makefile.in	2017-09-13 16:11:30.200805972 +0200
@@ -20,7 +20,7 @@
 CWARNINGS =
 
 ISCCFGLIBS =	../../lib/isccfg/libisccfg.@A@
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 IRSLIBS =	../../lib/irs/libirs.@A@
@@ -32,8 +32,8 @@
 
 DEPLIBS =	${DNSDEPLIBS} ${IRSDEPLIBS} ${ISCCFGDEPLIBS} ${ISCDEPLIBS}
 
-LIBS =		${DNSLIBS} ${IRSLIBS} ${ISCCFGLIBS} ${ISCLIBS} @LIBS@
-NOSYMLIBS =	${DNSLIBS} ${IRSLIBS} ${ISCCFGLIBS} ${ISCNOSYMLIBS} @LIBS@
+LIBS =		${DNSLIBS} ${IRSLIBS} ${ISCCFGLIBS} ${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
+NOSYMLIBS =	${DNSLIBS} ${IRSLIBS} ${ISCCFGLIBS} ${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 SUBDIRS =
 
diff -ru bind-9.11.2.orig/bin/dig/Makefile.in bind-9.11.2/bin/dig/Makefile.in
--- bind-9.11.2.orig/bin/dig/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/dig/Makefile.in	2017-09-13 16:12:11.496564031 +0200
@@ -22,7 +22,7 @@
 CWARNINGS =
 
 ISCCFGLIBS =	../../lib/isccfg/libisccfg.@A@
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 BIND9LIBS =	../../lib/bind9/libbind9.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
@@ -38,10 +38,10 @@
 		${ISCCFGDEPLIBS} ${LWRESDEPLIBS}
 
 LIBS =		${LWRESLIBS} ${BIND9LIBS} ${ISCCFGLIBS} \
-		${ISCLIBS} @IDNLIBS@ @LIBS@
+		${ISCLIBS} @DNS_CRYPTO_LIBS@ @IDNLIBS@ @LIBS@
 
 NOSYMLIBS =	${LWRESLIBS} ${BIND9LIBS} ${ISCCFGLIBS} \
-		${ISCNOSYMLIBS} @IDNLIBS@ @LIBS@
+		${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @IDNLIBS@ @LIBS@
 
 SUBDIRS =
 
diff -ru bind-9.11.2.orig/bin/dnssec/Makefile.in bind-9.11.2/bin/dnssec/Makefile.in
--- bind-9.11.2.orig/bin/dnssec/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/dnssec/Makefile.in	2017-09-13 16:12:31.476447075 +0200
@@ -20,7 +20,7 @@
 		@CRYPTO@ -DPK11_LIB_LOCATION=\"@PKCS11_PROVIDER@\"
 CWARNINGS =
 
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
 
@@ -29,9 +29,9 @@
 
 DEPLIBS =	${DNSDEPLIBS} ${ISCDEPLIBS}
 
-LIBS =		${DNSLIBS} ${ISCLIBS} @LIBS@
+LIBS =		${DNSLIBS} ${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
-NOSYMLIBS =	${DNSLIBS} ${ISCNOSYMLIBS} @LIBS@
+NOSYMLIBS =	${DNSLIBS} ${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 # Alphabetically
 TARGETS =	dnssec-keygen@EXEEXT@ dnssec-signzone@EXEEXT@ \
diff -ru bind-9.11.2.orig/bin/Makefile.in bind-9.11.2/bin/Makefile.in
--- bind-9.11.2.orig/bin/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/Makefile.in	2017-09-13 16:13:03.324260771 +0200
@@ -10,7 +10,7 @@
 VPATH =		@srcdir@
 top_srcdir =	@top_srcdir@
 
-SUBDIRS =	named rndc dig delv dnssec tools tests nsupdate \
+SUBDIRS =	named rndc dig delv dnssec tools nsupdate \
 		check confgen @NZD_TOOLS@ @PYTHON_TOOLS@ @PKCS11_TOOLS@
 TARGETS =
 
diff -ru bind-9.11.2.orig/bin/named/Makefile.in bind-9.11.2/bin/named/Makefile.in
--- bind-9.11.2.orig/bin/named/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/named/Makefile.in	2017-09-13 16:13:34.788076866 +0200
@@ -44,7 +44,7 @@
 
 CWARNINGS =
 
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 ISCCFGLIBS =	../../lib/isccfg/libisccfg.@A@
 ISCCCLIBS =	../../lib/isccc/libisccc.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
@@ -64,11 +64,11 @@
 
 LIBS =		${LWRESLIBS} ${DNSLIBS} ${BIND9LIBS} \
 		${ISCCFGLIBS} ${ISCCCLIBS} ${ISCLIBS} \
-		${DLZDRIVER_LIBS} ${DBDRIVER_LIBS} @LIBS@
+		${DLZDRIVER_LIBS} ${DBDRIVER_LIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 NOSYMLIBS =	${LWRESLIBS} ${DNSLIBS} ${BIND9LIBS} \
 		${ISCCFGLIBS} ${ISCCCLIBS} ${ISCNOSYMLIBS} \
-		${DLZDRIVER_LIBS} ${DBDRIVER_LIBS} @LIBS@
+		${DLZDRIVER_LIBS} ${DBDRIVER_LIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 SUBDIRS =	unix
 
diff -ru bind-9.11.2.orig/bin/nsupdate/Makefile.in bind-9.11.2/bin/nsupdate/Makefile.in
--- bind-9.11.2.orig/bin/nsupdate/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/nsupdate/Makefile.in	2017-09-13 16:14:07.755884315 +0200
@@ -26,7 +26,7 @@
 CWARNINGS =
 
 LWRESLIBS =	../../lib/lwres/liblwres.@A@
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 BIND9LIBS =	../../lib/bind9/libbind9.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
@@ -40,9 +40,9 @@
 
 DEPLIBS =	${DNSDEPLIBS} ${BIND9DEPLIBS} ${ISCDEPLIBS} ${ISCCFGDEPLIBS}
 
-LIBS =		${LWRESLIBS} ${DNSLIBS} ${BIND9LIBS} ${ISCCFGLIBS} ${ISCLIBS} @LIBS@
+LIBS =		${LWRESLIBS} ${DNSLIBS} ${BIND9LIBS} ${ISCCFGLIBS} ${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
-NOSYMLIBS =	${LWRESLIBS} ${DNSLIBS} ${BIND9LIBS} ${ISCCFGLIBS} ${ISCNOSYMLIBS} @LIBS@
+NOSYMLIBS =	${LWRESLIBS} ${DNSLIBS} ${BIND9LIBS} ${ISCCFGLIBS} ${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 SUBDIRS =
 
diff -ru bind-9.11.2.orig/bin/rndc/Makefile.in bind-9.11.2/bin/rndc/Makefile.in
--- bind-9.11.2.orig/bin/rndc/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/rndc/Makefile.in	2017-09-13 16:14:38.291706093 +0200
@@ -24,7 +24,7 @@
 ISCCCLIBS =	../../lib/isccc/libisccc.@A@
 ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 BIND9LIBS =	../../lib/bind9/libbind9.@A@
 
 ISCCFGDEPLIBS =	../../lib/isccfg/libisccfg.@A@
@@ -33,8 +33,8 @@
 DNSDEPLIBS =	../../lib/dns/libdns.@A@
 BIND9DEPLIBS =	../../lib/bind9/libbind9.@A@
 
-LIBS =		${ISCLIBS} @LIBS@
-NOSYMLIBS =	${ISCNOSYMLIBS} @LIBS@
+LIBS =		${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
+NOSYMLIBS =	${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 RNDCDEPLIBS =	${ISCCFGDEPLIBS} ${ISCCCDEPLIBS} ${BIND9DEPLIBS} ${DNSDEPLIBS} ${ISCDEPLIBS}
 
diff -ru bind-9.11.2.orig/bin/tools/Makefile.in bind-9.11.2/bin/tools/Makefile.in
--- bind-9.11.2.orig/bin/tools/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/bin/tools/Makefile.in	2017-09-13 16:15:59.075235122 +0200
@@ -19,9 +19,9 @@
 CDEFINES =	@CRYPTO@ -DVERSION=\"${VERSION}\"
 CWARNINGS =
 
-DNSLIBS =	../../lib/dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../../lib/dns/libdns.@A@
 BIND9LIBS =	../../lib/bind9/libbind9.@A@
-ISCLIBS =	../../lib/isc/libisc.@A@ @DNS_CRYPTO_LIBS@
+ISCLIBS =	../../lib/isc/libisc.@A@
 ISCNOSYMLIBS =	../../lib/isc/libisc-nosymtbl.@A@
 ISCCFGLIBS = 	../../lib/isccfg/libisccfg.@A@
 LWRESLIBS =	../../lib/lwres/liblwres.@A@
@@ -32,8 +32,8 @@
 ISCCFGDEPLIBS = ../../lib/isccfg/libisccfg.@A@
 LWRESDEPLIBS =	../../lib/lwres/liblwres.@A@
 
-LIBS =		${ISCLIBS} @LIBS@
-NOSYMLIBS =	${ISCNOSYMLIBS} @LIBS@
+LIBS =		${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
+NOSYMLIBS =	${ISCNOSYMLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 SUBDIRS =
 
diff -ru bind-9.11.2.orig/configure.in bind-9.11.2/configure.in
--- bind-9.11.2.orig/configure.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/configure.in	2017-09-13 16:21:26.765330737 +0200
@@ -2335,9 +2335,9 @@
 		esac
 		;;
 	*)
-		if test -f "$use_libxml2/bin/xml2-config" ; then
-			libxml2_libs=`$use_libxml2/bin/xml2-config --libs`
-			libxml2_cflags=`$use_libxml2/bin/xml2-config --cflags`
+		if test -f "$use_libxml2" ; then
+			libxml2_libs=`$use_libxml2 --libs`
+			libxml2_cflags=`$use_libxml2 --cflags`
 		fi
 		;;
 esac
diff -ru bind-9.11.2.orig/lib/irs/Makefile.in bind-9.11.2/lib/irs/Makefile.in
--- bind-9.11.2.orig/lib/irs/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/lib/irs/Makefile.in	2017-09-13 16:17:32.502691292 +0200
@@ -35,7 +35,7 @@
 		gai_strerror.c getaddrinfo.c getnameinfo.c \
 		resconf.c
 
-LIBS =		@LIBS@
+LIBS =		@LIBS@ -L../isc -lisc -L../dns -ldns -L../isccfg -lisccfg
 
 SUBDIRS =	include
 TESTDIRS =	@UNITTESTS@
diff -ru bind-9.11.2.orig/lib/isc/Makefile.in bind-9.11.2/lib/isc/Makefile.in
--- bind-9.11.2.orig/lib/isc/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/lib/isc/Makefile.in	2017-09-13 16:25:11.723950416 +0200
@@ -108,26 +108,16 @@
 	${AR} ${ARFLAGS} $@ ${OBJS} ${SYMTBLOBJS}
 	${RANLIB} $@
 
-libisc-nosymtbl.@SA@: ${OBJS}
-	${AR} ${ARFLAGS} $@ ${OBJS}
-	${RANLIB} $@
-
 libisc.la: ${OBJS} ${SYMTBLOBJS}
 	${LIBTOOL_MODE_LINK} \
 		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisc.la -rpath ${libdir} \
 		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
 		${OBJS} ${SYMTBLOBJS} ${LIBS}
 
-libisc-nosymtbl.la: ${OBJS}
-	${LIBTOOL_MODE_LINK} \
-		${CC} ${ALL_CFLAGS} ${LDFLAGS} -o libisc-nosymtbl.la -rpath ${libdir} \
-		-version-info ${LIBINTERFACE}:${LIBREVISION}:${LIBAGE} \
-		${OBJS} ${LIBS}
-
-timestamp: libisc.@A@ libisc-nosymtbl.@A@
+timestamp: libisc.@A@
 	touch timestamp
 
-testdirs: libisc.@A@ libisc-nosymtbl.@A@
+testdirs: libisc.@A@
 
 installdirs:
 	$(SHELL) ${top_srcdir}/mkinstalldirs ${DESTDIR}${libdir}
@@ -139,5 +129,4 @@
 	${LIBTOOL_MODE_UNINSTALL} rm -f ${DESTDIR}${libdir}/libisc.@A@
 
 clean distclean::
-	rm -f libisc.@A@ libisc-nosymtbl.@A@ libisc.la \
-	libisc-nosymtbl.la timestamp
+	rm -f libisc.@A@ libisc.la timestamp
diff -ru bind-9.11.2.orig/lib/Makefile.in bind-9.11.2/lib/Makefile.in
--- bind-9.11.2.orig/lib/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/lib/Makefile.in	2017-09-13 16:19:00.166181699 +0200
@@ -14,7 +14,7 @@
 # Attempt to disable parallel processing.
 .NOTPARALLEL:
 .NO_PARALLEL:
-SUBDIRS =	isc isccc dns isccfg bind9 lwres irs tests samples
+SUBDIRS =	isc isccc dns isccfg bind9 lwres irs samples
 TARGETS =
 
 @BIND9_MAKE_RULES@
diff -ru bind-9.11.2.orig/lib/samples/Makefile.in bind-9.11.2/lib/samples/Makefile.in
--- bind-9.11.2.orig/lib/samples/Makefile.in	2017-07-24 07:36:50.000000000 +0200
+++ bind-9.11.2/lib/samples/Makefile.in	2017-09-13 16:19:36.149972702 +0200
@@ -22,7 +22,7 @@
 CWARNINGS =
 
 ISCLIBS =	../isc/libisc.@A@
-DNSLIBS =	../dns/libdns.@A@ @DNS_CRYPTO_LIBS@
+DNSLIBS =	../dns/libdns.@A@
 ISCCFGLIBS =	../isccfg/libisccfg.@A@
 IRSLIBS =	../irs/libirs.@A@
 
@@ -33,7 +33,7 @@
 
 DEPLIBS =	${DNSDEPLIBS} ${ISCCFGDEPLIBS} ${ISCDEPLIBS}
 
-LIBS =		${DNSLIBS} ${ISCCFGLIBS} ${ISCLIBS} @LIBS@
+LIBS =		${DNSLIBS} ${ISCCFGLIBS} ${ISCLIBS} @DNS_CRYPTO_LIBS@ @LIBS@
 
 SUBDIRS =
 
