diff -ru busybox-1.28.3.orig/scripts/trylink busybox-1.28.3/scripts/trylink
--- busybox-1.28.3.orig/scripts/trylink	2018-03-26 01:59:01.000000000 +0200
+++ busybox-1.28.3/scripts/trylink	2018-05-13 21:10:02.921178839 +0200
@@ -250,13 +250,12 @@
     LBB_STATIC=""
     test "$CONFIG_FEATURE_LIBBUSYBOX_STATIC" = y && LBB_STATIC="-Wl,-static"
 
-    EXE="$sharedlib_dir/libbusybox.so.${BB_VER}_unstripped"
+    EXE="$sharedlib_dir/libbusybox.so.${BB_VER}"
     try $CC $CFLAGS $LDFLAGS \
 	    -o $EXE \
 	    -shared -fPIC $LBB_STATIC \
-	    -Wl,--enable-new-dtags \
-	    -Wl,-z,combreloc \
 	    -Wl,-soname="libbusybox.so.$BB_VER" \
+	    -Wl,--out-dsolib,$sharedlib_dir/libbusybox.lib.a \
 	    -Wl,--undefined=lbb_main \
 	    $SORT_COMMON \
 	    $SORT_SECTION \
@@ -268,7 +267,6 @@
 	cat $EXE.out
 	exit 1
     }
-    $STRIP -s --remove-section=.note --remove-section=.comment $EXE -o "$sharedlib_dir/libbusybox.so.$BB_VER"
     chmod a+x "$sharedlib_dir/libbusybox.so.$BB_VER"
     echo "libbusybox: $sharedlib_dir/libbusybox.so.$BB_VER"
 fi
