diff -ru cmake-3.7.2.orig/Utilities/cmcurl/lib/vtls/openssl.c cmake-3.7.2/Utilities/cmcurl/lib/vtls/openssl.c
--- cmake-3.7.2.orig/Utilities/cmcurl/lib/vtls/openssl.c        2017-01-13 15:05:42.000000000 +0100
+++ cmake-3.7.2/Utilities/cmcurl/lib/vtls/openssl.c     2022-11-04 12:08:55.736358938 +0100
@@ -108,8 +108,7 @@
 #define OPENSSL_NO_SSL2
 #endif

-#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) && /* OpenSSL 1.1.0+ */ \
-  !defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x10100000L) /* OpenSSL 1.1.0+ */
 #define SSLeay_add_ssl_algorithms() SSL_library_init()
 #define SSLEAY_VERSION_NUMBER OPENSSL_VERSION_NUMBER
 #define HAVE_X509_GET0_EXTENSIONS 1 /* added in 1.1.0 -pre1 */
@@ -117,8 +116,7 @@
 #define HAVE_OPAQUE_RSA_DSA_DH 1 /* since 1.1.0 -pre5 */
 #endif

-#if (OPENSSL_VERSION_NUMBER >= 0x1000200fL) && /* 1.0.2 or later */ \
-  !defined(LIBRESSL_VERSION_NUMBER)
+#if (OPENSSL_VERSION_NUMBER >= 0x1000200fL) /* 1.0.2 or later */
 #define HAVE_X509_GET0_SIGNATURE 1
 #endif

@@ -1315,8 +1313,7 @@
   ch = SSL_get_peer_cert_chain(connssl->handle);
   st = SSL_CTX_get_cert_store(connssl->ctx);

-#if ((OPENSSL_VERSION_NUMBER <= 0x1000201fL) /* Fixed after 1.0.2a */ || \
-     defined(LIBRESSL_VERSION_NUMBER))
+#if (OPENSSL_VERSION_NUMBER <= 0x1000201fL) /* Fixed after 1.0.2a */
   /* The authorized responder cert in the OCSP response MUST be signed by the
      peer cert's issuer (see RFC6960 section 4.2.2.2). If that's a root cert,
      no problem, but if it's an intermediate cert OpenSSL has a bug where it
diff -ru cmake-3.7.2.orig/Source/cmServerProtocol.cxx cmake-3.7.2/Source/cmServerProtocol.cxx
--- cmake-3.7.2.orig/Source/cmServerProtocol.cxx        2017-01-13 15:05:41.000000000 +0100
+++ cmake-3.7.2/Source/cmServerProtocol.cxx     2021-09-10 19:16:52.942413410 +0200
@@ -26,6 +26,7 @@
 #include <algorithm>
 #include <string>
 #include <vector>
+#include <limits>

 // Get rid of some windows macros:
 #undef max
diff -ru cmake-3.7.2.orig/Utilities/cmlibarchive/libarchive/archive_openssl_hmac_private.h cmake-3.7.2/Utilities/cmlibarchive/libarchive/archive_openssl_hmac_private.h
--- cmake-3.7.2.orig/Utilities/cmlibarchive/libarchive/archive_openssl_hmac_private.h	2017-01-13 15:05:42.000000000 +0100
+++ cmake-3.7.2/Utilities/cmlibarchive/libarchive/archive_openssl_hmac_private.h	2018-05-08 22:39:23.249070904 +0200
@@ -28,7 +28,8 @@
 #include <openssl/hmac.h>
 #include <openssl/opensslv.h>
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || \
+    (defined(LIBRESSL_VERSION_NUMBER) && LIBRESSL_VERSION_NUMBER < 0x2070000fL)
 #include <stdlib.h> /* malloc, free */
 #include <string.h> /* memset */
 static inline HMAC_CTX *HMAC_CTX_new(void)
