diff -ru curl-8.7.1.orig/curl-config.in curl-8.7.1/curl-config.in
--- curl-8.7.1.orig/curl-config.in      2024-03-06 22:16:03.000000000 +0100
+++ curl-8.7.1/curl-config.in   2024-04-09 00:07:35.445813553 +0200
@@ -152,13 +152,13 @@
         if test "X@includedir@" = "X/usr/include"; then
           echo "$CPPFLAG_CURL_STATICLIB"
         else
-          echo "${CPPFLAG_CURL_STATICLIB}-I@includedir@"
+          echo "$CPPFLAG_CURL_STATICLIB"
         fi
         ;;

     --libs)
         if test "X@libdir@" != "X/usr/lib" -a "X@libdir@" != "X/usr/lib64"; then
-           CURLLIBDIR="-L@libdir@ "
+           CURLLIBDIR=""
         else
            CURLLIBDIR=""
         fi
diff -ru curl-8.7.1.orig/m4/curl-compilers.m4 curl-8.7.1/m4/curl-compilers.m4
--- curl-8.7.1.orig/m4/curl-compilers.m4	2024-03-25 09:48:59.000000000 +0100
+++ curl-8.7.1/m4/curl-compilers.m4	2024-04-08 23:23:12.804956865 +0200
@@ -41,33 +41,6 @@
   flags_opt_off="unknown"
   #
   flags_prefer_cppflags="no"
-  #
-  CURL_CHECK_COMPILER_DEC_C
-  CURL_CHECK_COMPILER_HPUX_C
-  CURL_CHECK_COMPILER_IBM_C
-  CURL_CHECK_COMPILER_INTEL_C
-  CURL_CHECK_COMPILER_CLANG
-  CURL_CHECK_COMPILER_GNU_C
-  CURL_CHECK_COMPILER_SGI_MIPSPRO_C
-  CURL_CHECK_COMPILER_SGI_MIPS_C
-  CURL_CHECK_COMPILER_SUNPRO_C
-  CURL_CHECK_COMPILER_TINY_C
-  #
-  if test "$compiler_id" = "unknown"; then
-  cat <<_EOF 1>&2
-***
-*** Warning: This configure script does not have information about the
-*** compiler you are using, relative to the flags required to enable or
-*** disable generation of debug info, optimization options or warnings.
-***
-*** Whatever settings are present in CFLAGS will be used for this run.
-***
-*** If you wish to help the curl project to better support your compiler
-*** you can report this and the required info on the libcurl development
-*** mailing list: https://lists.haxx.selistinfo/curl-library/
-***
-_EOF
-  fi
 ])
 
 
