diff -ru curl-8.7.1.orig/configure curl-8.7.1/configure
--- curl-8.7.1.orig/configure	2024-04-08 23:55:17.997659030 +0200
+++ curl-8.7.1/configure	2024-04-09 00:03:10.404057041 +0200
@@ -15627,6 +15627,16 @@
   hardcode_libdir_flag_spec='-L$libdir'
   ;;
 
+midipix*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
+
 # This must be glibc/ELF.
 linux* | k*bsd*-gnu | kopensolaris*-gnu | gnu*)
   version_type=linux # correct to gnu/linux during the next big refactor
