--- curl-7.51.0/configure.orig	2016-10-31 10:41:43.000000000 +0100
+++ curl-7.51.0/configure	2016-12-14 19:53:31.265092100 +0100
@@ -14831,6 +14831,18 @@
   dynamic_linker='GNU/Linux ld.so'
   ;;
 
+midipix*)
+  version_type=linux
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec="\${libname}\${release}\${shared_ext}\$versuffix \${libname}\${release}\${shared_ext}\$major \$libname\${shared_ext}"
+  soname_spec="\${libname}\${release}\${shared_ext}\$major"
+  shlibpath_var=LD_LIBRARY_PATH
+  shlibpath_overrides_runpath=no
+  hardcode_into_libs=no
+  dynamic_linker='GNU/Linux ld.so'
+  ;;
+
 netbsdelf*-gnu)
   version_type=linux
   need_lib_prefix=no
--- curl-7.51.0/lib/Makefile.in.orig	2016-10-31 10:41:45.000000000 +0100
+++ curl-7.51.0/lib/Makefile.in	2016-12-14 19:54:05.594475300 +0100
@@ -249,7 +249,7 @@
 am__v_lt_1 = 
 libcurl_la_LINK = $(LIBTOOL) $(AM_V_lt) --tag=CC $(AM_LIBTOOLFLAGS) \
 	$(LIBTOOLFLAGS) --mode=link $(CCLD) $(libcurl_la_CFLAGS) \
-	$(CFLAGS) $(libcurl_la_LDFLAGS) $(LDFLAGS) -o $@
+	$(CFLAGS) $(LDFLAGS) -o $@
 libcurlu_la_LIBADD =
 am__objects_7 = libcurlu_la-file.lo libcurlu_la-timeval.lo \
 	libcurlu_la-base64.lo libcurlu_la-hostip.lo \
@@ -889,7 +889,7 @@
 	vtls/$(DEPDIR)/$(am__dirstamp)
 
 libcurl.la: $(libcurl_la_OBJECTS) $(libcurl_la_DEPENDENCIES) $(EXTRA_libcurl_la_DEPENDENCIES) 
-	$(AM_V_CCLD)$(libcurl_la_LINK) -rpath $(libdir) $(libcurl_la_OBJECTS) $(libcurl_la_LIBADD) $(LIBS)
+	$(AM_V_CCLD)$(libcurl_la_LINK) -rpath $(libdir) $(libcurl_la_OBJECTS) $(libcurl_la_LIBADD) $(LIBS) $(libcurl_la_LDFLAGS)
 vauth/libcurlu_la-vauth.lo: vauth/$(am__dirstamp) \
 	vauth/$(DEPDIR)/$(am__dirstamp)
 vauth/libcurlu_la-cleartext.lo: vauth/$(am__dirstamp) \
