diff -ru dante-1.4.3.orig/configure dante-1.4.3/configure
--- dante-1.4.3.orig/configure	2021-04-30 21:29:06.000000000 +0200
+++ dante-1.4.3/configure	2024-02-22 00:14:19.279836011 +0100
@@ -3314,8 +3314,6 @@
 #NOTE: save CFLAGS; wish to compile without -O2 when debugging
 oCFLAGS="$CFLAGS"
 oLDFLAGS="$LDFLAGS"
-unset CFLAGS
-unset LDFLAGS
 case `pwd` in
   *\ * | *\	*)
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Libtool does not cope well with whitespace in \`pwd\`" >&5
@@ -15347,10 +15345,8 @@
 $as_echo_n "checking whether realloc with a NULL pointer calls malloc... " >&6; }
 if test "$cross_compiling" = yes; then :
 
-$as_echo "#define HAVE_NOMALLOC_REALLOC 1" >>confdefs.h
-
-     { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
+     { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming yes" >&5
+$as_echo "assuming yes" >&6; }
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -16929,8 +16925,9 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking read/send-side pipe system" >&5
 $as_echo_n "checking read/send-side pipe system... " >&6; }
 if test "$cross_compiling" = yes; then :
-      { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling" >&5
-$as_echo "cross-compiling" >&6; }
+      { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming send-based" >&5
+$as_echo "assuming send-based" >&6; }
+$as_echo "#define HAVE_PIPEBUFFER_SEND_BASED 1" >>confdefs.h
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -17232,13 +17229,10 @@
  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for errorless select behavior with high timeouts" >&5
 $as_echo_n "checking for errorless select behavior with high timeouts... " >&6; }
  if test "$cross_compiling" = yes; then :
-         { $as_echo "$as_me:${as_lineno-$LINENO}: result: unknown" >&5
-$as_echo "unknown" >&6; }
-       { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: unable to determine max select value, using default" >&5
-$as_echo "$as_me: WARNING: unable to determine max select value, using default" >&2;}
-
+         { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling: assuming yes -- midipix specific" >&5
+$as_echo "yes" >&6; }
 cat >>confdefs.h <<_ACEOF
-#define HAVE_SELECT_MAXTIMEOUT $DEFAULTMAX
+#define HAVE_SELECT_MAXTIMEOUT 0
 _ACEOF
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -23234,12 +23228,13 @@
 
 #AC_CHECK_FUNCS will add HAVE_foo define as long as function exists,
 #check overselves as we only want define set if function is also working.
-ac_cv_func_pselect=no
+ac_cv_func_pselect=yes
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for working pselect()" >&5
 $as_echo_n "checking for working pselect()... " >&6; }
 if test "$cross_compiling" = yes; then :
-       { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming no" >&5
-$as_echo "assuming no" >&6; }
+       { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming yes" >&5
+$as_echo "assuming yes" >&6; }
+$as_echo "#define HAVE_PSELECT 1" >>confdefs.h
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -23292,8 +23287,9 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for working sockatmark" >&5
 $as_echo_n "checking for working sockatmark... " >&6; }
 if test "$cross_compiling" = yes; then :
-       { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming no" >&5
-$as_echo "assuming no" >&6; }
+       { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming yes" >&5
+$as_echo "assuming yes" >&6; }
+$as_echo "#define HAVE_SOCKATMARK 1" >>confdefs.h
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
@@ -23894,8 +23890,9 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for CLOCK_MONOTONIC clock_gettime() support" >&5
 $as_echo_n "checking for CLOCK_MONOTONIC clock_gettime() support... " >&6; }
 if test "$cross_compiling" = yes; then :
-       { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming no" >&5
-$as_echo "assuming no" >&6; }
+       { $as_echo "$as_me:${as_lineno-$LINENO}: result: assuming yes" >&5
+$as_echo "assuming yes" >&6; }
+$as_echo "#define HAVE_CLOCK_GETTIME_MONOTONIC 1" >>confdefs.h
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
Only in dante-1.4.3: configure.orig
diff -ru dante-1.4.3.orig/dlib/Makefile.in dante-1.4.3/dlib/Makefile.in
--- dante-1.4.3.orig/dlib/Makefile.in	2021-04-30 21:30:18.000000000 +0200
+++ dante-1.4.3/dlib/Makefile.in	2024-02-22 00:14:19.283169345 +0100
@@ -409,7 +409,7 @@
 	../libscompat/pselect.c ../libscompat/setproctitle.c \
 	../libscompat/strlcpy.c ../libscompat/strvis.c \
 	../libscompat/vsyslog.c ../libscompat/issetugid.c
-libdsocks_la_LDFLAGS = -avoid-version -all-dynamic @MAPOPT@
+libdsocks_la_LDFLAGS = -avoid-version @MAPOPT@
 libdsocks_la_LIBADD = @LIBDSCSRC@ @DLIBDEPS@ @LIBSCOMPAT@
 libdsocks_la_DEPENDENCIES = Makefile.am _reconf @LIBDSCSRC@ \
 	$(am__append_1)
Only in dante-1.4.3/dlib: Makefile.in.orig
diff -ru dante-1.4.3.orig/include/symbols_preload.txt dante-1.4.3/include/symbols_preload.txt
--- dante-1.4.3.orig/include/symbols_preload.txt	2024-02-22 00:37:06.753086184 +0100
+++ dante-1.4.3/include/symbols_preload.txt	2024-02-22 00:37:21.979752748 +0100
@@ -2,11 +2,9 @@
 bind
 bindresvport
 connect
-freehostent
 getaddrinfo
 gethostbyname
 gethostbyname2
-getipnodebyname
 getpeername
 getsockname
 getsockopt
@@ -16,7 +14,6 @@
 recv
 recvfrom
 recvmsg
-rresvport
 send
 sendmsg
 sendto
