--- dbus-1.11.20/test/Makefile.am.orig	2017-09-25 16:00:11.000000000 +0000
+++ dbus-1.11.20/test/Makefile.am	2018-10-14 20:14:12.616000000 +0000
@@ -541,7 +541,7 @@
 ## copy tests to builddir so that generated tests and static tests
 ## are all in one place.
 all-local: copy-config-local uninstalled-config-local
-	$(AM_V_at)$(MKDIR_P) -m700 XDG_RUNTIME_DIR
+	$(AM_V_at)$(MKDIR_P) -m 700 XDG_RUNTIME_DIR
 
 copy-config-local:
 	$(AM_V_at)$(MKDIR_P) data/valid-config-files/session.d
