diff -ru dbus-1.11.20.orig/bus/Makefile.in dbus-1.11.20/bus/Makefile.in
--- dbus-1.11.20.orig/bus/Makefile.in	2017-10-03 08:46:44.000000000 +0200
+++ dbus-1.11.20/bus/Makefile.in	2017-10-10 01:13:19.263921202 +0200
@@ -164,8 +164,8 @@
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1)
-dbus_daemon_DEPENDENCIES = $(top_builddir)/dbus/libdbus-1.la \
-	$(top_builddir)/dbus/libdbus-internal.la $(am__DEPENDENCIES_1) \
+dbus_daemon_DEPENDENCIES = $(top_builddir)/dbus/libdbus-internal.la \
+	$(top_builddir)/dbus/libdbus-1.la $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_2)
 AM_V_lt = $(am__v_lt_@AM_V@)
 am__v_lt_ = $(am__v_lt_@AM_DEFAULT_V@)
@@ -182,8 +182,8 @@
 am__DEPENDENCIES_3 = $(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1) \
 	$(am__DEPENDENCIES_1) $(am__DEPENDENCIES_1)
 dbus_daemon_launch_helper_DEPENDENCIES =  \
-	$(top_builddir)/dbus/libdbus-1.la \
-	$(top_builddir)/dbus/libdbus-internal.la $(am__DEPENDENCIES_3)
+	$(top_builddir)/dbus/libdbus-internal.la \
+	$(top_builddir)/dbus/libdbus-1.la $(am__DEPENDENCIES_3)
 am__objects_4 =  \
 	dbus_daemon_launch_helper_test-config-loader-expat.$(OBJEXT) \
 	dbus_daemon_launch_helper_test-config-parser-common.$(OBJEXT) \
@@ -196,8 +196,8 @@
 dbus_daemon_launch_helper_test_OBJECTS =  \
 	$(am_dbus_daemon_launch_helper_test_OBJECTS)
 dbus_daemon_launch_helper_test_DEPENDENCIES =  \
-	$(top_builddir)/dbus/libdbus-1.la \
-	$(top_builddir)/dbus/libdbus-internal.la $(am__DEPENDENCIES_3)
+	$(top_builddir)/dbus/libdbus-internal.la \
+	$(top_builddir)/dbus/libdbus-1.la $(am__DEPENDENCIES_3)
 am__test_bus_SOURCES_DIST = activation.c activation.h \
 	activation-exit-codes.h apparmor.c apparmor.h audit.c audit.h \
 	bus.c bus.h config-loader-expat.c config-parser.c \
@@ -223,8 +223,8 @@
 	$(am__objects_5)
 test_bus_launch_helper_OBJECTS = $(am_test_bus_launch_helper_OBJECTS)
 test_bus_launch_helper_DEPENDENCIES =  \
-	$(top_builddir)/dbus/libdbus-1.la \
-	$(top_builddir)/dbus/libdbus-internal.la $(am__DEPENDENCIES_3)
+	$(top_builddir)/dbus/libdbus-internal.la \
+	$(top_builddir)/dbus/libdbus-1.la $(am__DEPENDENCIES_3)
 am_test_bus_system_OBJECTS = config-loader-expat.$(OBJEXT) \
 	config-parser-common.$(OBJEXT) config-parser-trivial.$(OBJEXT) \
 	utils.$(OBJEXT) test-system.$(OBJEXT)
@@ -685,8 +685,8 @@
 	main.c
 
 dbus_daemon_LDADD = \
-	$(top_builddir)/dbus/libdbus-1.la	\
-	$(top_builddir)/dbus/libdbus-internal.la	\
+	$(top_builddir)/dbus/libdbus-internal.la 	\
+	$(top_builddir)/dbus/libdbus-1.la		\
 	$(EFENCE)					\
 	$(DBUS_BUS_LIBS)
 
@@ -709,8 +709,8 @@
 	$(LAUNCH_HELPER_SOURCES)
 
 dbus_daemon_launch_helper_LDADD = \
+	$(top_builddir)/dbus/libdbus-internal.la \
 	$(top_builddir)/dbus/libdbus-1.la \
-	$(top_builddir)/dbus/libdbus-internal.la	\
 	$(DBUS_LAUNCHER_LIBS)
 
 dbus_daemon_launch_helper_test_SOURCES = \
diff -ru dbus-1.11.20.orig/dbus/dbus-sysdeps.h dbus-1.11.20/dbus/dbus-sysdeps.h
--- dbus-1.11.20.orig/dbus/dbus-sysdeps.h	2017-09-27 16:15:31.000000000 +0200
+++ dbus-1.11.20/dbus/dbus-sysdeps.h	2017-10-10 01:24:40.402809152 +0200
@@ -53,7 +53,7 @@
  * DBusPollFD
  */ 
 #ifdef HAVE_POLL
-#include <sys/poll.h>
+#include <poll.h>
 #endif
 
 #ifdef DBUS_WINCE
diff -ru dbus-1.11.20.orig/dbus/dbus-sysdeps-unix.c dbus-1.11.20/dbus/dbus-sysdeps-unix.c
--- dbus-1.11.20.orig/dbus/dbus-sysdeps-unix.c	2017-10-02 10:41:48.000000000 +0200
+++ dbus-1.11.20/dbus/dbus-sysdeps-unix.c	2017-10-10 01:24:57.590686601 +0200
@@ -69,7 +69,7 @@
 #include <sys/uio.h>
 #endif
 #ifdef HAVE_POLL
-#include <sys/poll.h>
+#include <poll.h>
 #endif
 #ifdef HAVE_BACKTRACE
 #include <execinfo.h>
diff -ru dbus-1.11.20.orig/dbus/Makefile.in dbus-1.11.20/dbus/Makefile.in
--- dbus-1.11.20.orig/dbus/Makefile.in	2017-10-03 08:46:45.000000000 +0200
+++ dbus-1.11.20/dbus/Makefile.in	2017-10-10 00:54:28.948451539 +0200
@@ -250,7 +250,7 @@
 @DBUS_WIN_TRUE@am_libdbus_init_win_la_OBJECTS = dbus-init-win.lo
 libdbus_init_win_la_OBJECTS = $(am_libdbus_init_win_la_OBJECTS)
 @DBUS_WIN_TRUE@am_libdbus_init_win_la_rpath =
-libdbus_internal_la_DEPENDENCIES = $(am__DEPENDENCIES_1) libdbus-1.la
+libdbus_internal_la_DEPENDENCIES = $(am__DEPENDENCIES_1)
 am__libdbus_internal_la_SOURCES_DIST = dbus-asv-util.c dbus-asv-util.h \
 	dbus-auth-script.c dbus-auth-script.h dbus-auth-util.c \
 	dbus-credentials-util.c dbus-mainloop.c dbus-mainloop.h \
@@ -838,7 +838,7 @@
 	$(dbus_res_ldflag) \
 	$(NULL)
 
-libdbus_internal_la_LIBADD = $(LIBDBUS_LIBS) libdbus-1.la
+libdbus_internal_la_LIBADD = $(LIBDBUS_LIBS)
 @DBUS_WIN_TRUE@AM_CXXFLAGS = \
 @DBUS_WIN_TRUE@	$(CODE_COVERAGE_CXXFLAGS) \
 @DBUS_WIN_TRUE@	$(NULL)
