--- dbus-1.11.20/test/Makefile.in.orig	2017-10-03 06:46:45.000000000 +0000
+++ dbus-1.11.20/test/Makefile.in	2018-10-14 20:56:47.156000000 +0000
@@ -2288,7 +2288,7 @@
 @DBUS_ENABLE_INSTALLED_TESTS_TRUE@		$(installable_tests); }
 
 all-local: copy-config-local uninstalled-config-local
-	$(AM_V_at)$(MKDIR_P) -m700 XDG_RUNTIME_DIR
+	$(AM_V_at)$(MKDIR_P) -m 700 XDG_RUNTIME_DIR
 
 copy-config-local:
 	$(AM_V_at)$(MKDIR_P) data/valid-config-files/session.d
