diff -ru dropbear-2017.75.orig/configure dropbear-2017.75/configure
--- dropbear-2017.75.orig/configure	2017-05-18 16:47:03.000000000 +0200
+++ dropbear-2017.75/configure	2017-08-23 17:34:07.708548199 +0200
@@ -6793,20 +6793,22 @@
 
 fi
 
-
-# Solaris needs ptmx
-if test -z "$no_ptmx_check" ; then
-	if test x"$cross_compiling" = x"no" ; then
-		if test -e /dev/ptmx ; then
-
+#Why no config.cache checking?!
 $as_echo "#define USE_DEV_PTMX /**/" >>confdefs.h
 
-		fi
-	else
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: Not checking for /dev/ptmx, we're cross-compiling" >&5
-$as_echo "$as_me: Not checking for /dev/ptmx, we're cross-compiling" >&6;}
-	fi
-fi
+# Solaris needs ptmx
+#if test -z "$no_ptmx_check" ; then
+#	if test x"$cross_compiling" = x"no" ; then
+#		if test -e /dev/ptmx ; then
+#
+#$as_echo "#define USE_DEV_PTMX /**/" >>confdefs.h
+#
+#		fi
+#	else
+#		{ $as_echo "$as_me:${as_lineno-$LINENO}: Not checking for /dev/ptmx, we're cross-compiling" >&5
+#$as_echo "$as_me: Not checking for /dev/ptmx, we're cross-compiling" >&6;}
+#	fi
+#fi
 
 if test -z "$no_ptc_check" ; then
 	if test x"$cross_compiling" = x"no" ; then
diff -ru dropbear-2017.75.orig/includes.h dropbear-2017.75/includes.h
--- dropbear-2017.75.orig/includes.h	2017-05-18 16:47:02.000000000 +0200
+++ dropbear-2017.75/includes.h	2017-08-23 17:23:59.379752797 +0200
@@ -156,7 +156,7 @@
 typedef u_int32_t uint32_t;
 #endif /* HAVE_UINT32_T */
 
-#ifdef SO_PRIORITY
+#if defined(SO_PRIORITY) && !defined(__midipix__)
 #include <linux/types.h>
 #include <linux/pkt_sched.h>
 #endif
diff -ru dropbear-2017.75.orig/netio.c dropbear-2017.75/netio.c
--- dropbear-2017.75.orig/netio.c	2017-05-18 16:47:02.000000000 +0200
+++ dropbear-2017.75/netio.c	2017-08-23 17:25:34.759249812 +0200
@@ -333,7 +333,7 @@
 	}
 #endif
 
-#ifdef SO_PRIORITY
+#if defined(SO_PRIORITY) && !defined(__midipix__)
 	if (prio == DROPBEAR_PRIO_LOWDELAY) {
 		so_prio_val = TC_PRIO_INTERACTIVE;
 	} else if (prio == DROPBEAR_PRIO_BULK) {
diff -ru dropbear-2017.75.orig/options.h dropbear-2017.75/options.h
--- dropbear-2017.75.orig/options.h	2017-05-18 16:47:02.000000000 +0200
+++ dropbear-2017.75/options.h	2017-08-23 17:27:15.962716376 +0200
@@ -52,7 +52,7 @@
 several kB in binary size however will make the symmetrical ciphers and hashes
 slower, perhaps by 50%. Recommended for small systems that aren't doing
 much traffic. */
-#define DROPBEAR_SMALL_CODE
+/*#define DROPBEAR_SMALL_CODE*/ /* NO! */
 
 /* Enable X11 Forwarding - server only */
 #define ENABLE_X11FWD
@@ -347,7 +347,7 @@
 #define DEFAULT_IDLE_TIMEOUT 0
 
 /* The default path. This will often get replaced by the shell */
-#define DEFAULT_PATH "/usr/bin:/bin"
+#define DEFAULT_PATH "/bin"
 
 /* Some other defines (that mostly should be left alone) are defined
  * in sysoptions.h */
