--- findutils-4.5.19/configure.orig	2015-12-28 14:39:42.000000000 +0100
+++ findutils-4.5.19/configure	2016-12-18 00:49:18.483403200 +0100
@@ -55251,3 +55251,23 @@
 $as_echo "$as_me: WARNING: unrecognized options: $ac_unrecognized_opts" >&2;}
 fi
 
+
+cat >> config.h << EOF
+
+#define HAVE_ENDPWENT 1
+
+#undef fchownat
+#undef fchdir
+#undef fseeko
+#undef fstatat
+#undef getcwd
+#undef gmtime
+#undef localtime
+#undef lstat
+#undef malloc
+#undef memcmp
+#undef mktime
+#undef realloc
+#undef strnlen
+
+EOF
