diff -ru --new-file a/configure b/configure
--- a/configure	2014-07-19 06:24:44.000000000 -0400
+++ b/configure	2015-05-31 23:05:19.895583486 -0400
@@ -54948,3 +54950,23 @@
 $as_echo "$as_me: WARNING: unrecognized options: $ac_unrecognized_opts" >&2;}
 fi
 
+
+cat >> config.h << EOF
+
+#define HAVE_ENDPWENT 1
+
+#undef fchownat
+#undef fchdir
+#undef fseeko
+#undef fstatat
+#undef getcwd
+#undef gmtime
+#undef localtime
+#undef lstat
+#undef malloc
+#undef memcmp
+#undef mktime
+#undef realloc
+#undef strnlen
+
+EOF
