diff -Nru glew-2.1.0.orig/config/Makefile.midipix glew-2.1.0/config/Makefile.midipix
--- glew-2.1.0.orig/config/Makefile.midipix	1970-01-01 01:00:00.000000000 +0100
+++ glew-2.1.0/config/Makefile.midipix	2018-07-19 22:26:01.674783470 +0200
@@ -0,0 +1,17 @@
+NAME = $(GLEW_NAME)
+CC = gcc
+LD = gcc
+LDFLAGS.GL = -lGL -lX11
+NAME = GLEW
+WARN = -Wall -W
+POPT = -O2
+CFLAGS.EXTRA += -fPIC -I./include
+CFLAGS.EXTRA += -Wcast-qual
+CFLAGS.EXTRA += -ansi -pedantic
+CFLAGS.EXTRA += -fno-stack-protector
+BIN.SUFFIX =
+LIB.SONAME = lib$(NAME).so.$(SO_MAJOR)
+LIB.DEVLNK = lib$(NAME).so
+LIB.SHARED = lib$(NAME).so.$(SO_VERSION)
+LIB.STATIC = lib$(NAME).a
+LDFLAGS.SO = -shared -Wl,-soname=$(LIB.SONAME) -Wl,--out-dsolib,lib/lib$(NAME).lib.a
