--- glib-2.54.0/configure.orig	2017-09-11 02:18:49.000000000 +0200
+++ glib-2.54.0/configure	2017-09-20 19:03:31.573166756 +0200
@@ -16675,6 +16675,7 @@
 
 
 
+can_build_shared=yes;
   # Report what library types will actually be built
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking if libtool supports shared libraries" >&5
 $as_echo_n "checking if libtool supports shared libraries... " >&6; }
@@ -29484,7 +29485,7 @@
    -Wall -Wstrict-prototypes -Werror=declaration-after-statement \
    -Werror=missing-prototypes -Werror=implicit-function-declaration \
    -Werror=pointer-arith -Werror=init-self -Werror=format-security \
-   -Werror=format=2 -Werror=missing-include-dirs; do
+   -Werror=missing-include-dirs; do
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports flag $flag in envvar CFLAGS" >&5
 $as_echo_n "checking if $CC supports flag $flag in envvar CFLAGS... " >&6; }
--- glib-2.54.0/gio/gdbusmenumodel.c.orig	2017-07-14 01:03:38.000000000 +0200
+++ glib-2.54.0/gio/gdbusmenumodel.c	2017-09-20 19:03:31.565166706 +0200
@@ -625,7 +625,7 @@
   while (g_variant_iter_loop (&iter, "@a{sv}", &item))
     g_sequence_insert_before (point, g_dbus_menu_group_create_item (item));
 
-  if (g_sequence_is_empty (items))
+  if (g_sequence_get_length (items) == 0)
     {
       g_hash_table_remove (group->menus, GINT_TO_POINTER (menu_id));
       items = NULL;
--- glib-2.54.0/gio/gio-tool-mount.c.orig	2017-07-14 01:03:38.000000000 +0200
+++ glib-2.54.0/gio/gio-tool-mount.c	2017-09-20 19:03:31.565166706 +0200
@@ -780,7 +780,7 @@
               g_object_unref (icon);
             }
 
-          g_print ("%*sis_removable=%d\n", indent + 2, "", g_drive_is_removable (drive));
+          //g_print ("%*sis_removable=%d\n", indent + 2, "", g_drive_is_removable (drive));
           g_print ("%*sis_media_removable=%d\n", indent + 2, "", g_drive_is_media_removable (drive));
           g_print ("%*shas_media=%d\n", indent + 2, "", g_drive_has_media (drive));
           g_print ("%*sis_media_check_automatic=%d\n", indent + 2, "", g_drive_is_media_check_automatic (drive));
--- glib-2.54.0/gio/glocalfilemonitor.c.orig	2017-07-14 01:03:38.000000000 +0200
+++ glib-2.54.0/gio/glocalfilemonitor.c	2017-09-20 19:03:31.565166706 +0200
@@ -504,7 +504,7 @@
   g_mutex_lock (&fms->lock);
 
   /* Create events for any pending changes that are due to fire */
-  while (!g_sequence_is_empty (fms->pending_changes))
+  while (g_sequence_get_length (fms->pending_changes))
     {
       GSequenceIter *iter = g_sequence_get_begin_iter (fms->pending_changes);
       PendingChange *pending = g_sequence_get (iter);
@@ -572,7 +572,7 @@
       while ((event = g_queue_pop_head (&fms->event_queue)))
         queued_event_free (event);
 
-      g_assert (g_sequence_is_empty (fms->pending_changes));
+      g_assert (g_sequence_get_length (fms->pending_changes) == 0);
       g_assert (g_hash_table_size (fms->pending_changes_table) == 0);
       g_assert (fms->event_queue.length == 0);
       fms->instance = NULL;
@@ -592,7 +592,7 @@
 
   /* should already have been cleared in dispose of the monitor */
   g_assert (fms->instance == NULL);
-  g_assert (g_sequence_is_empty (fms->pending_changes));
+  g_assert (g_sequence_get_length (fms->pending_changes) == 0);
   g_assert (g_hash_table_size (fms->pending_changes_table) == 0);
   g_assert (fms->event_queue.length == 0);
 
--- glib-2.54.0/gio/gmenuexporter.c.orig	2017-07-14 01:03:38.000000000 +0200
+++ glib-2.54.0/gio/gmenuexporter.c	2017-09-20 19:03:31.569166731 +0200
@@ -377,7 +377,7 @@
       guint id = GPOINTER_TO_INT (key);
       GMenuExporterMenu *menu = val;
 
-      if (!g_sequence_is_empty (menu->item_links))
+      if (g_sequence_get_length (menu->item_links))
         {
           g_variant_builder_open (builder, G_VARIANT_TYPE ("(uuaa{sv})"));
           g_variant_builder_add (builder, "u", group->id);
--- glib-2.54.0/gio/inotify/Makefile.in.orig	2017-09-11 02:18:51.000000000 +0200
+++ glib-2.54.0/gio/inotify/Makefile.in	2017-09-20 19:03:31.561166681 +0200
@@ -180,7 +180,7 @@
 	"$(DESTDIR)$(installed_test_metadir)" \
 	"$(DESTDIR)$(installed_testdir)"
 LTLIBRARIES = $(installed_test_LTLIBRARIES) $(noinst_LTLIBRARIES)
-libinotify_la_LIBADD =
+libinotify_la_LIBADD = -L../../glib -lglib-2.0
 am__objects_1 =
 am_libinotify_la_OBJECTS = libinotify_la-inotify-kernel.lo \
 	libinotify_la-inotify-sub.lo libinotify_la-inotify-path.lo \
--- glib-2.54.0/glib-2.0.pc.in.orig	2016-10-22 07:17:22.000000000 +0200
+++ glib-2.54.0/glib-2.0.pc.in	2017-09-20 19:03:31.561166681 +0200
@@ -5,7 +5,7 @@
 
 glib_genmarshal=glib-genmarshal
 gobject_query=gobject-query
-glib_mkenums=glib-mkenums
+glib_mkenums=@exec_prefix@/bin/glib-mkenums
 
 Name: GLib
 Description: C Utility Library
