--- glib-2.46.2/gio/inotify/Makefile.in.orig	2015-11-06 23:28:59.000000000 +0100
+++ glib-2.46.2/gio/inotify/Makefile.in	2016-09-07 05:20:16.670049836 +0200
@@ -180,7 +180,7 @@
 	"$(DESTDIR)$(installed_test_metadir)" \
 	"$(DESTDIR)$(installed_testdir)"
 LTLIBRARIES = $(installed_test_LTLIBRARIES) $(noinst_LTLIBRARIES)
-libinotify_la_LIBADD =
+libinotify_la_LIBADD = -L../../glib -lglib-2.0
 am__objects_1 =
 am_libinotify_la_OBJECTS = libinotify_la-inotify-kernel.lo \
 	libinotify_la-inotify-sub.lo libinotify_la-inotify-path.lo \
--- glib-2.46.2/glib-2.0.pc.in.orig	2015-03-23 04:22:18.000000000 +0100
+++ glib-2.46.2/glib-2.0.pc.in	2017-08-25 22:20:03.833998381 +0200
@@ -5,7 +5,7 @@
 
 glib_genmarshal=glib-genmarshal
 gobject_query=gobject-query
-glib_mkenums=glib-mkenums
+glib_mkenums=@exec_prefix@/bin/glib-mkenums
 
 Name: GLib
 Description: C Utility Library
--- glib-2.46.2/configure.orig	2015-11-06 22:28:54.000000000 +0000
+++ glib-2.46.2/configure	2017-09-06 17:48:13.444000000 +0000
@@ -28873,7 +28873,7 @@
    -Wall -Wstrict-prototypes -Werror=declaration-after-statement \
    -Werror=missing-prototypes -Werror=implicit-function-declaration \
    -Werror=pointer-arith -Werror=init-self -Werror=format-security \
-   -Werror=format=2 -Werror=missing-include-dirs; do
+   -Werror=missing-include-dirs; do
 
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking if $CC supports flag $flag in envvar CFLAGS" >&5
 $as_echo_n "checking if $CC supports flag $flag in envvar CFLAGS... " >&6; }
