diff -ru gnutls-3.6.6.orig/lib/random.c gnutls-3.6.6/lib/random.c
--- gnutls-3.6.6.orig/lib/random.c	2018-04-01 13:56:13.000000000 +0200
+++ gnutls-3.6.6/lib/random.c	2019-01-31 20:38:57.815274977 +0100
@@ -36,8 +36,8 @@
 #endif
 
 /* Per thread context of random generator, and a flag to indicate initialization */
-static _Thread_local void *gnutls_rnd_ctx;
-static _Thread_local unsigned rnd_initialized = 0;
+static __thread void *gnutls_rnd_ctx;
+static __thread unsigned rnd_initialized = 0;
 
 struct rnd_ctx_list_st {
 	void *ctx;
