--- harfbuzz-1.7.5/src/hb-ot-kern-table.hh.orig	2018-02-27 21:36:40.772000000 +0000
+++ harfbuzz-1.7.5/src/hb-ot-kern-table.hh	2018-02-27 21:37:57.168000000 +0000
@@ -205,7 +205,7 @@
   {
     TRACE_SANITIZE (this);
     return_trace (c->check_struct (thiz()) &&
-		  thiz()->length >= thiz()->min_size &&
+		  thiz()->length >= T::min_size &&
 		  c->check_array (thiz(), 1, thiz()->length) &&
 		  thiz()->subtable.sanitize (c, thiz()->format));
   }
