--- htop-2.0.1/configure.orig	2016-03-07 21:12:35.000000000 +0000
+++ htop-2.0.1/configure	2016-07-10 14:39:42.654635796 +0000
@@ -13313,7 +13313,7 @@
    fi
    if test "x$htop_script_success" = xno; then
 
-   htop_config_script=$("ncursesw5-config" --libs 2> /dev/null)
+   htop_config_script=$("${PKG_PREFIX}/bin/ncursesw6-config" --libs 2> /dev/null)
    htop_script_success=no
    htop_save_LDFLAGS="$LDFLAGS"
    if test ! "x$htop_config_script" = x; then
