diff -ru httpd-2.4.52.orig/configure httpd-2.4.52/configure
--- httpd-2.4.52.orig/configure	2021-12-16 14:49:07.000000000 +0100
+++ httpd-2.4.52/configure	2022-02-18 17:49:53.294104051 +0100
@@ -6530,123 +6530,7 @@
   export CC; export CPP
 fi
 
-
-# Check whether --with-pcre was given.
-if test ${with_pcre+y}
-then :
-  withval=$with_pcre;
-fi
-
-
-# Extract the first word of "pcre-config", so it can be a program name with args.
-set dummy pcre-config; ac_word=$2
-{ printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
-printf %s "checking for $ac_word... " >&6; }
-if test ${ac_cv_path_PCRE_CONFIG+y}
-then :
-  printf %s "(cached) " >&6
-else $as_nop
-  case $PCRE_CONFIG in
-  [\\/]* | ?:[\\/]*)
-  ac_cv_path_PCRE_CONFIG="$PCRE_CONFIG" # Let the user override the test with a path.
-  ;;
-  *)
-  as_save_IFS=$IFS; IFS=$PATH_SEPARATOR
-for as_dir in $PATH
-do
-  IFS=$as_save_IFS
-  case $as_dir in #(((
-    '') as_dir=./ ;;
-    */) ;;
-    *) as_dir=$as_dir/ ;;
-  esac
-    for ac_exec_ext in '' $ac_executable_extensions; do
-  if as_fn_executable_p "$as_dir$ac_word$ac_exec_ext"; then
-    ac_cv_path_PCRE_CONFIG="$as_dir$ac_word$ac_exec_ext"
-    printf "%s\n" "$as_me:${as_lineno-$LINENO}: found $as_dir$ac_word$ac_exec_ext" >&5
-    break 2
-  fi
-done
-  done
-IFS=$as_save_IFS
-
-  test -z "$ac_cv_path_PCRE_CONFIG" && ac_cv_path_PCRE_CONFIG="false"
-  ;;
-esac
-fi
-PCRE_CONFIG=$ac_cv_path_PCRE_CONFIG
-if test -n "$PCRE_CONFIG"; then
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: $PCRE_CONFIG" >&5
-printf "%s\n" "$PCRE_CONFIG" >&6; }
-else
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-fi
-
-
-if test -d "$with_pcre" && test -x "$with_pcre/bin/pcre-config"; then
-   PCRE_CONFIG=$with_pcre/bin/pcre-config
-elif test -x "$with_pcre"; then
-   PCRE_CONFIG=$with_pcre
-fi
-
-if test "$PCRE_CONFIG" != "false"; then
-  if $PCRE_CONFIG --version >/dev/null 2>&1; then :; else
-    as_fn_error $? "Did not find pcre-config script at $PCRE_CONFIG" "$LINENO" 5
-  fi
-  case `$PCRE_CONFIG --version` in
-  [1-5].*)
-    as_fn_error $? "Need at least pcre version 6.0" "$LINENO" 5
-    ;;
-  esac
-  { printf "%s\n" "$as_me:${as_lineno-$LINENO}: Using external PCRE library from $PCRE_CONFIG" >&5
-printf "%s\n" "$as_me: Using external PCRE library from $PCRE_CONFIG" >&6;}
-
-  if test "x$PCRE_INCLUDES" = "x"; then
-    test "x$silent" != "xyes" && echo "  setting PCRE_INCLUDES to \"`$PCRE_CONFIG --cflags`\""
-    PCRE_INCLUDES="`$PCRE_CONFIG --cflags`"
-  else
-    apr_addto_bugger="`$PCRE_CONFIG --cflags`"
-    for i in $apr_addto_bugger; do
-      apr_addto_duplicate="0"
-      for j in $PCRE_INCLUDES; do
-        if test "x$i" = "x$j"; then
-          apr_addto_duplicate="1"
-          break
-        fi
-      done
-      if test $apr_addto_duplicate = "0"; then
-        test "x$silent" != "xyes" && echo "  adding \"$i\" to PCRE_INCLUDES"
-        PCRE_INCLUDES="$PCRE_INCLUDES $i"
-      fi
-    done
-  fi
-
-
-  if test "x$PCRE_LIBS" = "x"; then
-    test "x$silent" != "xyes" && echo "  setting PCRE_LIBS to \"`$PCRE_CONFIG --libs`\""
-    PCRE_LIBS="`$PCRE_CONFIG --libs`"
-  else
-    apr_addto_bugger="`$PCRE_CONFIG --libs`"
-    for i in $apr_addto_bugger; do
-      apr_addto_duplicate="0"
-      for j in $PCRE_LIBS; do
-        if test "x$i" = "x$j"; then
-          apr_addto_duplicate="1"
-          break
-        fi
-      done
-      if test $apr_addto_duplicate = "0"; then
-        test "x$silent" != "xyes" && echo "  adding \"$i\" to PCRE_LIBS"
-        PCRE_LIBS="$PCRE_LIBS $i"
-      fi
-    done
-  fi
-
-else
-  as_fn_error $? "pcre-config for libpcre not found. PCRE is required and available from http://pcre.org/" "$LINENO" 5
-fi
-
+  PCRE_LIBS=-lpcre
   APACHE_VAR_SUBST="$APACHE_VAR_SUBST PCRE_LIBS"
 
 
@@ -41250,9 +41134,9 @@
 
 
 if test x${apu_found} != xobsolete; then
-  AP_LIBS="$AP_LIBS `$apu_config --avoid-ldap --link-libtool --libs`"
+  AP_LIBS="$AP_LIBS \-lapr-1 \-laprutil-1"
 fi
-AP_LIBS="$AP_LIBS `$apr_config --link-libtool --libs`"
+AP_LIBS="$AP_LIBS \-lapr-1 \-laprutil-1"
 
   APACHE_VAR_SUBST="$APACHE_VAR_SUBST AP_LIBS"
 
