--- ImageMagick-7.0.9-12/configure.orig	2019-12-27 01:06:32.000000000 +0000
+++ ImageMagick-7.0.9-12/configure	2019-12-27 19:40:26.904000000 +0000
@@ -22639,13 +22639,7 @@
 
 # Only allow building loadable modules if we are building shared libraries
 if test "$with_modules" != 'no' ; then
-    if test "$libtool_build_shared_libs" = 'no'; then
-        { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: Modules may only be built if building shared libraries is enabled." >&5
-$as_echo "$as_me: WARNING: Modules may only be built if building shared libraries is enabled." >&2;}
-    build_modules='no'
-  else
     build_modules='yes'
-  fi
 fi
 if test "$build_modules" != 'no' ; then
 
