diff -ru indent-2.2.12.orig/Makefile.in indent-2.2.12/Makefile.in
--- indent-2.2.12.orig/Makefile.in	2020-11-15 00:08:42.898242760 +0100
+++ indent-2.2.12/Makefile.in	2020-11-15 09:33:23.686422521 +0100
@@ -378,7 +378,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = no-texinfo.tex
-SUBDIRS = intl src doc po man regression
+SUBDIRS = intl src doc po man
 BUILT_SOURCES = 
 DISTFILES = $(DIST_COMMON:README=README.md) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 EXTRA_DIST = README.md \
