--- infozip/Makefile.orig	2018-05-14 22:16:50.436000000 +0000
+++ infozip/Makefile	2018-06-06 09:34:02.456000000 +0000
@@ -45,16 +45,16 @@
 $(OBJC): $(ZIP_H)
 
 zip: $(OBJZ) $(OBJI)
-	$(CC) -o zip $(LFLAGS1) $(OBJZ) $(OBJI) $(LDFLAGS)
+	$(CC) -o zip $(LFLAGS1) $(LDFLAGS) $(OBJZ) $(OBJI) $(LDFLAGS)
 
 zipnote: $(OBJN)
-	$(CC) -o zipnote $(LFLAGS1) $(OBJN) $(LDFLAGS)
+	$(CC) -o zipnote $(LFLAGS1) $(LDFLAGS) $(OBJN) $(LDFLAGS)
 
 zipcloak: $(OBJC)
-	$(CC) -o zipcloak $(LFLAGS1) $(OBJC) $(LDFLAGS)
+	$(CC) -o zipcloak $(LFLAGS1) $(LDFLAGS) $(OBJC) $(LDFLAGS)
 
 zipsplit: $(OBJS)
-	$(CC) -o zipsplit $(LFLAGS1) $(OBJS) $(LDFLAGS)
+	$(CC) -o zipsplit $(LFLAGS1) $(LDFLAGS) $(OBJS) $(LDFLAGS)
 
 ZIPS = zip zipcloak zipnote zipsplit
 
