diff -ru ldns-1.7.0.orig/dnssec.c ldns-1.7.0/dnssec.c
--- ldns-1.7.0.orig/dnssec.c	2016-12-20 11:48:22.000000000 +0100
+++ ldns-1.7.0/dnssec.c	2022-03-04 03:18:52.759678825 +0100
@@ -375,7 +375,7 @@
 		BN_free(Y);
 		return NULL;
 	}
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 #ifndef S_SPLINT_S
 	dsa->p = P;
 	dsa->q = Q;
@@ -461,7 +461,7 @@
 		BN_free(modulus);
 		return NULL;
 	}
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 #ifndef S_SPLINT_S
 	rsa->n = modulus;
 	rsa->e = exponent;
diff -ru ldns-1.7.0.orig/host2str.c ldns-1.7.0/host2str.c
--- ldns-1.7.0.orig/host2str.c	2016-12-20 11:48:22.000000000 +0100
+++ ldns-1.7.0/host2str.c	2022-03-04 03:18:52.759678825 +0100
@@ -2040,7 +2040,7 @@
 					const BIGNUM *n=NULL, *e=NULL, *d=NULL,
 						*p=NULL, *q=NULL, *dmp1=NULL,
 						*dmq1=NULL, *iqmp=NULL;
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 					n = rsa->n;
 					e = rsa->e;
 					d = rsa->d;
@@ -2092,7 +2092,7 @@
 				if(1) {
 					const BIGNUM *p=NULL, *q=NULL, *g=NULL,
 						*priv_key=NULL, *pub_key=NULL;
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 #ifndef S_SPLINT_S
 					p = dsa->p;
 					q = dsa->q;
diff -ru ldns-1.7.0.orig/keys.c ldns-1.7.0/keys.c
--- ldns-1.7.0.orig/keys.c	2016-12-20 11:48:22.000000000 +0100
+++ ldns-1.7.0/keys.c	2022-03-04 03:18:52.759678825 +0100
@@ -898,7 +898,7 @@
 	}
 #endif /* splint */
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 # ifndef S_SPLINT_S
 	rsa->n = n;
 	rsa->e = e;
@@ -1018,7 +1018,7 @@
 	}
 #endif /* splint */
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 # ifndef S_SPLINT_S
 	dsa->p = p;
 	dsa->q = q;
@@ -1713,7 +1713,7 @@
 	if (!k) {
 		return false;
 	}
-#if OPENSSL_VERSION_NUMBER < 0x10100000 || defined(HAVE_LIBRESSL)
+#if OPENSSL_VERSION_NUMBER < 0x10100000 || LIBRESSL_VERSION_NUMBER < 0x3050000fL
 	n = k->n;
 	e = k->e;
 #else
Only in ldns-1.7.0: ldns-1.7.0.local.patch
diff -ru ldns-1.7.0.orig/Makefile.in ldns-1.7.0/Makefile.in
--- ldns-1.7.0.orig/Makefile.in	2016-12-20 11:48:22.000000000 +0100
+++ ldns-1.7.0/Makefile.in	2022-03-04 03:19:09.959679170 +0100
@@ -148,7 +148,7 @@
 	fi
 
 drill/drill: $(DRILL_LOBJS) $(LIB)
-	$(LINK_EXE) $(DRILL_LOBJS) $(LIBS) $(LIBSSL_LIBS) -lldns -o drill/drill
+	$(LINK_EXE) $(DRILL_LOBJS) $(LIBS) $(LIBSSL_LIBS) -L./lib -lldns -o drill/drill
 
 drill/drill.1: $(srcdir)/drill/drill.1.in
 	$(edit) $(srcdir)/drill/drill.1.in > drill/drill.1
@@ -179,21 +179,21 @@
 	fi
 
 $(EXAMPLE_PROGS):
-	$(LINK_EXE) $@.lo $(LIBS) -lldns -o $@
+	$(LINK_EXE) $@.lo $(LIBS) -L./lib -lldns -o $@
 
 $(TESTNS):
-	$(LINK_EXE) $(TESTNS_LOBJS) $(LIBS) -lldns -o $(TESTNS)
+	$(LINK_EXE) $(TESTNS_LOBJS) $(LIBS) -L./lib -lldns -o $(TESTNS)
 
 $(LDNS_DPA):
-	$(LINK_EXE) $(LDNS_DPA_LOBJS) $(LIBS) $(LIBPCAP_LIBS) -lldns \
+	$(LINK_EXE) $(LDNS_DPA_LOBJS) $(LIBS) $(LIBPCAP_LIBS) -L./lib -lldns \
 		 -o $(LDNS_DPA)
 
 $(LDNS_DANE):
-	$(LINK_EXE) $(LDNS_DANE_LOBJS) $(LIBS) $(LIBSSL_SSL_LIBS) -lldns \
+	$(LINK_EXE) $(LDNS_DANE_LOBJS) $(LIBS) $(LIBSSL_SSL_LIBS) -L./lib -lldns \
 		 -o $(LDNS_DANE)
 
 $(EX_SSL_PROGS):
-	$(LINK_EXE) $@.lo $(LIBS) $(LIBSSL_LIBS) -lldns -o $@
+	$(LINK_EXE) $@.lo $(LIBS) $(LIBSSL_LIBS) -L./lib -lldns -o $@
 
 examples/ldns-dane.1: $(srcdir)/examples/ldns-dane.1.in
 	$(edit) $(srcdir)/examples/ldns-dane.1.in > examples/ldns-dane.1
@@ -228,14 +228,14 @@
 
 linktest: $(srcdir)/linktest.c libldns.la
 	$(COMP_LIB) $(LIBSSL_CPPFLAGS) -c $(srcdir)/linktest.c -o linktest.lo
-	$(LINK_EXE) linktest.lo $(LIBS) $(LIBSSL_LIBS) -lldns -o linktest
+	$(LINK_EXE) linktest.lo $(LIBS) $(LIBSSL_LIBS) -L./lib -lldns -o linktest
 
 lib: libldns.la
 
 lib-export-all: libldns.la-export-all
 
 libldns.la: $(LDNS_ALL_LOBJS)
-	$(LINK_LIB) $(LIBSSL_LDFLAGS) $(LIBSSL_LIBS) -export-symbols-regex '^(ldns_|b32_[pn]to[pn]|mktime_from_utc|qsort_rr_compare_nsec3)' -o libldns.la $(LDNS_ALL_LOBJS) -rpath $(libdir) $(RUNTIME_PATH)
+	$(LINK_LIB) $(LIBSSL_LDFLAGS) -export-symbols-regex '^(ldns_|b32_[pn]to[pn]|mktime_from_utc|qsort_rr_compare_nsec3)' -o libldns.la $(LDNS_ALL_LOBJS) $(LIBSSL_LIBS) -rpath $(libdir) $(RUNTIME_PATH)
 
 libldns.la-export-all: $(LDNS_ALL_LOBJS)
 	$(LINK_LIB) -o libldns.la $(LDNS_ALL_LOBJS) -rpath $(libdir) $(RUNTIME_PATH)
