diff -ru libarchive-3.7.2.orig/configure libarchive-3.7.2/configure
--- libarchive-3.7.2.orig/configure	2023-09-12 00:05:34.000000000 +0200
+++ libarchive-3.7.2/configure	2024-02-26 13:44:22.491965148 +0100
@@ -12906,6 +12906,15 @@
   dynamic_linker='NetBSD ld.elf_so'
   ;;
 
+midipix*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 netbsd*)
   version_type=sunos
   need_lib_prefix=no
@@ -16925,7 +16935,7 @@
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_LIBXML2_PC_LIBS=`$PKG_CONFIG --libs "libxml-2.0" 2>/dev/null`
+  pkg_cv_LIBXML2_PC_LIBS=`$PKG_CONFIG --libs-only-l "libxml-2.0" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
