diff -ru a/src/libassuan-config.in b/src/libassuan-config.in
--- a/src/libassuan-config.in	2018-10-25 05:20:28.000000000 -0400
+++ b/src/libassuan-config.in	2020-03-11 18:47:54.410306584 -0400
@@ -131,13 +131,5 @@
 fi
 
 if test "$echo_libs" = "yes"; then
-    if test "@libdir@" != "/usr/lib" ; then
-      libdirs="-L@libdir@"
-      for i in $lib $extralibs ; do
-	if test "$i" = "-L@libdir@" ; then
-	  libdirs=""
-	fi
-      done
-    fi
     echo $libdirs $lib $extralibs
 fi
