--- libcaca/bootstrap.orig	2020-02-09 17:08:18.969163074 +0000
+++ libcaca/bootstrap	2020-02-09 18:07:03.387297417 +0000
@@ -84,7 +84,7 @@
 
 # Check for pkg-config
 if test "$pkgconfig" = "yes"; then
-  if ! pkg-config --version >/dev/null 2>&1; then
+  if ! $PKG_CONFIG --version >/dev/null 2>&1; then
     echo "$0: pkg-config not found"
     exit 1
   fi
--- libcaca/configure.ac.orig	2021-10-21 13:53:20.773431316 +0200
+++ libcaca/configure.ac	2021-10-21 14:01:22.878201384 +0200
@@ -1,6 +1,6 @@
 
 AC_INIT([libcaca],[0.99.beta20])
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 AC_CONFIG_AUX_DIR(.auto)
 AC_CANONICAL_TARGET
 AM_INIT_AUTOMAKE([foreign no-define tar-ustar silent-rules subdir-objects])
