diff -ru libgcrypt-1.8.2.orig/configure libgcrypt-1.8.2/configure
--- libgcrypt-1.8.2.orig/configure	2017-12-13 14:54:17.000000000 +0100
+++ libgcrypt-1.8.2/configure	2018-05-05 00:17:43.366304238 +0200
@@ -13074,6 +13074,10 @@
         PRINTABLE_OS_NAME="GNU/Linux"
         ;;
 
+    *-midipix*)
+        PRINTABLE_OS_NAME="Midipix"
+        ;;
+
     *)
         PRINTABLE_OS_NAME=`uname -s || echo "Unknown"`
         ;;
