diff -ru libpcap-libpcap-1.9.0.orig/configure libpcap-libpcap-1.9.0/configure
--- libpcap-libpcap-1.9.0.orig/configure	2018-07-22 19:08:39.000000000 +0200
+++ libpcap-libpcap-1.9.0/configure	2019-04-26 05:17:01.024964350 +0200
@@ -3997,7 +3997,7 @@
 	    aix*)
 		    ;;
 
-	    freebsd*|netbsd*|openbsd*|dragonfly*|linux*|osf*)
+	    freebsd*|netbsd*|openbsd*|dragonfly*|linux*|osf*|midipix*)
 	    	    #
 		    # Platforms where the linker is the GNU linker
 		    # or accepts command-line arguments like
@@ -8179,7 +8179,7 @@
 	MAN_MISC_INFO=5
 	;;
 
-linux*|freebsd*|netbsd*|openbsd*|dragonfly*|kfreebsd*|gnu*)
+linux*|freebsd*|netbsd*|openbsd*|dragonfly*|kfreebsd*|gnu*|midipix*)
 	DYEXT="so"
 
 	#
