diff -ru cyrus-sasl-2.1.27.orig/common/Makefile.in cyrus-sasl-2.1.27/common/Makefile.in
--- cyrus-sasl-2.1.27.orig/common/Makefile.in	2018-10-09 16:58:13.000000000 +0200
+++ cyrus-sasl-2.1.27/common/Makefile.in	2021-06-01 21:00:34.299354616 +0200
@@ -428,6 +428,7 @@
 # See <http://www.gnu.org/software/libtool/manual/libtool.html#Versioning>
 # CURRENT:REVISION:AGE
 plugin_common_version = 3:0:0
+crypto_compat_version = 0:0:0
 AM_CPPFLAGS = -fPIC -I$(top_srcdir)/include -I$(top_builddir)/include
 noinst_LTLIBRARIES = libplugin_common.la libcrypto_compat.la
 libplugin_common_la_SOURCES = plugin_common.c plugin_common.h
