diff -ru libtheora-1.1.1.orig/Makefile.in libtheora-1.1.1/Makefile.in
--- libtheora-1.1.1.orig/Makefile.in    2009-10-01 20:04:07.000000000 +0200
+++ libtheora-1.1.1/Makefile.in 2020-12-10 14:17:43.482548317 +0100
@@ -164,7 +164,7 @@
 @THEORA_ENABLE_EXAMPLES_TRUE@EXAMPLES_DIR = examples
 @THEORA_ENABLE_EXAMPLES_FALSE@EXAMPLES_DIR =

-SUBDIRS = lib include doc tests m4 $(EXAMPLES_DIR)
+SUBDIRS = lib include tests m4 $(EXAMPLES_DIR)

 # we include the whole debian/ dir in EXTRA_DIST because there's a problem
 # with autotools and HFS+ MacOSX file systems that caused debian/Makefile.am
diff -ru libtheora-1.1.1.orig/lib/Makefile.in libtheora-1.1.1/lib/Makefile.in
--- libtheora-1.1.1.orig/lib/Makefile.in	2009-10-01 20:04:07.000000000 +0200
+++ libtheora-1.1.1/lib/Makefile.in	2018-08-04 17:38:41.912201898 +0200
@@ -179,7 +179,7 @@
 	x86_vc
 
 
-lib_LTLIBRARIES = libtheoradec.la libtheoraenc.la libtheora.la
+lib_LTLIBRARIES = libtheoradec.la libtheora.la libtheoraenc.la
 
 @THEORA_DISABLE_ENCODE_TRUE@encoder_uniq_sources = \
 @THEORA_DISABLE_ENCODE_TRUE@	encoder_disabled.c
@@ -357,7 +357,7 @@
 libtheoradec_la_LIBADD =
 am_libtheoradec_la_OBJECTS = $(am__objects_3)
 libtheoradec_la_OBJECTS = $(am_libtheoradec_la_OBJECTS)
-libtheoraenc_la_LIBADD =
+libtheoraenc_la_LIBADD = $(top_srcdir)/lib/libtheora.la
 @THEORA_DISABLE_ENCODE_FALSE@am__objects_8 = mmxfrag.lo mmxidct.lo \
 @THEORA_DISABLE_ENCODE_FALSE@	mmxstate.lo x86state.lo
 @THEORA_DISABLE_ENCODE_FALSE@am__objects_9 =
