diff -ru libtirpc-1.0.3.orig/src/bindresvport.c libtirpc-1.0.3/src/bindresvport.c
--- libtirpc-1.0.3.orig/src/bindresvport.c	2018-03-14 14:55:12.000000000 +0100
+++ libtirpc-1.0.3/src/bindresvport.c	2018-08-04 19:37:31.972078008 +0200
@@ -61,7 +61,7 @@
         return bindresvport_sa(sd, (struct sockaddr *)sin);
 }
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__midipix__)
 
 #define STARTPORT 600
 #define LOWPORT 512
diff -ru libtirpc-1.0.3.orig/src/clnt_dg.c libtirpc-1.0.3/src/clnt_dg.c
--- libtirpc-1.0.3.orig/src/clnt_dg.c	2018-03-14 14:55:12.000000000 +0100
+++ libtirpc-1.0.3/src/clnt_dg.c	2018-08-04 19:41:06.325266026 +0200
@@ -54,12 +54,16 @@
 #include <err.h>
 #include "rpc_com.h"
 
-#ifdef IP_RECVERR
+#if defined(IP_RECVERR) && !defined(__midipix__)
 #include <asm/types.h>
 #include <linux/errqueue.h>
 #include <sys/uio.h>
 #endif
 
+#ifdef __midipix__
+#include <sys/uio.h>
+#endif
+
 
 #define MAX_DEFAULT_FDS                 20000
 
@@ -421,7 +425,7 @@
                 }
                 break;
         }
-#ifdef IP_RECVERR
+#if defined(IP_RECVERR) && !defined(__midipix__)
       if (fd.revents & POLLERR)
 	{
 	  struct msghdr msg;
diff -ru libtirpc-1.0.3.orig/src/rtime.c libtirpc-1.0.3/src/rtime.c
--- libtirpc-1.0.3.orig/src/rtime.c	2018-03-14 14:55:12.000000000 +0100
+++ libtirpc-1.0.3/src/rtime.c	2018-08-04 19:47:49.089270334 +0200
@@ -46,7 +46,7 @@
 #include <unistd.h>
 #include <errno.h>
 #include <sys/types.h>
-#include <sys/poll.h>
+#include <poll.h>
 #include <sys/socket.h>
 #include <sys/time.h>
 #include <netinet/in.h>
diff -ru libtirpc-1.0.3.orig/src/svc_run.c libtirpc-1.0.3/src/svc_run.c
--- libtirpc-1.0.3.orig/src/svc_run.c	2018-03-14 14:55:12.000000000 +0100
+++ libtirpc-1.0.3/src/svc_run.c	2018-08-04 19:48:02.569155655 +0200
@@ -37,7 +37,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <unistd.h>
-#include <sys/poll.h>
+#include <poll.h>
 
 
 #include <rpc/rpc.h>
diff -ru libtirpc-1.0.3.orig/tirpc/reentrant.h libtirpc-1.0.3/tirpc/reentrant.h
--- libtirpc-1.0.3.orig/tirpc/reentrant.h	2018-03-14 14:55:12.000000000 +0100
+++ libtirpc-1.0.3/tirpc/reentrant.h	2018-08-04 19:23:48.165632436 +0200
@@ -36,7 +36,7 @@
  * These definitions are only guaranteed to be valid on Linux. 
  */
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__midipix__)
 
 #include <pthread.h>
 
