diff -ru libtasn1-4.16.0.orig/configure libtasn1-4.16.0/configure
--- libtasn1-4.16.0.orig/configure	2020-02-01 23:22:23.000000000 +0100
+++ libtasn1-4.16.0/configure	2020-04-12 15:39:06.225324685 +0200
@@ -12621,6 +12621,16 @@
   dynamic_linker='NetBSD ld.elf_so'
   ;;
 
+midipix*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
+  shlibpath_var=LD_LIBRARY_PATH
+  ;;
+
 netbsd*)
   version_type=sunos
   need_lib_prefix=no
diff -ru libtasn1-4.16.0.orig/Makefile.in libtasn1-4.16.0/Makefile.in
--- libtasn1-4.16.0.orig/Makefile.in	2020-02-01 23:22:24.000000000 +0100
+++ libtasn1-4.16.0/Makefile.in	2020-04-12 22:44:21.614252148 +0200
@@ -557,7 +557,7 @@
 	windows/test-tree/test-tree.vcproj windows/libtasn14win.mk \
 	CONTRIBUTING.md cfg.mk maint.mk AUTHORS NEWS ChangeLog THANKS \
 	LICENSE cfg.mk maint.mk .clcopying README.md
-SUBDIRS = lib src fuzz tests $(am__append_1)
+SUBDIRS = lib src fuzz $(am__append_1)
 @CODE_COVERAGE_ENABLED_TRUE@GITIGNOREFILES = $(GITIGNOREFILES) $(CODE_COVERAGE_OUTPUT_FILE) $(CODE_COVERAGE_OUTPUT_DIRECTORY)
 @CODE_COVERAGE_ENABLED_TRUE@code_coverage_v_lcov_cap = $(code_coverage_v_lcov_cap_$(V))
 @CODE_COVERAGE_ENABLED_TRUE@code_coverage_v_lcov_cap_ = $(code_coverage_v_lcov_cap_$(AM_DEFAULT_VERBOSITY))
