diff -ru libxslt-1.1.32.orig/Makefile.in libxslt-1.1.32/Makefile.in
--- libxslt-1.1.32.orig/Makefile.in	2017-11-02 21:34:22.000000000 +0100
+++ libxslt-1.1.32/Makefile.in	2018-08-04 15:15:09.283854093 +0200
@@ -415,10 +415,7 @@
 SUBDIRS = \
 	libxslt \
 	libexslt \
-	xsltproc \
-	doc \
-	$(PYTHON_SUBDIR) \
-	tests
+	xsltproc
 
 DIST_SUBDIRS = libxslt libexslt xsltproc python doc tests
 confexecdir = $(libdir)
