diff -ru lighttpd-1.4.65.orig/configure lighttpd-1.4.65/configure
--- lighttpd-1.4.65.orig/configure	2022-06-29 20:12:39.278502239 +0200
+++ lighttpd-1.4.65/configure	2022-06-29 20:18:18.776131546 +0200
@@ -18859,7 +18859,7 @@
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_PCRE2_LIBS=`$PKG_CONFIG --libs "libpcre2-8" 2>/dev/null`
+  pkg_cv_PCRE2_LIBS=`$PKG_CONFIG --libs-only-l "libpcre2-8" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
