diff -ru lighttpd-1.4.70.orig/src/Makefile.in lighttpd-1.4.70/src/Makefile.in
--- lighttpd-1.4.70.orig/src/Makefile.in        2023-05-10 09:21:16.000000000 +0200
+++ lighttpd-1.4.70/src/Makefile.in     2023-05-21 14:12:46.984527740 +0200
@@ -1469,7 +1469,7 @@
 @NO_RDYNAMIC_TRUE@common_ldflags = -avoid-version -no-undefined
 @NO_RDYNAMIC_TRUE@liblightcomp_la_SOURCES = $(common_src)
 @NO_RDYNAMIC_TRUE@liblightcomp_la_CFLAGS = $(AM_CFLAGS) $(LIBEV_CFLAGS)
-@NO_RDYNAMIC_TRUE@liblightcomp_la_LDFLAGS = $(common_ldflags) --export-all-symbols
+@NO_RDYNAMIC_TRUE@liblightcomp_la_LDFLAGS = $(common_ldflags)
 @NO_RDYNAMIC_TRUE@liblightcomp_la_LIBADD = $(PCRE_LIB) $(CRYPTO_LIB) $(FAM_LIBS) $(LIBEV_LIBS) $(ATTR_LIB) $(WS2_32_LIB)
 @NO_RDYNAMIC_FALSE@common_libadd =
 @NO_RDYNAMIC_TRUE@common_libadd = liblightcomp.la
diff -ru lighttpd-1.4.68.orig/configure lighttpd-1.4.68/configure
--- lighttpd-1.4.68.orig/configure	2023-01-03 14:05:40.000000000 +0100
+++ lighttpd-1.4.68/configure	2023-01-07 13:13:44.702656206 +0100
@@ -18388,7 +18388,7 @@
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_GNUTLS_LIBS=`$PKG_CONFIG --libs "gnutls >= 3.3.0" 2>/dev/null`
+  pkg_cv_GNUTLS_LIBS=`$PKG_CONFIG --libs-only-l "gnutls >= 3.3.0" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
@@ -18771,7 +18771,7 @@
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_PCRE2_LIBS=`$PKG_CONFIG --libs "libpcre2-8" 2>/dev/null`
+  pkg_cv_PCRE2_LIBS=`$PKG_CONFIG --libs-only-l "libpcre2-8" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
