Only in lighttpd-1.4.63/src: Makefile.
diff -ru lighttpd-1.4.63.orig/src/Makefile.in lighttpd-1.4.63/src/Makefile.in
--- lighttpd-1.4.63.orig/src/Makefile.in	2021-12-04 15:40:42.000000000 +0100
+++ lighttpd-1.4.63/src/Makefile.in	2021-12-11 13:14:04.707392975 +0100
@@ -1858,7 +1858,7 @@
 @LIGHTTPD_STATIC_TRUE@	$(am__append_45) $(am__append_48) \
 @LIGHTTPD_STATIC_TRUE@	$(am__append_53) $(am__append_56) \
 @LIGHTTPD_STATIC_TRUE@	$(am__append_59) $(am__append_61)
-@LIGHTTPD_STATIC_FALSE@lighttpd_LDADD = $(PCRE_LIB) $(DL_LIB) $(SENDFILE_LIB) $(ATTR_LIB) $(common_libadd) $(CRYPTO_LIB) $(XXHASH_LIBS) $(FAM_LIBS) $(LIBEV_LIBS) $(LIBUNWIND_LIBS)
+@LIGHTTPD_STATIC_FALSE@lighttpd_LDADD = $(common_libadd) $(PCRE_LIB) $(DL_LIB) $(SENDFILE_LIB) $(ATTR_LIB) $(CRYPTO_LIB) $(XXHASH_LIBS) $(FAM_LIBS) $(LIBEV_LIBS) $(LIBUNWIND_LIBS)
 @LIGHTTPD_STATIC_TRUE@lighttpd_LDADD = $(common_libadd) $(CRYPT_LIB) \
 @LIGHTTPD_STATIC_TRUE@	$(CRYPTO_LIB) $(XXHASH_LIBS) $(XML_LIBS) \
 @LIGHTTPD_STATIC_TRUE@	$(SQLITE_LIBS) $(UUID_LIBS) $(ELFTC_LIB) \
