diff -ru lighttpd-1.4.70.orig/src/Makefile.in lighttpd-1.4.70/src/Makefile.in
--- lighttpd-1.4.70.orig/src/Makefile.in        2023-05-10 09:21:16.000000000 +0200
+++ lighttpd-1.4.70/src/Makefile.in     2023-05-21 14:12:46.984527740 +0200
@@ -1469,7 +1469,7 @@
 @NO_RDYNAMIC_TRUE@common_ldflags = -avoid-version -no-undefined
 @NO_RDYNAMIC_TRUE@liblightcomp_la_SOURCES = $(common_src)
 @NO_RDYNAMIC_TRUE@liblightcomp_la_CFLAGS = $(AM_CFLAGS) $(LIBEV_CFLAGS)
-@NO_RDYNAMIC_TRUE@liblightcomp_la_LDFLAGS = $(common_ldflags) --export-all-symbols
+@NO_RDYNAMIC_TRUE@liblightcomp_la_LDFLAGS = $(common_ldflags)
 @NO_RDYNAMIC_TRUE@liblightcomp_la_LIBADD = $(PCRE_LIB) $(CRYPTO_LIB) $(FAM_LIBS) $(LIBEV_LIBS) $(ATTR_LIB) $(WS2_32_LIB)
 @NO_RDYNAMIC_FALSE@common_libadd =
 @NO_RDYNAMIC_TRUE@common_libadd = liblightcomp.la
