--- lighttpd-1.4.45/src/network.c.orig	2017-01-14 05:05:17.000000000 +0000
+++ lighttpd-1.4.45/src/network.c	2018-02-17 20:39:33.920000000 +0000
@@ -53,6 +53,7 @@
 void
 network_accept_tcp_nagle_disable (const int fd)
 {
+#ifndef __midipix__
     static int noinherit_tcpnodelay = -1;
     int opt;
 
@@ -70,6 +71,7 @@
 
     opt = 1;
     (void)setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &opt, sizeof(opt));
+#endif
 }
 
 static handler_t network_server_handle_fdevent(server *srv, void *context, int revents) {
