diff -ru links-2.29.orig/configure links-2.29/configure
--- links-2.29.orig/configure	2023-03-18 17:13:07.000000000 +0100
+++ links-2.29/configure	2023-12-06 15:01:23.081475928 +0100
@@ -7178,7 +7178,7 @@
 
             echo $ac_n "checking OPENSSL_LIBS""... $ac_c" 1>&6
 echo "configure:7183: checking OPENSSL_LIBS" >&5
-            OPENSSL_LIBS=`$PKG_CONFIG --libs "openssl"`
+            OPENSSL_LIBS=`$PKG_CONFIG --libs-only-l "openssl"`
             echo "$ac_t""$OPENSSL_LIBS" 1>&6
         else
             OPENSSL_CFLAGS=""
