diff -ru ltris-1.2.2.orig/configure ltris-1.2.2/configure
--- ltris-1.2.2.orig/configure	2020-12-28 12:58:30.000000000 +0100
+++ ltris-1.2.2/configure	2021-01-12 16:47:35.063828462 +0100
@@ -9089,8 +9089,8 @@
   if test "$SDL_CONFIG" = "no" ; then
     no_sdl=yes
   else
-    SDL_CFLAGS=`$SDL_CONFIG $sdlconf_args --cflags`
-    SDL_LIBS=`$SDL_CONFIG $sdlconf_args --libs`
+    SDL_CFLAGS=`$PKG_CONFIG --cflags sdl`
+    SDL_LIBS=`$PKG_CONFIG --libs sdl`
 
     sdl_major_version=`$SDL_CONFIG $sdl_args --version | \
            sed 's/\([0-9]*\).\([0-9]*\).\([0-9]*\)/\1/'`
@@ -9249,8 +9249,8 @@
 
   rm -f conf.sdltest
 
-CFLAGS="$CFLAGS $(sdl-config --cflags)"
-LIBS="$LIBS $(sdl-config --libs)"
+CFLAGS="$CFLAGS $(PKG_CONFIG --cflags sdl)"
+LIBS="$LIBS $(PKG_CONFIG --libs sdl)"
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for main in -lSDL_mixer" >&5
 $as_echo_n "checking for main in -lSDL_mixer... " >&6; }
