diff -ru mailutils-3.4.orig/imap4d/Makefile.am mailutils-3.4/imap4d/Makefile.am
--- mailutils-3.4.orig/imap4d/Makefile.am	2017-06-08 15:08:20.000000000 +0200
+++ mailutils-3.4/imap4d/Makefile.am	2018-05-18 18:00:06.234274714 +0200
@@ -78,8 +78,9 @@
  ${MU_LIB_MAILDIR}\
  ${MU_LIB_AUTH}\
  @MU_AUTHLIBS@ \
+ @MU_TCPWRAP_LIBRARIES@\
  ${MU_LIB_MAILUTILS}\
- @SERV_AUTHLIBS@ @MU_COMMON_LIBRARIES@ @MU_TCPWRAP_LIBRARIES@
+ @SERV_AUTHLIBS@ @MU_COMMON_LIBRARIES@
 
 ## This kludge is necessary to correctly establish imap4d -> MU_AUTHLIBS
 ## dependencies. Automake stupidly refuses to include them.
diff -ru mailutils-3.4.orig/lib/Makefile.am mailutils-3.4/lib/Makefile.am
--- mailutils-3.4.orig/lib/Makefile.am	2017-06-08 15:08:20.000000000 +0200
+++ mailutils-3.4/lib/Makefile.am	2018-05-18 17:47:07.863936646 +0200
@@ -35,7 +35,7 @@
  strexit.c\
  mu_umaxtostr.c\
  mu_umaxtostr.h
-libmuaux_la_LIBADD=gnu/libgnu.la
+libmuaux_la_LIBADD = gnu/libgnu.la ../libmailutils/libmailutils.la
 libmuaux_la_LDFLAGS = -version-info @VI_CURRENT@:@VI_REVISION@:@VI_AGE@
 
 libmutcpwrap_a_SOURCES = tcpwrap.c
diff -ru mailutils-3.4.orig/maidag/Makefile.am mailutils-3.4/maidag/Makefile.am
--- mailutils-3.4.orig/maidag/Makefile.am	2017-06-08 15:08:22.000000000 +0200
+++ mailutils-3.4/maidag/Makefile.am	2018-05-18 18:03:29.831157165 +0200
@@ -44,6 +44,7 @@
  ${MU_LIB_AUTH}\
  ${MU_LIB_MAILER}\
  @MU_AUTHLIBS@\
+ @MU_TCPWRAP_LIBRARIES@\
  ${MU_LIB_MAILUTILS} \
  @MU_COMMON_LIBRARIES@\
  @LIBMU_SCM@\
@@ -52,8 +53,7 @@
  $(LIBMU_DBM)\
  @GUILE_LIBS@\
  @PYTHON_LIBS@\
- @DBMLIBS@\
- @MU_TCPWRAP_LIBRARIES@
+ @DBMLIBS@
 
 AM_CPPFLAGS = -I${top_srcdir} @MU_APP_COMMON_INCLUDES@ @GUILE_INCLUDES@ \
  @PYTHON_INCLUDES@
diff -ru mailutils-3.4.orig/pop3d/Makefile.am mailutils-3.4/pop3d/Makefile.am
--- mailutils-3.4.orig/pop3d/Makefile.am	2017-06-08 15:08:22.000000000 +0200
+++ mailutils-3.4/pop3d/Makefile.am	2018-05-18 17:55:20.005420723 +0200
@@ -57,11 +57,11 @@
  ${MU_LIB_MAILDIR}\
  ${MU_LIB_AUTH}\
  @MU_AUTHLIBS@ \
+ @MU_TCPWRAP_LIBRARIES@\
  ${MU_LIB_MAILUTILS}\
  @MU_COMMON_LIBRARIES@\
  ${LIBMU_DBM}\
- @DBMLIBS@\
- @MU_TCPWRAP_LIBRARIES@
+ @DBMLIBS@
 
 popauth_SOURCES = popauth.c
 popauth_LDADD = \
