diff -ru mesa-18.0.0.orig/src/gallium/auxiliary/os/os_misc.c mesa-18.0.0/src/gallium/auxiliary/os/os_misc.c
--- mesa-18.0.0.orig/src/gallium/auxiliary/os/os_misc.c	2018-03-27 19:35:14.000000000 +0200
+++ mesa-18.0.0/src/gallium/auxiliary/os/os_misc.c	2018-03-31 17:16:02.843416010 +0200
@@ -47,7 +47,7 @@
 #endif
 
 
-#if defined(PIPE_OS_LINUX) || defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_SOLARIS)
+#if defined(PIPE_OS_LINUX) || defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_SOLARIS) || defined(PIPE_OS_MIDIPIX)
 #  include <unistd.h>
 #elif defined(PIPE_OS_APPLE) || defined(PIPE_OS_BSD)
 #  include <sys/sysctl.h>
@@ -124,7 +124,7 @@
 bool
 os_get_total_physical_memory(uint64_t *size)
 {
-#if defined(PIPE_OS_LINUX) || defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_SOLARIS)
+#if defined(PIPE_OS_LINUX) || defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_SOLARIS) || defined(PIPE_OS_MIDIPIX)
    const long phys_pages = sysconf(_SC_PHYS_PAGES);
    const long page_size = sysconf(_SC_PAGE_SIZE);
 
diff -ru mesa-18.0.0.orig/src/gallium/auxiliary/rtasm/rtasm_x86sse.h mesa-18.0.0/src/gallium/auxiliary/rtasm/rtasm_x86sse.h
--- mesa-18.0.0.orig/src/gallium/auxiliary/rtasm/rtasm_x86sse.h	2018-03-27 19:35:14.000000000 +0200
+++ mesa-18.0.0/src/gallium/auxiliary/rtasm/rtasm_x86sse.h	2018-03-31 17:12:28.412711185 +0200
@@ -140,7 +140,7 @@
 {
 #ifdef PIPE_ARCH_X86
    return X86_32;
-#elif (defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_WINDOWS)) && defined(PIPE_ARCH_X86_64)
+#elif (defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_WINDOWS) || defined(PIPE_OS_MIDIPIX)) && defined(PIPE_ARCH_X86_64)
    return X86_64_WIN64_ABI;
 #elif defined(PIPE_ARCH_X86_64)
    return X86_64_STD_ABI;
diff -ru mesa-18.0.0.orig/src/gallium/auxiliary/util/u_network.c mesa-18.0.0/src/gallium/auxiliary/util/u_network.c
--- mesa-18.0.0.orig/src/gallium/auxiliary/util/u_network.c	2018-03-27 19:35:14.000000000 +0200
+++ mesa-18.0.0/src/gallium/auxiliary/util/u_network.c	2018-03-31 17:14:07.064114855 +0200
@@ -10,7 +10,7 @@
 #  include <windows.h>
 #  include <ws2tcpip.h>
 #elif defined(PIPE_OS_LINUX) || defined(PIPE_OS_HAIKU) || \
-   defined(PIPE_OS_APPLE) || defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_SOLARIS)
+   defined(PIPE_OS_APPLE) || defined(PIPE_OS_CYGWIN) || defined(PIPE_OS_SOLARIS) || defined(PIPE_OS_MIDIPIX)
 #  include <sys/socket.h>
 #  include <netinet/in.h>
 #  include <unistd.h>
diff -ru mesa-18.0.0.orig/src/gallium/include/pipe/p_config.h mesa-18.0.0/src/gallium/include/pipe/p_config.h
--- mesa-18.0.0.orig/src/gallium/include/pipe/p_config.h	2018-03-27 19:35:15.000000000 +0200
+++ mesa-18.0.0/src/gallium/include/pipe/p_config.h	2018-03-31 17:16:40.779187231 +0200
@@ -213,6 +213,11 @@
 #define PIPE_OS_UNIX
 #endif
 
+#if defined(__midipix__)
+#define PIPE_OS_MIDIPIX
+#define PIPE_OS_UNIX
+#endif
+
 /*
  * Try to auto-detect the subsystem.
  * 
diff -ru mesa-18.0.0.orig/src/util/xmlconfig.c mesa-18.0.0/src/util/xmlconfig.c
--- mesa-18.0.0.orig/src/util/xmlconfig.c	2018-03-27 19:35:15.000000000 +0200
+++ mesa-18.0.0/src/util/xmlconfig.c	2018-03-31 17:36:30.362136400 +0200
@@ -40,7 +40,7 @@
 
 #undef GET_PROGRAM_NAME
 
-#if (defined(__GNU_LIBRARY__) || defined(__GLIBC__)) && !defined(__UCLIBC__)
+#if (defined(__GNU_LIBRARY__) || defined(__GLIBC__) || defined(__midipix__)) && !defined(__UCLIBC__)
 #    if !defined(__GLIBC__) || (__GLIBC__ < 2)
 /* These aren't declared in any libc5 header */
 extern char *program_invocation_name, *program_invocation_short_name;
