--- minizip-ng-2.10.4/CMakeLists.txt.orig	2020-12-07 00:32:18.000000000 +0000
+++ minizip-ng-2.10.4/CMakeLists.txt	2021-01-22 14:22:55.215576991 +0000
@@ -639,7 +639,6 @@
 endif()
 
 target_link_libraries(${PROJECT_NAME} PUBLIC ${MINIZIP_LIB} ${MINIZIP_DEP})
-target_link_directories(${PROJECT_NAME} PUBLIC ${MINIZIP_LBD})
 target_compile_definitions(${PROJECT_NAME} PRIVATE ${STDLIB_DEF} ${MINIZIP_DEF})
 target_include_directories(${PROJECT_NAME} PRIVATE ${MINIZIP_INC})
 target_include_directories(${PROJECT_NAME} PUBLIC
