diff -ru moe-1.12.orig/configure moe-1.12/configure
--- moe-1.12.orig/configure	2022-01-20 17:28:29.000000000 +0100
+++ moe-1.12/configure	2022-02-03 23:41:43.452057383 +0100
@@ -23,9 +23,9 @@
 sysconfdir='$(prefix)/etc'
 CXX=g++
 CPPFLAGS=
-CXXFLAGS='-Wall -W -O2'
+CXXFLAGS ?= '-Wall -W -O2'
 LDFLAGS=
-LIBS=-lncurses
+LIBS='-lncurses -ltinfo'
 
 # checking whether we are using GNU C++.
 /bin/sh -c "${CXX} --version" > /dev/null 2>&1 || { CXX=c++ ; CXXFLAGS=-O2 ; }
