diff -ru ncdu-1.18.orig/configure ncdu-1.18/configure
--- ncdu-1.18.orig/configure	2022-12-06 10:47:01.000000000 +0100
+++ ncdu-1.18/configure	2022-12-23 16:05:20.100553280 +0100
@@ -6464,7 +6464,7 @@
   ac_status=$?
   printf "%s\n" "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_NCURSES_LIBS=`$PKG_CONFIG --libs "ncursesw" 2>/dev/null`
+  pkg_cv_NCURSES_LIBS=`$PKG_CONFIG --libs-only-l "ncursesw" 2>/dev/null`
 		      test "x$?" != "x0" && pkg_failed=yes
 else
   pkg_failed=yes
