--- ncurses-6.0/configure.orig	2015-08-05 09:20:32.000000000 +0000
+++ ncurses-6.0/configure	2016-02-23 11:33:12.718112802 +0000
@@ -5856,7 +5856,7 @@
 		LD_RPATH_OPT="-rpath "
 	fi
 	;;
-(linux*|gnu*|k*bsd*-gnu)
+(linux*|gnu*|k*bsd*-gnu|midipix*)
 	LD_RPATH_OPT="-Wl,-rpath,"
 	;;
 (openbsd[2-9].*|mirbsd*)
@@ -6188,7 +6188,7 @@
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
-	(linux*|gnu*|k*bsd*-gnu)
+	(linux*|gnu*|k*bsd*-gnu|midipix*)
 		if test "$DFT_LWR_MODEL" = "shared" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
--- ncurses-6.0/misc/Makefile.in.orig	2015-08-05 23:15:41.000000000 +0000
+++ ncurses-6.0/misc/Makefile.in	2016-06-23 08:40:43.767712775 +0000
@@ -110,8 +110,8 @@
 install.libs :: $(DESTDIR)$(bindir) ncurses-config
 	$(INSTALL_SCRIPT) ncurses-config $(DESTDIR)$(bindir)/$(NCURSES_CONFIG)
 
-@MAKE_PC_FILES@install.libs :: pc-files $(DESTDIR)$(PKG_CONFIG_LIBDIR)
-@MAKE_PC_FILES@	$(SHELL) -c 'for name in *.pc; do $(INSTALL_DATA) $$name $(DESTDIR)$(PKG_CONFIG_LIBDIR)/$$name; done'
+@MAKE_PC_FILES@install.libs :: pc-files $(PKG_CONFIG_LIBDIR)
+@MAKE_PC_FILES@	$(SHELL) -c 'for name in *.pc; do $(INSTALL_DATA) $$name $(PKG_CONFIG_LIBDIR)/$$name; done'
 
 @MAKE_PC_FILES@sources :: pc-files
 @MAKE_PC_FILES@pc-files :
