--- ncurses-6.0/misc/Makefile.in.orig	2015-08-05 23:15:41.000000000 +0000
+++ ncurses-6.0/misc/Makefile.in	2019-03-13 12:31:25.236000000 +0000
@@ -84,17 +84,17 @@
 install : @MISC_INSTALL_DATA@ install.libs
 
 install.data :	terminfo.tmp \
-		$(DESTDIR)$(libdir) \
-		$(DESTDIR)$(datadir) \
-		$(DESTDIR)$(tabsetdir)
+		$(DESTDIR)/lib \
+		$(DESTDIR)/share \
+		$(DESTDIR)/share/tabset
 	DESTDIR=${DESTDIR} \
-	prefix=${prefix} \
-	exec_prefix=${exec_prefix} \
-	bindir=${bindir} \
+	prefix="" \
+	exec_prefix="" \
+	bindir=bin \
 	top_srcdir=${top_srcdir} \
 	srcdir=${srcdir} \
-	datadir=${datadir} \
-	ticdir=${ticdir} \
+	datadir=share \
+	ticdir=share/terminfo \
 	source=terminfo.tmp \
 	cross_compiling=@cross_compiling@ \
 	$(SHELL) ./run_tic.sh
@@ -102,7 +102,7 @@
 		$(SHELL) -c 'for i in * ; do \
 			if test -f $$i ; then \
 			echo installing $$i; \
-			$(INSTALL_DATA) $$i $(DESTDIR)$(tabsetdir)/$$i; \
+			$(INSTALL_DATA) $$i $(DESTDIR)/share/tabset/$$i; \
 			fi; done'
 
 NCURSES_CONFIG	= ncurses@DFT_ARG_SUFFIX@@cf_cv_abi_version@-config
@@ -110,8 +110,8 @@
 install.libs :: $(DESTDIR)$(bindir) ncurses-config
 	$(INSTALL_SCRIPT) ncurses-config $(DESTDIR)$(bindir)/$(NCURSES_CONFIG)
 
-@MAKE_PC_FILES@install.libs :: pc-files $(DESTDIR)$(PKG_CONFIG_LIBDIR)
-@MAKE_PC_FILES@	$(SHELL) -c 'for name in *.pc; do $(INSTALL_DATA) $$name $(DESTDIR)$(PKG_CONFIG_LIBDIR)/$$name; done'
+@MAKE_PC_FILES@install.libs :: pc-files $(PKG_CONFIG_LIBDIR)
+@MAKE_PC_FILES@	$(SHELL) -c 'for name in *.pc; do $(INSTALL_DATA) $$name $(PKG_CONFIG_LIBDIR)/$$name; done'
 
 @MAKE_PC_FILES@sources :: pc-files
 @MAKE_PC_FILES@pc-files :
--- ncurses-6.0/misc/run_tic.in.orig	2012-10-06 19:00:51.000000000 +0000
+++ ncurses-6.0/misc/run_tic.in	2019-03-13 12:31:25.236000000 +0000
@@ -42,16 +42,8 @@
 #	make install.data
 
 : ${suffix:=@EXEEXT@}
-: ${DESTDIR:=@DESTDIR@}
-: ${prefix:=@prefix@}
-: ${exec_prefix:=@exec_prefix@}
-: ${bindir:=@bindir@}
-: ${top_srcdir:=@top_srcdir@}
-: ${srcdir:=@srcdir@}
 : ${datarootdir:=@datarootdir@}
-: ${datadir:=@datadir@}
 : ${TIC_PATH:=@TIC_PATH@}
-: ${ticdir:=@TERMINFO@}
 : ${source:=@TERMINFO_SRC@}
 : ${LN_S:="@LN_S@"}
 : ${cross_compiling:=no}
