diff -ru a/configure b/configure
--- a/configure	2018-01-20 00:27:18.000000000 +0000
+++ b/configure	2019-08-19 09:46:30.361489923 +0000
@@ -6317,7 +6317,7 @@
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
-	(linux*|gnu*|k*bsd*-gnu)
+	(linux*|gnu*|k*bsd*-gnu|midipix*)
 		if test "$DFT_LWR_MODEL" = "shared" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
diff -ru a/misc/gen-pkgconfig.in b/misc/gen-pkgconfig.in
--- a/misc/gen-pkgconfig.in	2018-01-01 22:51:23.000000000 +0000
+++ b/misc/gen-pkgconfig.in	2019-08-19 10:00:35.426023772 +0000
@@ -64,17 +64,7 @@
 MENU_LIBRARY="${MENU_NAME}@USE_ARG_SUFFIX@"
 FORM_LIBRARY="${FORM_NAME}@USE_ARG_SUFFIX@"
 
-CFLAGS="@PKG_CFLAGS@"
-if test "$includedir" != "/usr/include" ; then
-	includetop=`echo "$includedir" | sed -e 's,/include/[^/]*$,/include,'`
-	test "$includetop" = "/usr/include" && includetop="$includedir"
-	if test "x$includetop" != "x$includedir"
-	then
-		CFLAGS="$CFLAGS -I${includetop}"
-	fi
-	CFLAGS="$CFLAGS -I\${includedir}"
-fi
-
+CFLAGS=
 LDFLAGS=
 if test "$libdir" != "/usr/lib" ; then
 	LDFLAGS="$LDFLAGS -L\${libdir}"
