diff -ru a/configure b/configure
--- a/configure	2018-01-20 00:27:18.000000000 +0000
+++ b/configure	2019-08-19 14:51:02.694029707 +0000
@@ -6317,7 +6317,7 @@
 		fi
 		cf_cv_rm_so_locs=yes
 		;;
-	(linux*|gnu*|k*bsd*-gnu)
+	(linux*|gnu*|k*bsd*-gnu|midipix*)
 		if test "$DFT_LWR_MODEL" = "shared" ; then
 			LOCAL_LDFLAGS="${LD_RPATH_OPT}\$(LOCAL_LIBDIR)"
 			LOCAL_LDFLAGS2="$LOCAL_LDFLAGS"
diff -ru a/misc/gen-pkgconfig.in b/misc/gen-pkgconfig.in
--- a/misc/gen-pkgconfig.in	2018-01-01 22:51:23.000000000 +0000
+++ b/misc/gen-pkgconfig.in	2019-08-19 14:51:02.694029707 +0000
@@ -64,17 +64,7 @@
 MENU_LIBRARY="${MENU_NAME}@USE_ARG_SUFFIX@"
 FORM_LIBRARY="${FORM_NAME}@USE_ARG_SUFFIX@"
 
-CFLAGS="@PKG_CFLAGS@"
-if test "$includedir" != "/usr/include" ; then
-	includetop=`echo "$includedir" | sed -e 's,/include/[^/]*$,/include,'`
-	test "$includetop" = "/usr/include" && includetop="$includedir"
-	if test "x$includetop" != "x$includedir"
-	then
-		CFLAGS="$CFLAGS -I${includetop}"
-	fi
-	CFLAGS="$CFLAGS -I\${includedir}"
-fi
-
+CFLAGS='-I${includedir}'
 LDFLAGS=
 if test "$libdir" != "/usr/lib" ; then
 	LDFLAGS="$LDFLAGS -L\${libdir}"
--- ncurses-6.1/misc/run_tic.in.orig	2017-10-24 00:16:22.000000000 +0000
+++ ncurses-6.1/misc/run_tic.in	2019-10-11 16:25:12.380000000 +0000
@@ -42,16 +42,8 @@
 #	make install.data
 
 : ${suffix:=@EXEEXT@}
-: ${DESTDIR:=@DESTDIR@}
-: ${prefix:=@prefix@}
-: ${exec_prefix:=@exec_prefix@}
-: ${bindir:=@bindir@}
-: ${top_srcdir:=@top_srcdir@}
-: ${srcdir:=@srcdir@}
 : ${datarootdir:=@datarootdir@}
-: ${datadir:=@datadir@}
 : ${TIC_PATH:=@TIC_PATH@}
-: ${ticdir:=@TERMINFO@}
 : ${source:=@TERMINFO_SRC@}
 : ${LN_S:="@LN_S@"}
 : ${cross_compiling:=no}
--- ncurses-6.1/misc/Makefile.in.orig	2017-12-31 20:34:42.000000000 +0000
+++ ncurses-6.1/misc/Makefile.in	2019-10-11 16:27:12.376000000 +0000
@@ -83,17 +83,17 @@
 install : @MISC_INSTALL_DATA@ install.libs
 
 install.data :	terminfo.tmp \
-		$(DESTDIR)$(libdir) \
-		$(DESTDIR)$(datadir) \
-		$(DESTDIR)$(tabsetdir)
+		$(DESTDIR)/lib \
+		$(DESTDIR)/share \
+		$(DESTDIR)/share/tabset
 	DESTDIR=${DESTDIR} \
-	prefix=${prefix} \
-	exec_prefix=${exec_prefix} \
-	bindir=${bindir} \
+	prefix="" \
+	exec_prefix="" \
+	bindir=bin \
 	top_srcdir=${top_srcdir} \
 	srcdir=${srcdir} \
-	datadir=${datadir} \
-	ticdir=${ticdir} \
+	datadir=share \
+	ticdir=share/terminfo \
 	source=terminfo.tmp \
 	cross_compiling=@cross_compiling@ \
 	$(SHELL) ./run_tic.sh
@@ -101,7 +101,7 @@
 		$(SHELL) -c 'for i in * ; do \
 			if test -f $$i ; then \
 			echo installing $$i; \
-			$(INSTALL_DATA) $$i $(DESTDIR)$(tabsetdir)/$$i; \
+			$(INSTALL_DATA) $$i $(DESTDIR)/share/tabset/$$i; \
 			fi; done'
 
 NCURSES_CONFIG	= ncurses@DFT_ARG_SUFFIX@@cf_cv_abi_version@@cf_config_suffix@-config
@@ -113,12 +113,12 @@
 # pkg-config is not installed.  Work around that by creating the library
 # directory during this rule:
 @MAKE_PC_FILES@install.libs :: pc-files
-@MAKE_PC_FILES@	@$(SHELL) -c 'case "x$(DESTDIR)$(PKG_CONFIG_LIBDIR)" in \
+@MAKE_PC_FILES@	@$(SHELL) -c 'case "x$(PKG_CONFIG_LIBDIR)" in \
 @MAKE_PC_FILES@		(x/*) \
-@MAKE_PC_FILES@			mkdir -p $(DESTDIR)$(PKG_CONFIG_LIBDIR); \
+@MAKE_PC_FILES@			mkdir -p $(PKG_CONFIG_LIBDIR); \
 @MAKE_PC_FILES@			for name in *.pc; do \
 @MAKE_PC_FILES@				echo installing $$name; \
-@MAKE_PC_FILES@				$(INSTALL_DATA) $$name $(DESTDIR)$(PKG_CONFIG_LIBDIR)/$$name; \
+@MAKE_PC_FILES@				$(INSTALL_DATA) $$name $(PKG_CONFIG_LIBDIR)/$$name; \
 @MAKE_PC_FILES@			done \
 @MAKE_PC_FILES@			;; \
 @MAKE_PC_FILES@		(*) \
diff -ru a/misc/ncurses-config.in b/misc/ncurses-config.in
--- a/misc/ncurses-config.in 2018-01-20 00:27:18.000000000 +0000
+++ b/misc/ncurses-config.in 2020-03-11 15:38:05.000000000 +0000
@@ -111,12 +111,7 @@
 		;;
 	--libs)
-		if test "$libdir" = /usr/lib
-		then
-			LIBDIR=
-		else
-			LIBDIR=-L$libdir
-		fi
+		LIBDIR=
 		if test @TINFO_NAME@ = @LIB_NAME@ ; then
 		sed -e 's,^[ ]*,,' -e 's, [ ]*, ,g' -e 's,[ ]*$,,' <<-ENDECHO
 			$LIBDIR @EXTRA_LDFLAGS@ -l${THIS} @LIBS@
