diff -ru nettle-3.8.1.orig/x86_64/fat/aes128-decrypt-2.asm nettle-3.8.1/x86_64/fat/aes128-decrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/aes128-decrypt-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/aes128-decrypt-2.asm        2023-02-17 13:40:13.492569768 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/aes128-decrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_aes128_decrypt_aesni,"r"
+        .global __imp__nettle_aes128_decrypt_aesni
+__imp__nettle_aes128_decrypt_aesni:
+        .quad   _nettle_aes128_decrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/aes128-encrypt-2.asm nettle-3.8.1/x86_64/fat/aes128-encrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/aes128-encrypt-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/aes128-encrypt-2.asm        2023-02-17 13:39:44.528569389 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/aes128-encrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_aes128_encrypt_aesni,"r"
+        .global __imp__nettle_aes128_encrypt_aesni
+__imp__nettle_aes128_encrypt_aesni:
+        .quad   _nettle_aes128_encrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/aes192-decrypt-2.asm nettle-3.8.1/x86_64/fat/aes192-decrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/aes192-decrypt-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/aes192-decrypt-2.asm        2023-02-17 13:41:00.400570382 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/aes192-decrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_aes192_decrypt_aesni,"r"
+        .global __imp__nettle_aes192_decrypt_aesni
+__imp__nettle_aes192_decrypt_aesni:
+        .quad   _nettle_aes192_decrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/aes192-encrypt-2.asm nettle-3.8.1/x86_64/fat/aes192-encrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/aes192-encrypt-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/aes192-encrypt-2.asm        2023-02-17 13:40:37.196570078 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/aes192-encrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_aes192_encrypt_aesni,"r"
+        .global __imp__nettle_aes192_encrypt_aesni
+__imp__nettle_aes192_encrypt_aesni:
+        .quad   _nettle_aes192_encrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/aes256-decrypt-2.asm nettle-3.8.1/x86_64/fat/aes256-decrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/aes256-decrypt-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/aes256-decrypt-2.asm        2023-02-17 13:44:55.409072440 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/aes256-decrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_aes256_decrypt_aesni,"r"
+        .global __imp__nettle_aes256_decrypt_aesni
+__imp__nettle_aes256_decrypt_aesni:
+        .quad   _nettle_aes256_decrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/aes256-encrypt-2.asm nettle-3.8.1/x86_64/fat/aes256-encrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/aes256-encrypt-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/aes256-encrypt-2.asm        2023-02-17 13:44:31.165072123 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/aes256-encrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_aes256_encrypt_aesni,"r"
+        .global __imp__nettle_aes256_encrypt_aesni
+__imp__nettle_aes256_encrypt_aesni:
+        .quad   _nettle_aes256_encrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/cbc-aes128-encrypt-2.asm nettle-3.8.1/x86_64/fat/cbc-aes128-encrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/cbc-aes128-encrypt-2.asm       2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/cbc-aes128-encrypt-2.asm    2023-02-17 13:45:43.369073067 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/cbc-aes128-encrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_cbc_aes128_encrypt_aesni,"r"
+        .global __imp__nettle_cbc_aes128_encrypt_aesni
+__imp__nettle_cbc_aes128_encrypt_aesni:
+        .quad   _nettle_cbc_aes128_encrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/cbc-aes192-encrypt-2.asm nettle-3.8.1/x86_64/fat/cbc-aes192-encrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/cbc-aes192-encrypt-2.asm       2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/cbc-aes192-encrypt-2.asm    2023-02-17 13:46:09.457073408 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/cbc-aes192-encrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_cbc_aes192_encrypt_aesni,"r"
+        .global __imp__nettle_cbc_aes192_encrypt_aesni
+__imp__nettle_cbc_aes192_encrypt_aesni:
+        .quad   _nettle_cbc_aes192_encrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/cbc-aes256-encrypt-2.asm nettle-3.8.1/x86_64/fat/cbc-aes256-encrypt-2.asm
--- nettle-3.8.1.orig/x86_64/fat/cbc-aes256-encrypt-2.asm       2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/cbc-aes256-encrypt-2.asm    2023-02-17 13:46:31.545073697 +0100
@@ -34,3 +34,11 @@

 define(`fat_transform', `_$1_aesni')
 include_src(`x86_64/aesni/cbc-aes256-encrypt.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_cbc_aes256_encrypt_aesni,"r"
+        .global __imp__nettle_cbc_aes256_encrypt_aesni
+__imp__nettle_cbc_aes256_encrypt_aesni:
+        .quad   _nettle_cbc_aes256_encrypt_aesni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/sha1-compress-2.asm nettle-3.8.1/x86_64/fat/sha1-compress-2.asm
--- nettle-3.8.1.orig/x86_64/fat/sha1-compress-2.asm    2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/sha1-compress-2.asm 2023-02-17 12:37:28.782267505 +0100
@@ -32,3 +32,11 @@

 define(`fat_transform', `_$1_sha_ni')
 include_src(`x86_64/sha_ni/sha1-compress.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_sha1_compress_sha_ni,"r"
+        .global __imp__nettle_sha1_compress_sha_ni
+__imp__nettle_sha1_compress_sha_ni:
+        .quad   _nettle_sha1_compress_sha_ni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/sha1-compress.asm nettle-3.8.1/x86_64/fat/sha1-compress.asm
--- nettle-3.8.1.orig/x86_64/fat/sha1-compress.asm      2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/sha1-compress.asm   2023-02-17 12:37:57.546267881 +0100
@@ -32,3 +32,11 @@

 define(`fat_transform', `_$1_x86_64')
 include_src(`x86_64/sha1-compress.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_sha1_compress_x86_64,"r"
+        .global __imp__nettle_sha1_compress_x86_64
+__imp__nettle_sha1_compress_x86_64:
+        .quad   _nettle_sha1_compress_x86_64
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/ghash-set-key-2.asm nettle-3.8.1/x86_64/fat/ghash-set-key-2.asm
--- nettle-3.8.1.orig/x86_64/fat/ghash-set-key-2.asm    2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/ghash-set-key-2.asm 2023-02-17 12:32:21.594437764 +0100
@@ -35,3 +35,11 @@

 define(`fat_transform', `$1_pclmul')
 include_src(`x86_64/pclmul/ghash-set-key.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_ghash_set_key_pclmul,"r"
+        .global __imp__nettle_ghash_set_key_pclmul
+__imp__nettle_ghash_set_key_pclmul:
+        .quad   _nettle_ghash_set_key_pclmul
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/sha256-compress-2.asm nettle-3.8.1/x86_64/fat/sha256-compress-2.asm
--- nettle-3.8.1.orig/x86_64/fat/sha256-compress-2.asm  2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/sha256-compress-2.asm       2023-02-17 12:35:29.438265944 +0100
@@ -32,3 +32,11 @@

 define(`fat_transform', `$1_sha_ni')
 include_src(`x86_64/sha_ni/sha256-compress.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_sha256_compress_sha_ni,"r"
+        .global __imp__nettle_sha256_compress_sha_ni
+__imp__nettle_sha256_compress_sha_ni:
+        .quad   _nettle_sha256_compress_sha_ni
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/sha256-compress.asm nettle-3.8.1/x86_64/fat/sha256-compress.asm
--- nettle-3.8.1.orig/x86_64/fat/sha256-compress.asm    2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/sha256-compress.asm 2023-02-17 12:34:02.826264811 +0100
@@ -32,3 +32,11 @@

 define(`fat_transform', `$1_x86_64')
 include_src(`x86_64/sha256-compress.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_sha256_compress_x86_64,"r"
+        .global __imp__nettle_sha256_compress_x86_64
+__imp__nettle_sha256_compress_x86_64:
+        .quad   _nettle_sha256_compress_x86_64
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/ghash-update-2.asm nettle-3.8.1/x86_64/fat/ghash-update-2.asm
--- nettle-3.8.1.orig/x86_64/fat/ghash-update-2.asm     2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/ghash-update-2.asm  2023-02-17 12:29:12.138435285 +0100
@@ -35,3 +35,11 @@

 define(`fat_transform', `$1_pclmul')
 include_src(`x86_64/pclmul/ghash-update.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_ghash_update_pclmul,"r"
+        .global __imp__nettle_ghash_update_pclmul
+__imp__nettle_ghash_update_pclmul:
+        .quad   _nettle_ghash_update_pclmul
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/memxor-2.asm nettle-3.8.1/x86_64/fat/memxor-2.asm
--- nettle-3.8.1.orig/x86_64/fat/memxor-2.asm   2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/memxor-2.asm        2023-02-17 12:27:46.702434168 +0100
@@ -34,3 +34,11 @@
 define(`fat_transform', `_$1_sse2')
 define(`USE_SSE2', `yes')
 include_src(`x86_64/memxor.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_memxor_sse2,"r"
+        .global __imp__nettle_memxor_sse2
+__imp__nettle_memxor_sse2:
+        .quad   _nettle_memxor_sse2
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/memxor.asm nettle-3.8.1/x86_64/fat/memxor.asm
--- nettle-3.8.1.orig/x86_64/fat/memxor.asm     2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/memxor.asm  2023-02-17 12:25:20.797229091 +0100
@@ -33,3 +33,11 @@

 define(`fat_transform', `_$1_x86_64')
 include_src(`x86_64/memxor.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_memxor_x86_64,"r"
+        .global __imp__nettle_memxor_x86_64
+__imp__nettle_memxor_x86_64:
+        .quad   _nettle_memxor_x86_64
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/cpuid.asm nettle-3.8.1/x86_64/fat/cpuid.asm
--- nettle-3.8.1.orig/x86_64/fat/cpuid.asm      2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/cpuid.asm   2023-02-17 12:20:01.842500301 +0100
@@ -57,3 +57,10 @@
        ret
 EPILOGUE(_nettle_cpuid)

+#ifdef __midipix__
+        .section .got$_nettle_cpuid,"r"
+        .global __imp__nettle_cpuid
+__imp__nettle_cpuid:
+        .quad   _nettle_cpuid
+        .linkonce discard
+#endif
diff -ru nettle-3.8.1.orig/x86_64/fat/ghash-update.asm nettle-3.8.1/x86_64/fat/ghash-update.asm
--- nettle-3.8.1.orig/x86_64/fat/ghash-update.asm       2022-07-27 21:23:20.000000000 +0200
+++ nettle-3.8.1/x86_64/fat/ghash-update.asm    2023-02-17 12:22:38.869226973 +0100
@@ -35,3 +35,11 @@

 define(`fat_transform', `$1_table')
 include_src(`x86_64/ghash-update.asm')
+
+#ifdef __midipix__
+        .section .got$_nettle_ghash_update_table,"r"
+        .global __imp__nettle_ghash_update_table
+__imp__nettle_ghash_update_table:
+        .quad   _nettle_ghash_update_table
+        .linkonce discard
+#endif
diff -ru nettle-3.7.2.orig/x86_64/salsa20-2core.asm nettle-3.7.2/x86_64/salsa20-2core.asm
--- nettle-3.7.2.orig/x86_64/salsa20-2core.asm  2021-03-21 09:32:25.000000000 +0100
+++ nettle-3.7.2/x86_64/salsa20-2core.asm       2021-04-10 14:56:55.970009499 +0200
@@ -316,3 +316,11 @@
        W64_EXIT(3, 13)
        ret
 EPILOGUE(_nettle_salsa20_2core)
+
+#ifdef __midipix__
+        .section .got$_nettle_salsa20_2core,"r"
+        .global __imp__nettle_salsa20_2core
+__imp__nettle_salsa20_2core:
+        .quad   _nettle_salsa20_2core
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/aes-decrypt-internal.asm nettle-3.6/x86_64/aes-decrypt-internal.asm
--- nettle-3.6.orig/x86_64/aes-decrypt-internal.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/aes-decrypt-internal.asm	2020-09-07 13:38:12.793134936 +0200
@@ -150,3 +150,11 @@
 	W64_EXIT(6, 0)
 	ret
 EPILOGUE(_nettle_aes_decrypt)
+
+#ifdef __midipix__
+	.section .got$_nettle_aes_decrypt,"r"
+	.global __imp__nettle_aes_decrypt
+__imp__nettle_aes_decrypt:
+	.quad   _nettle_aes_decrypt
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/aes-encrypt-internal.asm nettle-3.6/x86_64/aes-encrypt-internal.asm
--- nettle-3.6.orig/x86_64/aes-encrypt-internal.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/aes-encrypt-internal.asm	2020-09-07 13:39:17.917132897 +0200
@@ -151,3 +151,11 @@
 	W64_EXIT(6, 0)
 	ret
 EPILOGUE(_nettle_aes_encrypt)
+
+#ifdef __midipix__
+	.section .got$_nettle_aes_encrypt,"r"
+	.global __imp__nettle_aes_encrypt
+__imp__nettle_aes_encrypt:
+	.quad   _nettle_aes_encrypt
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/camellia-crypt-internal.asm nettle-3.6/x86_64/camellia-crypt-internal.asm
--- nettle-3.6.orig/x86_64/camellia-crypt-internal.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/camellia-crypt-internal.asm	2020-09-07 13:40:27.389130722 +0200
@@ -200,3 +200,11 @@
 	W64_EXIT(6, 0)
 	ret
 EPILOGUE(_nettle_camellia_crypt)
+
+#ifdef __midipix__
+	.section .got$_nettle_camellia_crypt,"r"
+	.global __imp__nettle_camellia_crypt
+__imp__nettle_camellia_crypt:
+	.quad   _nettle_camellia_crypt
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/chacha-core-internal.asm nettle-3.6/x86_64/chacha-core-internal.asm
--- nettle-3.6.orig/x86_64/chacha-core-internal.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/chacha-core-internal.asm	2020-09-07 13:41:25.405128905 +0200
@@ -126,3 +126,11 @@
 	W64_EXIT(3, 6)
 	ret
 EPILOGUE(_nettle_chacha_core)
+
+#ifdef __midipix__
+	.section .got$_nettle_chacha_core,"r"
+	.global __imp__nettle_chacha_core
+__imp__nettle_chacha_core:
+	.quad   _nettle_chacha_core
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-curve25519-modp.asm nettle-3.6/x86_64/ecc-curve25519-modp.asm
--- nettle-3.6.orig/x86_64/ecc-curve25519-modp.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-curve25519-modp.asm	2020-09-07 14:01:46.533804910 +0200
@@ -92,3 +92,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_curve25519_modp)
+
+#ifdef __midipix__
+        .section .got$_nettle_ecc_curve25519_modp,"r"
+        .global __imp__nettle_ecc_curve25519_modp
+__imp__nettle_ecc_curve25519_modp:
+        .quad   _nettle_ecc_curve25519_modp
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-curve448-modp.asm nettle-3.6/x86_64/ecc-curve448-modp.asm
--- nettle-3.6.orig/x86_64/ecc-curve448-modp.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-curve448-modp.asm	2020-09-07 14:02:55.749802742 +0200
@@ -142,3 +142,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_curve448_modp)
+
+#ifdef __midipix__
+	.section .got$_nettle_ecc_curve448_modp,"r"
+	.global __imp__nettle_ecc_curve448_modp
+__imp__nettle_ecc_curve448_modp:
+	.quad   _nettle_ecc_curve448_modp
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-secp192r1-modp.asm nettle-3.6/x86_64/ecc-secp192r1-modp.asm
--- nettle-3.6.orig/x86_64/ecc-secp192r1-modp.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-secp192r1-modp.asm	2020-09-07 14:04:12.289800346 +0200
@@ -86,3 +86,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_secp192r1_modp)
+
+#ifdef __midipix__
+	.section .got$_nettle_ecc_secp192r1_modp,"r"
+	.global __imp__nettle_ecc_secp192r1_modp
+__imp__nettle_ecc_secp192r1_modp:
+	.quad   _nettle_ecc_secp192r1_modp
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-secp224r1-modp.asm nettle-3.6/x86_64/ecc-secp224r1-modp.asm
--- nettle-3.6.orig/x86_64/ecc-secp224r1-modp.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-secp224r1-modp.asm	2020-09-07 14:05:23.821798106 +0200
@@ -129,3 +129,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_secp224r1_modp)
+
+#ifdef __midipix__
+	.section .got$_nettle_ecc_secp224r1_modp,"r"
+	.global __imp__nettle_ecc_secp224r1_modp
+__imp__nettle_ecc_secp224r1_modp:
+	.quad   _nettle_ecc_secp224r1_modp
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-secp256r1-redc.asm nettle-3.6/x86_64/ecc-secp256r1-redc.asm
--- nettle-3.6.orig/x86_64/ecc-secp256r1-redc.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-secp256r1-redc.asm	2020-09-07 14:06:51.854540772 +0200
@@ -127,3 +127,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_secp256r1_redc)
+
+#ifdef __midipix__
+	.section .got$_nettle_ecc_secp256r1_redc,"r"
+	.global __imp__nettle_ecc_secp256r1_redc
+__imp__nettle_ecc_secp256r1_redc:
+	.quad   _nettle_ecc_secp256r1_redc
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-secp384r1-modp.asm nettle-3.6/x86_64/ecc-secp384r1-modp.asm
--- nettle-3.6.orig/x86_64/ecc-secp384r1-modp.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-secp384r1-modp.asm	2020-09-07 14:08:10.706538303 +0200
@@ -232,3 +232,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_secp384r1_modp)
+
+#ifdef __midipix__
+	.section .got$_nettle_ecc_secp384r1_modp,"r"
+	.global __imp__nettle_ecc_secp384r1_modp
+__imp__nettle_ecc_secp384r1_modp:
+	.quad   _nettle_ecc_secp384r1_modp
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/ecc-secp521r1-modp.asm nettle-3.6/x86_64/ecc-secp521r1-modp.asm
--- nettle-3.6.orig/x86_64/ecc-secp521r1-modp.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/ecc-secp521r1-modp.asm	2020-09-07 14:09:27.370535902 +0200
@@ -156,3 +156,11 @@
 	W64_EXIT(2, 0)
 	ret
 EPILOGUE(_nettle_ecc_secp521r1_modp)
+
+#ifdef __midipix__
+	.section .got$_nettle_ecc_secp521r1_modp,"r"
+	.global __imp__nettle_ecc_secp521r1_modp
+__imp__nettle_ecc_secp521r1_modp:
+	.quad   _nettle_ecc_secp521r1_modp
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/md5-compress.asm nettle-3.6/x86_64/md5-compress.asm
--- nettle-3.6.orig/x86_64/md5-compress.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/md5-compress.asm	2020-09-07 13:44:12.135638069 +0200
@@ -174,3 +174,11 @@
 
 	ret
 EPILOGUE(nettle_md5_compress)
+
+#ifdef __midipix__
+	.section .got$nettle_md5_compress,"r"
+	.global __imp_nettle_md5_compress
+__imp_nettle_md5_compress:
+	.quad   nettle_md5_compress
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/memxor3.asm nettle-3.6/x86_64/memxor3.asm
--- nettle-3.6.orig/x86_64/memxor3.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/memxor3.asm	2020-09-07 13:45:42.443635242 +0200
@@ -261,3 +261,11 @@
 	
 
 EPILOGUE(nettle_memxor3)
+
+#ifdef __midipix__
+	.section .got$nettle_memxor3,"r"
+	.global __imp_nettle_memxor3
+__imp_nettle_memxor3:
+	.quad   nettle_memxor3
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/poly1305-internal.asm nettle-3.6/x86_64/poly1305-internal.asm
--- nettle-3.6.orig/x86_64/poly1305-internal.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/poly1305-internal.asm	2020-09-07 13:51:14.239748557 +0200
@@ -184,3 +184,22 @@
 	ret
 EPILOGUE(_nettle_poly1305_digest)
 
+#ifdef __midipix__
+	.section .got$_nettle_poly1305_set_key,"r"
+	.global __imp__nettle_poly1305_set_key
+__imp__nettle_poly1305_set_key:
+	.quad   _nettle_poly1305_set_key
+	.linkonce discard
+
+	.section .got$_nettle_poly1305_block,"r"
+	.global __imp__nettle_poly1305_block
+__imp__nettle_poly1305_block:
+	.quad   _nettle_poly1305_block
+	.linkonce discard
+
+	.section .got$_nettle_poly1305_digest,"r"
+	.global __imp__nettle_poly1305_digest
+__imp__nettle_poly1305_digest:
+	.quad   _nettle_poly1305_digest
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/salsa20-core-internal.asm nettle-3.6/x86_64/salsa20-core-internal.asm
--- nettle-3.6.orig/x86_64/salsa20-core-internal.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/salsa20-core-internal.asm	2020-09-07 13:53:21.847744562 +0200
@@ -109,3 +109,11 @@
 	W64_EXIT(3, 9)
 	ret
 EPILOGUE(_nettle_salsa20_core)
+
+#ifdef __midipix__
+	.section .got$_nettle_salsa20_core,"r"
+	.global __imp__nettle_salsa20_core
+__imp__nettle_salsa20_core:
+	.quad   _nettle_salsa20_core
+	.linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/serpent-decrypt.asm nettle-3.6/x86_64/serpent-decrypt.asm
--- nettle-3.6.orig/x86_64/serpent-decrypt.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/serpent-decrypt.asm	2020-09-07 13:55:28.667790597 +0200
@@ -714,3 +714,11 @@
 	W64_EXIT(4, 13)
 	ret
 EPILOGUE(nettle_serpent_decrypt)
+
+#ifdef __midipix__
+        .section .got$nettle_serpent_decrypt,"r"
+        .global __imp_nettle_serpent_decrypt
+__imp_nettle_serpent_decrypt:
+        .quad   nettle_serpent_decrypt
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/serpent-encrypt.asm nettle-3.6/x86_64/serpent-encrypt.asm
--- nettle-3.6.orig/x86_64/serpent-encrypt.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/serpent-encrypt.asm	2020-09-07 13:56:00.179789610 +0200
@@ -749,3 +749,11 @@
 	W64_EXIT(4, 13)
 	ret
 EPILOGUE(nettle_serpent_encrypt)
+
+#ifdef __midipix__
+        .section .got$nettle_serpent_encrypt,"r"
+        .global __imp_nettle_serpent_encrypt
+__imp_nettle_serpent_encrypt:
+        .quad   nettle_serpent_encrypt
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/sha3-permute.asm nettle-3.6/x86_64/sha3-permute.asm
--- nettle-3.6.orig/x86_64/sha3-permute.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/sha3-permute.asm	2020-09-07 13:57:35.427786628 +0200
@@ -522,3 +522,11 @@
 	.quad	0x800000000000808A
 	.quad	0x0000000000008082
 	.quad	0x0000000000000001
+
+#ifdef __midipix__
+        .section .got$nettle_sha3_permute,"r"
+        .global __imp_nettle_sha3_permute
+__imp_nettle_sha3_permute:
+        .quad   nettle_sha3_permute
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/sha512-compress.asm nettle-3.6/x86_64/sha512-compress.asm
--- nettle-3.6.orig/x86_64/sha512-compress.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/sha512-compress.asm	2020-09-07 13:58:36.391784719 +0200
@@ -208,3 +208,11 @@
 	W64_EXIT(3, 0)
 	ret
 EPILOGUE(_nettle_sha512_compress)
+
+#ifdef __midipix__
+        .section .got$_nettle_sha512_compress,"r"
+        .global __imp__nettle_sha512_compress
+__imp__nettle_sha512_compress:
+        .quad   _nettle_sha512_compress
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/umac-nh.asm nettle-3.6/x86_64/umac-nh.asm
--- nettle-3.6.orig/x86_64/umac-nh.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/umac-nh.asm	2020-09-07 13:58:59.259784003 +0200
@@ -79,3 +79,11 @@
 	W64_EXIT(3, 7)
 	ret
 EPILOGUE(_nettle_umac_nh)
+
+#ifdef __midipix__
+        .section .got$_nettle_umac_nh,"r"
+        .global __imp__nettle_umac_nh
+__imp__nettle_umac_nh:
+        .quad   _nettle_umac_nh
+        .linkonce discard
+#endif
diff -ru nettle-3.6.orig/x86_64/umac-nh-n.asm nettle-3.6/x86_64/umac-nh-n.asm
--- nettle-3.6.orig/x86_64/umac-nh-n.asm	2020-04-29 20:11:44.000000000 +0200
+++ nettle-3.6/x86_64/umac-nh-n.asm	2020-09-07 13:59:24.655783208 +0200
@@ -273,3 +273,11 @@
 	W64_EXIT(5, 14)
 	ret
 EPILOGUE(_nettle_umac_nh_n)
+
+#ifdef __midipix__
+       .section .got$_nettle_umac_nh_n,"r"
+       .global __imp__nettle_umac_nh_n
+__imp__nettle_umac_nh_n:
+       .quad   _nettle_umac_nh_n
+       .linkonce discard
+#endif
