--- nginx-1.11.3.dist/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ nginx-1.11.3/Makefile	2017-02-02 18:11:45.710858114 +0000
@@ -0,0 +1,23 @@
+
+default:	build
+
+clean:
+	rm -rf Makefile objs
+
+build:
+	$(MAKE) -f objs/Makefile
+
+install:
+	$(MAKE) -f objs/Makefile install
+
+modules:
+	$(MAKE) -f objs/Makefile modules
+
+upgrade:
+	/sbin/nginx -t
+
+	kill -USR2 `cat /logs/nginx.pid`
+	sleep 1
+	test -f /logs/nginx.pid.oldbin
+
+	kill -QUIT `cat /logs/nginx.pid.oldbin`
--- nginx-1.11.3.dist/objs/ngx_auto_headers.h	1970-01-01 00:00:00.000000000 +0000
+++ nginx-1.11.3/objs/ngx_auto_headers.h	2017-02-02 18:11:43.338852504 +0000
@@ -0,0 +1,51 @@
+
+
+#ifndef NGX_HAVE_UNISTD_H
+#define NGX_HAVE_UNISTD_H  1
+#endif
+
+
+#ifndef NGX_HAVE_INTTYPES_H
+#define NGX_HAVE_INTTYPES_H  1
+#endif
+
+
+#ifndef NGX_HAVE_LIMITS_H
+#define NGX_HAVE_LIMITS_H  1
+#endif
+
+
+#ifndef NGX_HAVE_SYS_PARAM_H
+#define NGX_HAVE_SYS_PARAM_H  1
+#endif
+
+
+#ifndef NGX_HAVE_SYS_MOUNT_H
+#define NGX_HAVE_SYS_MOUNT_H  1
+#endif
+
+
+#ifndef NGX_HAVE_SYS_STATVFS_H
+#define NGX_HAVE_SYS_STATVFS_H  1
+#endif
+
+
+#ifndef NGX_HAVE_CRYPT_H
+#define NGX_HAVE_CRYPT_H  1
+#endif
+
+
+#ifndef NGX_LINUX
+#define NGX_LINUX  1
+#endif
+
+
+#ifndef NGX_HAVE_SYS_PRCTL_H
+#define NGX_HAVE_SYS_PRCTL_H  1
+#endif
+
+
+#ifndef NGX_HAVE_SYS_VFS_H
+#define NGX_HAVE_SYS_VFS_H  1
+#endif
+
--- nginx-1.11.3.dist/objs/ngx_auto_config.h	1970-01-01 00:00:00.000000000 +0000
+++ nginx-1.11.3/objs/ngx_auto_config.h	2017-02-02 18:25:55.660944596 +0000
@@ -0,0 +1,444 @@
+#define NGX_CONFIGURE " --prefix="
+
+#ifndef NGX_COMPILER
+#define NGX_COMPILER  "gcc 6.2.1 20160830 [gcc-6-branch revision 239856] (SUSE Linux) "
+#endif
+
+
+#ifndef NGX_HAVE_GCC_ATOMIC
+#define NGX_HAVE_GCC_ATOMIC  1
+#endif
+
+
+#ifndef NGX_HAVE_C99_VARIADIC_MACROS
+#define NGX_HAVE_C99_VARIADIC_MACROS  1
+#endif
+
+
+#ifndef NGX_HAVE_GCC_VARIADIC_MACROS
+#define NGX_HAVE_GCC_VARIADIC_MACROS  1
+#endif
+
+
+#ifndef NGX_HAVE_GCC_BSWAP64
+#define NGX_HAVE_GCC_BSWAP64  1
+#endif
+
+
+#ifndef NGX_HAVE_CLEAR_EVENT
+#define NGX_HAVE_CLEAR_EVENT  1
+#endif
+
+
+#ifndef NGX_HAVE_O_PATH
+#define NGX_HAVE_O_PATH  1
+#endif
+
+
+#ifndef NGX_HAVE_SENDFILE
+#define NGX_HAVE_SENDFILE  1
+#endif
+
+
+#ifndef NGX_HAVE_SENDFILE64
+#define NGX_HAVE_SENDFILE64  1
+#endif
+
+
+#ifndef NGX_HAVE_PR_SET_DUMPABLE
+#define NGX_HAVE_PR_SET_DUMPABLE  1
+#endif
+
+
+#ifndef NGX_HAVE_SCHED_SETAFFINITY
+#define NGX_HAVE_SCHED_SETAFFINITY  1
+#endif
+
+
+#ifndef NGX_HAVE_GNU_CRYPT_R
+#define NGX_HAVE_GNU_CRYPT_R  1
+#endif
+
+
+#ifndef NGX_HAVE_NONALIGNED
+#define NGX_HAVE_NONALIGNED  1
+#endif
+
+
+#ifndef NGX_CPU_CACHE_LINE
+#define NGX_CPU_CACHE_LINE  64
+#endif
+
+
+#define NGX_KQUEUE_UDATA_T  (void *)
+
+
+#ifndef NGX_HAVE_POSIX_FADVISE
+#define NGX_HAVE_POSIX_FADVISE  1
+#endif
+
+
+#ifndef NGX_HAVE_O_DIRECT
+#define NGX_HAVE_O_DIRECT  1
+#endif
+
+
+#ifndef NGX_HAVE_ALIGNED_DIRECTIO
+#define NGX_HAVE_ALIGNED_DIRECTIO  1
+#endif
+
+
+#ifndef NGX_HAVE_STATFS
+#define NGX_HAVE_STATFS  1
+#endif
+
+
+#ifndef NGX_HAVE_STATVFS
+#define NGX_HAVE_STATVFS  1
+#endif
+
+
+#ifndef NGX_HAVE_DLOPEN
+#define NGX_HAVE_DLOPEN  1
+#endif
+
+
+#ifndef NGX_HAVE_SCHED_YIELD
+#define NGX_HAVE_SCHED_YIELD  1
+#endif
+
+
+#ifndef NGX_HAVE_REUSEPORT
+#define NGX_HAVE_REUSEPORT  1
+#endif
+
+
+#ifdef NGX_HAVE_IP_BIND_ADDRESS_NO_PORT
+#undef NGX_HAVE_IP_BIND_ADDRESS_NO_PORT
+#endif
+
+
+#ifndef NGX_HAVE_TRANSPARENT_PROXY
+#define NGX_HAVE_TRANSPARENT_PROXY  1
+#endif
+
+
+#ifndef NGX_HAVE_IP_PKTINFO
+#define NGX_HAVE_IP_PKTINFO  1
+#endif
+
+
+#ifndef NGX_HAVE_IPV6_RECVPKTINFO
+#define NGX_HAVE_IPV6_RECVPKTINFO  1
+#endif
+
+
+#ifndef NGX_HAVE_DEFERRED_ACCEPT
+#define NGX_HAVE_DEFERRED_ACCEPT  1
+#endif
+
+
+#ifndef NGX_HAVE_KEEPALIVE_TUNABLE
+#define NGX_HAVE_KEEPALIVE_TUNABLE  1
+#endif
+
+
+#ifndef NGX_HAVE_TCP_FASTOPEN
+#define NGX_HAVE_TCP_FASTOPEN  1
+#endif
+
+
+#ifndef NGX_HAVE_TCP_INFO
+#define NGX_HAVE_TCP_INFO  1
+#endif
+
+
+#ifndef NGX_HAVE_ACCEPT4
+#define NGX_HAVE_ACCEPT4  1
+#endif
+
+
+#ifndef NGX_HAVE_EVENTFD
+#define NGX_HAVE_EVENTFD  1
+#endif
+
+
+#ifndef NGX_HAVE_SYS_EVENTFD_H
+#define NGX_HAVE_SYS_EVENTFD_H  1
+#endif
+
+
+#ifndef NGX_HAVE_UNIX_DOMAIN
+#define NGX_HAVE_UNIX_DOMAIN  1
+#endif
+
+
+#ifndef NGX_PTR_SIZE
+#define NGX_PTR_SIZE  8
+#endif
+
+
+#ifndef NGX_SIG_ATOMIC_T_SIZE
+#define NGX_SIG_ATOMIC_T_SIZE  4
+#endif
+
+
+#ifndef NGX_HAVE_LITTLE_ENDIAN
+#define NGX_HAVE_LITTLE_ENDIAN  1
+#endif
+
+
+#ifndef NGX_MAX_SIZE_T_VALUE
+#define NGX_MAX_SIZE_T_VALUE  9223372036854775807LL
+#endif
+
+
+#ifndef NGX_SIZE_T_LEN
+#define NGX_SIZE_T_LEN  (sizeof("-9223372036854775808") - 1)
+#endif
+
+
+#ifndef NGX_MAX_OFF_T_VALUE
+#define NGX_MAX_OFF_T_VALUE  9223372036854775807LL
+#endif
+
+
+#ifndef NGX_OFF_T_LEN
+#define NGX_OFF_T_LEN  (sizeof("-9223372036854775808") - 1)
+#endif
+
+
+#ifndef NGX_TIME_T_SIZE
+#define NGX_TIME_T_SIZE  8
+#endif
+
+
+#ifndef NGX_TIME_T_LEN
+#define NGX_TIME_T_LEN  (sizeof("-9223372036854775808") - 1)
+#endif
+
+
+#ifndef NGX_MAX_TIME_T_VALUE
+#define NGX_MAX_TIME_T_VALUE  9223372036854775807LL
+#endif
+
+
+#ifndef NGX_HAVE_PREAD
+#define NGX_HAVE_PREAD  1
+#endif
+
+
+#ifndef NGX_HAVE_PWRITE
+#define NGX_HAVE_PWRITE  1
+#endif
+
+
+#ifndef NGX_HAVE_PWRITEV
+#define NGX_HAVE_PWRITEV  1
+#endif
+
+
+#ifndef NGX_SYS_NERR
+#define NGX_SYS_NERR  135
+#endif
+
+
+#ifndef NGX_HAVE_LOCALTIME_R
+#define NGX_HAVE_LOCALTIME_R  1
+#endif
+
+
+#ifndef NGX_HAVE_POSIX_MEMALIGN
+#define NGX_HAVE_POSIX_MEMALIGN  1
+#endif
+
+
+#ifndef NGX_HAVE_MEMALIGN
+#define NGX_HAVE_MEMALIGN  1
+#endif
+
+
+#ifndef NGX_HAVE_MAP_ANON
+#define NGX_HAVE_MAP_ANON  1
+#endif
+
+
+#ifndef NGX_HAVE_MAP_DEVZERO
+#define NGX_HAVE_MAP_DEVZERO  1
+#endif
+
+
+#ifndef NGX_HAVE_SYSVSHM
+#define NGX_HAVE_SYSVSHM  1
+#endif
+
+
+#ifndef NGX_HAVE_POSIX_SEM
+#define NGX_HAVE_POSIX_SEM  1
+#endif
+
+
+#ifndef NGX_HAVE_MSGHDR_MSG_CONTROL
+#define NGX_HAVE_MSGHDR_MSG_CONTROL  1
+#endif
+
+
+#ifndef NGX_HAVE_FIONBIO
+#define NGX_HAVE_FIONBIO  1
+#endif
+
+
+#ifndef NGX_HAVE_GMTOFF
+#define NGX_HAVE_GMTOFF  1
+#endif
+
+
+#ifndef NGX_HAVE_D_TYPE
+#define NGX_HAVE_D_TYPE  1
+#endif
+
+
+#ifndef NGX_HAVE_SC_NPROCESSORS_ONLN
+#define NGX_HAVE_SC_NPROCESSORS_ONLN  1
+#endif
+
+
+#ifndef NGX_HAVE_OPENAT
+#define NGX_HAVE_OPENAT  1
+#endif
+
+
+#ifndef NGX_HAVE_GETADDRINFO
+#define NGX_HAVE_GETADDRINFO  1
+#endif
+
+
+#ifndef NGX_HTTP_CACHE
+#define NGX_HTTP_CACHE  1
+#endif
+
+
+#ifndef NGX_HTTP_GZIP
+#define NGX_HTTP_GZIP  1
+#endif
+
+
+#ifndef NGX_HTTP_SSI
+#define NGX_HTTP_SSI  1
+#endif
+
+
+#ifndef NGX_CRYPT
+#define NGX_CRYPT  1
+#endif
+
+
+#ifndef NGX_HTTP_X_FORWARDED_FOR
+#define NGX_HTTP_X_FORWARDED_FOR  1
+#endif
+
+
+#ifndef NGX_HTTP_X_FORWARDED_FOR
+#define NGX_HTTP_X_FORWARDED_FOR  1
+#endif
+
+
+#ifndef NGX_HTTP_UPSTREAM_ZONE
+#define NGX_HTTP_UPSTREAM_ZONE  1
+#endif
+
+
+#ifndef NGX_PCRE
+#define NGX_PCRE  1
+#endif
+
+
+#ifndef NGX_HAVE_PCRE_JIT
+#define NGX_HAVE_PCRE_JIT  1
+#endif
+
+
+#ifndef NGX_ZLIB
+#define NGX_ZLIB  1
+#endif
+
+
+#ifndef NGX_CONF_PREFIX
+#define NGX_CONF_PREFIX  "conf/"
+#endif
+
+
+#ifndef NGX_SBIN_PATH
+#define NGX_SBIN_PATH  "sbin/nginx"
+#endif
+
+
+#ifndef NGX_CONF_PATH
+#define NGX_CONF_PATH  "conf/nginx.conf"
+#endif
+
+
+#ifndef NGX_PID_PATH
+#define NGX_PID_PATH  "logs/nginx.pid"
+#endif
+
+
+#ifndef NGX_LOCK_PATH
+#define NGX_LOCK_PATH  "logs/nginx.lock"
+#endif
+
+
+#ifndef NGX_ERROR_LOG_PATH
+#define NGX_ERROR_LOG_PATH  "logs/error.log"
+#endif
+
+
+#ifndef NGX_HTTP_LOG_PATH
+#define NGX_HTTP_LOG_PATH  "logs/access.log"
+#endif
+
+
+#ifndef NGX_HTTP_CLIENT_TEMP_PATH
+#define NGX_HTTP_CLIENT_TEMP_PATH  "client_body_temp"
+#endif
+
+
+#ifndef NGX_HTTP_PROXY_TEMP_PATH
+#define NGX_HTTP_PROXY_TEMP_PATH  "proxy_temp"
+#endif
+
+
+#ifndef NGX_HTTP_FASTCGI_TEMP_PATH
+#define NGX_HTTP_FASTCGI_TEMP_PATH  "fastcgi_temp"
+#endif
+
+
+#ifndef NGX_HTTP_UWSGI_TEMP_PATH
+#define NGX_HTTP_UWSGI_TEMP_PATH  "uwsgi_temp"
+#endif
+
+
+#ifndef NGX_HTTP_SCGI_TEMP_PATH
+#define NGX_HTTP_SCGI_TEMP_PATH  "scgi_temp"
+#endif
+
+
+#ifndef NGX_SUPPRESS_WARN
+#define NGX_SUPPRESS_WARN  1
+#endif
+
+
+#ifndef NGX_SMP
+#define NGX_SMP  1
+#endif
+
+
+#ifndef NGX_USER
+#define NGX_USER  "nobody"
+#endif
+
+
+#ifndef NGX_GROUP
+#define NGX_GROUP  "nobody"
+#endif
+
--- nginx-1.11.3.dist/objs/ngx_modules.c	1970-01-01 00:00:00.000000000 +0000
+++ nginx-1.11.3/objs/ngx_modules.c	2017-02-02 18:11:45.262857053 +0000
@@ -0,0 +1,156 @@
+
+#include <ngx_config.h>
+#include <ngx_core.h>
+
+
+
+extern ngx_module_t  ngx_core_module;
+extern ngx_module_t  ngx_errlog_module;
+extern ngx_module_t  ngx_conf_module;
+extern ngx_module_t  ngx_regex_module;
+extern ngx_module_t  ngx_events_module;
+extern ngx_module_t  ngx_event_core_module;
+extern ngx_module_t  ngx_http_module;
+extern ngx_module_t  ngx_http_core_module;
+extern ngx_module_t  ngx_http_log_module;
+extern ngx_module_t  ngx_http_upstream_module;
+extern ngx_module_t  ngx_http_static_module;
+extern ngx_module_t  ngx_http_autoindex_module;
+extern ngx_module_t  ngx_http_index_module;
+extern ngx_module_t  ngx_http_auth_basic_module;
+extern ngx_module_t  ngx_http_access_module;
+extern ngx_module_t  ngx_http_limit_conn_module;
+extern ngx_module_t  ngx_http_limit_req_module;
+extern ngx_module_t  ngx_http_geo_module;
+extern ngx_module_t  ngx_http_map_module;
+extern ngx_module_t  ngx_http_split_clients_module;
+extern ngx_module_t  ngx_http_referer_module;
+extern ngx_module_t  ngx_http_rewrite_module;
+extern ngx_module_t  ngx_http_proxy_module;
+extern ngx_module_t  ngx_http_fastcgi_module;
+extern ngx_module_t  ngx_http_uwsgi_module;
+extern ngx_module_t  ngx_http_scgi_module;
+extern ngx_module_t  ngx_http_memcached_module;
+extern ngx_module_t  ngx_http_empty_gif_module;
+extern ngx_module_t  ngx_http_browser_module;
+extern ngx_module_t  ngx_http_upstream_hash_module;
+extern ngx_module_t  ngx_http_upstream_ip_hash_module;
+extern ngx_module_t  ngx_http_upstream_least_conn_module;
+extern ngx_module_t  ngx_http_upstream_keepalive_module;
+extern ngx_module_t  ngx_http_upstream_zone_module;
+extern ngx_module_t  ngx_http_write_filter_module;
+extern ngx_module_t  ngx_http_header_filter_module;
+extern ngx_module_t  ngx_http_chunked_filter_module;
+extern ngx_module_t  ngx_http_range_header_filter_module;
+extern ngx_module_t  ngx_http_gzip_filter_module;
+extern ngx_module_t  ngx_http_postpone_filter_module;
+extern ngx_module_t  ngx_http_ssi_filter_module;
+extern ngx_module_t  ngx_http_charset_filter_module;
+extern ngx_module_t  ngx_http_userid_filter_module;
+extern ngx_module_t  ngx_http_headers_filter_module;
+extern ngx_module_t  ngx_http_copy_filter_module;
+extern ngx_module_t  ngx_http_range_body_filter_module;
+extern ngx_module_t  ngx_http_not_modified_filter_module;
+
+ngx_module_t *ngx_modules[] = {
+    &ngx_core_module,
+    &ngx_errlog_module,
+    &ngx_conf_module,
+    &ngx_regex_module,
+    &ngx_events_module,
+    &ngx_event_core_module,
+    &ngx_http_module,
+    &ngx_http_core_module,
+    &ngx_http_log_module,
+    &ngx_http_upstream_module,
+    &ngx_http_static_module,
+    &ngx_http_autoindex_module,
+    &ngx_http_index_module,
+    &ngx_http_auth_basic_module,
+    &ngx_http_access_module,
+    &ngx_http_limit_conn_module,
+    &ngx_http_limit_req_module,
+    &ngx_http_geo_module,
+    &ngx_http_map_module,
+    &ngx_http_split_clients_module,
+    &ngx_http_referer_module,
+    &ngx_http_rewrite_module,
+    &ngx_http_proxy_module,
+    &ngx_http_fastcgi_module,
+    &ngx_http_uwsgi_module,
+    &ngx_http_scgi_module,
+    &ngx_http_memcached_module,
+    &ngx_http_empty_gif_module,
+    &ngx_http_browser_module,
+    &ngx_http_upstream_hash_module,
+    &ngx_http_upstream_ip_hash_module,
+    &ngx_http_upstream_least_conn_module,
+    &ngx_http_upstream_keepalive_module,
+    &ngx_http_upstream_zone_module,
+    &ngx_http_write_filter_module,
+    &ngx_http_header_filter_module,
+    &ngx_http_chunked_filter_module,
+    &ngx_http_range_header_filter_module,
+    &ngx_http_gzip_filter_module,
+    &ngx_http_postpone_filter_module,
+    &ngx_http_ssi_filter_module,
+    &ngx_http_charset_filter_module,
+    &ngx_http_userid_filter_module,
+    &ngx_http_headers_filter_module,
+    &ngx_http_copy_filter_module,
+    &ngx_http_range_body_filter_module,
+    &ngx_http_not_modified_filter_module,
+    NULL
+};
+
+char *ngx_module_names[] = {
+    "ngx_core_module",
+    "ngx_errlog_module",
+    "ngx_conf_module",
+    "ngx_regex_module",
+    "ngx_events_module",
+    "ngx_event_core_module",
+    "ngx_http_module",
+    "ngx_http_core_module",
+    "ngx_http_log_module",
+    "ngx_http_upstream_module",
+    "ngx_http_static_module",
+    "ngx_http_autoindex_module",
+    "ngx_http_index_module",
+    "ngx_http_auth_basic_module",
+    "ngx_http_access_module",
+    "ngx_http_limit_conn_module",
+    "ngx_http_limit_req_module",
+    "ngx_http_geo_module",
+    "ngx_http_map_module",
+    "ngx_http_split_clients_module",
+    "ngx_http_referer_module",
+    "ngx_http_rewrite_module",
+    "ngx_http_proxy_module",
+    "ngx_http_fastcgi_module",
+    "ngx_http_uwsgi_module",
+    "ngx_http_scgi_module",
+    "ngx_http_memcached_module",
+    "ngx_http_empty_gif_module",
+    "ngx_http_browser_module",
+    "ngx_http_upstream_hash_module",
+    "ngx_http_upstream_ip_hash_module",
+    "ngx_http_upstream_least_conn_module",
+    "ngx_http_upstream_keepalive_module",
+    "ngx_http_upstream_zone_module",
+    "ngx_http_write_filter_module",
+    "ngx_http_header_filter_module",
+    "ngx_http_chunked_filter_module",
+    "ngx_http_range_header_filter_module",
+    "ngx_http_gzip_filter_module",
+    "ngx_http_postpone_filter_module",
+    "ngx_http_ssi_filter_module",
+    "ngx_http_charset_filter_module",
+    "ngx_http_userid_filter_module",
+    "ngx_http_headers_filter_module",
+    "ngx_http_copy_filter_module",
+    "ngx_http_range_body_filter_module",
+    "ngx_http_not_modified_filter_module",
+    NULL
+};
+
--- nginx-1.11.3.dist/objs/Makefile	1970-01-01 00:00:00.000000000 +0000
+++ nginx-1.11.3/objs/Makefile	2017-02-02 18:37:11.770727563 +0000
@@ -0,0 +1,1196 @@
+
+CPP =	$(CC) -E
+LINK =	$(CC)
+
+
+ALL_INCS = -I src/core \
+	-I src/event \
+	-I src/event/modules \
+	-I src/os/unix \
+	-I objs \
+	-I src/http \
+	-I src/http/modules
+
+
+CORE_DEPS = src/core/nginx.h \
+	src/core/ngx_config.h \
+	src/core/ngx_core.h \
+	src/core/ngx_log.h \
+	src/core/ngx_palloc.h \
+	src/core/ngx_array.h \
+	src/core/ngx_list.h \
+	src/core/ngx_hash.h \
+	src/core/ngx_buf.h \
+	src/core/ngx_queue.h \
+	src/core/ngx_string.h \
+	src/core/ngx_parse.h \
+	src/core/ngx_parse_time.h \
+	src/core/ngx_inet.h \
+	src/core/ngx_file.h \
+	src/core/ngx_crc.h \
+	src/core/ngx_crc32.h \
+	src/core/ngx_murmurhash.h \
+	src/core/ngx_md5.h \
+	src/core/ngx_sha1.h \
+	src/core/ngx_rbtree.h \
+	src/core/ngx_radix_tree.h \
+	src/core/ngx_rwlock.h \
+	src/core/ngx_slab.h \
+	src/core/ngx_times.h \
+	src/core/ngx_shmtx.h \
+	src/core/ngx_connection.h \
+	src/core/ngx_cycle.h \
+	src/core/ngx_conf_file.h \
+	src/core/ngx_module.h \
+	src/core/ngx_resolver.h \
+	src/core/ngx_open_file_cache.h \
+	src/core/ngx_crypt.h \
+	src/core/ngx_proxy_protocol.h \
+	src/core/ngx_syslog.h \
+	src/event/ngx_event.h \
+	src/event/ngx_event_timer.h \
+	src/event/ngx_event_posted.h \
+	src/event/ngx_event_connect.h \
+	src/event/ngx_event_pipe.h \
+	src/os/unix/ngx_time.h \
+	src/os/unix/ngx_errno.h \
+	src/os/unix/ngx_alloc.h \
+	src/os/unix/ngx_files.h \
+	src/os/unix/ngx_channel.h \
+	src/os/unix/ngx_shmem.h \
+	src/os/unix/ngx_process.h \
+	src/os/unix/ngx_setaffinity.h \
+	src/os/unix/ngx_setproctitle.h \
+	src/os/unix/ngx_atomic.h \
+	src/os/unix/ngx_gcc_atomic_x86.h \
+	src/os/unix/ngx_thread.h \
+	src/os/unix/ngx_socket.h \
+	src/os/unix/ngx_os.h \
+	src/os/unix/ngx_user.h \
+	src/os/unix/ngx_dlopen.h \
+	src/os/unix/ngx_process_cycle.h \
+	src/os/unix/ngx_linux_config.h \
+	src/os/unix/ngx_linux.h \
+	src/core/ngx_regex.h \
+	objs/ngx_auto_config.h
+
+
+CORE_INCS = -I src/core \
+	-I src/event \
+	-I src/event/modules \
+	-I src/os/unix \
+	-I objs
+
+
+HTTP_DEPS = src/http/ngx_http.h \
+	src/http/ngx_http_request.h \
+	src/http/ngx_http_config.h \
+	src/http/ngx_http_core_module.h \
+	src/http/ngx_http_cache.h \
+	src/http/ngx_http_variables.h \
+	src/http/ngx_http_script.h \
+	src/http/ngx_http_upstream.h \
+	src/http/ngx_http_upstream_round_robin.h \
+	src/http/modules/ngx_http_ssi_filter_module.h
+
+
+HTTP_INCS = -I src/http \
+	-I src/http/modules
+
+
+build:	binary modules manpage
+
+binary:	objs/nginx
+
+objs/nginx:	objs/src/core/nginx.o \
+	objs/src/core/ngx_log.o \
+	objs/src/core/ngx_palloc.o \
+	objs/src/core/ngx_array.o \
+	objs/src/core/ngx_list.o \
+	objs/src/core/ngx_hash.o \
+	objs/src/core/ngx_buf.o \
+	objs/src/core/ngx_queue.o \
+	objs/src/core/ngx_output_chain.o \
+	objs/src/core/ngx_string.o \
+	objs/src/core/ngx_parse.o \
+	objs/src/core/ngx_parse_time.o \
+	objs/src/core/ngx_inet.o \
+	objs/src/core/ngx_file.o \
+	objs/src/core/ngx_crc32.o \
+	objs/src/core/ngx_murmurhash.o \
+	objs/src/core/ngx_md5.o \
+	objs/src/core/ngx_sha1.o \
+	objs/src/core/ngx_rbtree.o \
+	objs/src/core/ngx_radix_tree.o \
+	objs/src/core/ngx_slab.o \
+	objs/src/core/ngx_times.o \
+	objs/src/core/ngx_shmtx.o \
+	objs/src/core/ngx_connection.o \
+	objs/src/core/ngx_cycle.o \
+	objs/src/core/ngx_spinlock.o \
+	objs/src/core/ngx_rwlock.o \
+	objs/src/core/ngx_cpuinfo.o \
+	objs/src/core/ngx_conf_file.o \
+	objs/src/core/ngx_module.o \
+	objs/src/core/ngx_resolver.o \
+	objs/src/core/ngx_open_file_cache.o \
+	objs/src/core/ngx_crypt.o \
+	objs/src/core/ngx_proxy_protocol.o \
+	objs/src/core/ngx_syslog.o \
+	objs/src/event/ngx_event.o \
+	objs/src/event/ngx_event_timer.o \
+	objs/src/event/ngx_event_posted.o \
+	objs/src/event/ngx_event_accept.o \
+	objs/src/event/ngx_event_connect.o \
+	objs/src/event/ngx_event_pipe.o \
+	objs/src/os/unix/ngx_time.o \
+	objs/src/os/unix/ngx_errno.o \
+	objs/src/os/unix/ngx_alloc.o \
+	objs/src/os/unix/ngx_files.o \
+	objs/src/os/unix/ngx_socket.o \
+	objs/src/os/unix/ngx_recv.o \
+	objs/src/os/unix/ngx_readv_chain.o \
+	objs/src/os/unix/ngx_udp_recv.o \
+	objs/src/os/unix/ngx_send.o \
+	objs/src/os/unix/ngx_writev_chain.o \
+	objs/src/os/unix/ngx_udp_send.o \
+	objs/src/os/unix/ngx_channel.o \
+	objs/src/os/unix/ngx_shmem.o \
+	objs/src/os/unix/ngx_process.o \
+	objs/src/os/unix/ngx_daemon.o \
+	objs/src/os/unix/ngx_setaffinity.o \
+	objs/src/os/unix/ngx_setproctitle.o \
+	objs/src/os/unix/ngx_posix_init.o \
+	objs/src/os/unix/ngx_user.o \
+	objs/src/os/unix/ngx_dlopen.o \
+	objs/src/os/unix/ngx_process_cycle.o \
+	objs/src/os/unix/ngx_linux_init.o \
+	objs/src/os/unix/ngx_linux_sendfile_chain.o \
+	objs/src/core/ngx_regex.o \
+	objs/src/http/ngx_http.o \
+	objs/src/http/ngx_http_core_module.o \
+	objs/src/http/ngx_http_special_response.o \
+	objs/src/http/ngx_http_request.o \
+	objs/src/http/ngx_http_parse.o \
+	objs/src/http/modules/ngx_http_log_module.o \
+	objs/src/http/ngx_http_request_body.o \
+	objs/src/http/ngx_http_variables.o \
+	objs/src/http/ngx_http_script.o \
+	objs/src/http/ngx_http_upstream.o \
+	objs/src/http/ngx_http_upstream_round_robin.o \
+	objs/src/http/ngx_http_file_cache.o \
+	objs/src/http/ngx_http_write_filter_module.o \
+	objs/src/http/ngx_http_header_filter_module.o \
+	objs/src/http/modules/ngx_http_chunked_filter_module.o \
+	objs/src/http/modules/ngx_http_range_filter_module.o \
+	objs/src/http/modules/ngx_http_gzip_filter_module.o \
+	objs/src/http/ngx_http_postpone_filter_module.o \
+	objs/src/http/modules/ngx_http_ssi_filter_module.o \
+	objs/src/http/modules/ngx_http_charset_filter_module.o \
+	objs/src/http/modules/ngx_http_userid_filter_module.o \
+	objs/src/http/modules/ngx_http_headers_filter_module.o \
+	objs/src/http/ngx_http_copy_filter_module.o \
+	objs/src/http/modules/ngx_http_not_modified_filter_module.o \
+	objs/src/http/modules/ngx_http_static_module.o \
+	objs/src/http/modules/ngx_http_autoindex_module.o \
+	objs/src/http/modules/ngx_http_index_module.o \
+	objs/src/http/modules/ngx_http_auth_basic_module.o \
+	objs/src/http/modules/ngx_http_access_module.o \
+	objs/src/http/modules/ngx_http_limit_conn_module.o \
+	objs/src/http/modules/ngx_http_limit_req_module.o \
+	objs/src/http/modules/ngx_http_geo_module.o \
+	objs/src/http/modules/ngx_http_map_module.o \
+	objs/src/http/modules/ngx_http_split_clients_module.o \
+	objs/src/http/modules/ngx_http_referer_module.o \
+	objs/src/http/modules/ngx_http_rewrite_module.o \
+	objs/src/http/modules/ngx_http_proxy_module.o \
+	objs/src/http/modules/ngx_http_fastcgi_module.o \
+	objs/src/http/modules/ngx_http_uwsgi_module.o \
+	objs/src/http/modules/ngx_http_scgi_module.o \
+	objs/src/http/modules/ngx_http_memcached_module.o \
+	objs/src/http/modules/ngx_http_empty_gif_module.o \
+	objs/src/http/modules/ngx_http_browser_module.o \
+	objs/src/http/modules/ngx_http_upstream_hash_module.o \
+	objs/src/http/modules/ngx_http_upstream_ip_hash_module.o \
+	objs/src/http/modules/ngx_http_upstream_least_conn_module.o \
+	objs/src/http/modules/ngx_http_upstream_keepalive_module.o \
+	objs/src/http/modules/ngx_http_upstream_zone_module.o \
+	objs/ngx_modules.o
+
+	$(LINK) -o objs/nginx \
+	objs/src/core/nginx.o \
+	objs/src/core/ngx_log.o \
+	objs/src/core/ngx_palloc.o \
+	objs/src/core/ngx_array.o \
+	objs/src/core/ngx_list.o \
+	objs/src/core/ngx_hash.o \
+	objs/src/core/ngx_buf.o \
+	objs/src/core/ngx_queue.o \
+	objs/src/core/ngx_output_chain.o \
+	objs/src/core/ngx_string.o \
+	objs/src/core/ngx_parse.o \
+	objs/src/core/ngx_parse_time.o \
+	objs/src/core/ngx_inet.o \
+	objs/src/core/ngx_file.o \
+	objs/src/core/ngx_crc32.o \
+	objs/src/core/ngx_murmurhash.o \
+	objs/src/core/ngx_md5.o \
+	objs/src/core/ngx_sha1.o \
+	objs/src/core/ngx_rbtree.o \
+	objs/src/core/ngx_radix_tree.o \
+	objs/src/core/ngx_slab.o \
+	objs/src/core/ngx_times.o \
+	objs/src/core/ngx_shmtx.o \
+	objs/src/core/ngx_connection.o \
+	objs/src/core/ngx_cycle.o \
+	objs/src/core/ngx_spinlock.o \
+	objs/src/core/ngx_rwlock.o \
+	objs/src/core/ngx_cpuinfo.o \
+	objs/src/core/ngx_conf_file.o \
+	objs/src/core/ngx_module.o \
+	objs/src/core/ngx_resolver.o \
+	objs/src/core/ngx_open_file_cache.o \
+	objs/src/core/ngx_crypt.o \
+	objs/src/core/ngx_proxy_protocol.o \
+	objs/src/core/ngx_syslog.o \
+	objs/src/event/ngx_event.o \
+	objs/src/event/ngx_event_timer.o \
+	objs/src/event/ngx_event_posted.o \
+	objs/src/event/ngx_event_accept.o \
+	objs/src/event/ngx_event_connect.o \
+	objs/src/event/ngx_event_pipe.o \
+	objs/src/os/unix/ngx_time.o \
+	objs/src/os/unix/ngx_errno.o \
+	objs/src/os/unix/ngx_alloc.o \
+	objs/src/os/unix/ngx_files.o \
+	objs/src/os/unix/ngx_socket.o \
+	objs/src/os/unix/ngx_recv.o \
+	objs/src/os/unix/ngx_readv_chain.o \
+	objs/src/os/unix/ngx_udp_recv.o \
+	objs/src/os/unix/ngx_send.o \
+	objs/src/os/unix/ngx_writev_chain.o \
+	objs/src/os/unix/ngx_udp_send.o \
+	objs/src/os/unix/ngx_channel.o \
+	objs/src/os/unix/ngx_shmem.o \
+	objs/src/os/unix/ngx_process.o \
+	objs/src/os/unix/ngx_daemon.o \
+	objs/src/os/unix/ngx_setaffinity.o \
+	objs/src/os/unix/ngx_setproctitle.o \
+	objs/src/os/unix/ngx_posix_init.o \
+	objs/src/os/unix/ngx_user.o \
+	objs/src/os/unix/ngx_dlopen.o \
+	objs/src/os/unix/ngx_process_cycle.o \
+	objs/src/os/unix/ngx_linux_init.o \
+	objs/src/os/unix/ngx_linux_sendfile_chain.o \
+	objs/src/core/ngx_regex.o \
+	objs/src/http/ngx_http.o \
+	objs/src/http/ngx_http_core_module.o \
+	objs/src/http/ngx_http_special_response.o \
+	objs/src/http/ngx_http_request.o \
+	objs/src/http/ngx_http_parse.o \
+	objs/src/http/modules/ngx_http_log_module.o \
+	objs/src/http/ngx_http_request_body.o \
+	objs/src/http/ngx_http_variables.o \
+	objs/src/http/ngx_http_script.o \
+	objs/src/http/ngx_http_upstream.o \
+	objs/src/http/ngx_http_upstream_round_robin.o \
+	objs/src/http/ngx_http_file_cache.o \
+	objs/src/http/ngx_http_write_filter_module.o \
+	objs/src/http/ngx_http_header_filter_module.o \
+	objs/src/http/modules/ngx_http_chunked_filter_module.o \
+	objs/src/http/modules/ngx_http_range_filter_module.o \
+	objs/src/http/modules/ngx_http_gzip_filter_module.o \
+	objs/src/http/ngx_http_postpone_filter_module.o \
+	objs/src/http/modules/ngx_http_ssi_filter_module.o \
+	objs/src/http/modules/ngx_http_charset_filter_module.o \
+	objs/src/http/modules/ngx_http_userid_filter_module.o \
+	objs/src/http/modules/ngx_http_headers_filter_module.o \
+	objs/src/http/ngx_http_copy_filter_module.o \
+	objs/src/http/modules/ngx_http_not_modified_filter_module.o \
+	objs/src/http/modules/ngx_http_static_module.o \
+	objs/src/http/modules/ngx_http_autoindex_module.o \
+	objs/src/http/modules/ngx_http_index_module.o \
+	objs/src/http/modules/ngx_http_auth_basic_module.o \
+	objs/src/http/modules/ngx_http_access_module.o \
+	objs/src/http/modules/ngx_http_limit_conn_module.o \
+	objs/src/http/modules/ngx_http_limit_req_module.o \
+	objs/src/http/modules/ngx_http_geo_module.o \
+	objs/src/http/modules/ngx_http_map_module.o \
+	objs/src/http/modules/ngx_http_split_clients_module.o \
+	objs/src/http/modules/ngx_http_referer_module.o \
+	objs/src/http/modules/ngx_http_rewrite_module.o \
+	objs/src/http/modules/ngx_http_proxy_module.o \
+	objs/src/http/modules/ngx_http_fastcgi_module.o \
+	objs/src/http/modules/ngx_http_uwsgi_module.o \
+	objs/src/http/modules/ngx_http_scgi_module.o \
+	objs/src/http/modules/ngx_http_memcached_module.o \
+	objs/src/http/modules/ngx_http_empty_gif_module.o \
+	objs/src/http/modules/ngx_http_browser_module.o \
+	objs/src/http/modules/ngx_http_upstream_hash_module.o \
+	objs/src/http/modules/ngx_http_upstream_ip_hash_module.o \
+	objs/src/http/modules/ngx_http_upstream_least_conn_module.o \
+	objs/src/http/modules/ngx_http_upstream_keepalive_module.o \
+	objs/src/http/modules/ngx_http_upstream_zone_module.o \
+	objs/ngx_modules.o \
+	$(LDFLAGS) -ldl -lpthread -lcrypt -lpcre -lz
+	
+
+
+modules:
+
+objs/ngx_modules.o:	$(CORE_DEPS) \
+	objs/ngx_modules.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/ngx_modules.o \
+		objs/ngx_modules.c
+
+
+objs/src/core/nginx.o:	$(CORE_DEPS) \
+	src/core/nginx.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/nginx.o \
+		src/core/nginx.c
+
+
+objs/src/core/ngx_log.o:	$(CORE_DEPS) \
+	src/core/ngx_log.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_log.o \
+		src/core/ngx_log.c
+
+
+objs/src/core/ngx_palloc.o:	$(CORE_DEPS) \
+	src/core/ngx_palloc.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_palloc.o \
+		src/core/ngx_palloc.c
+
+
+objs/src/core/ngx_array.o:	$(CORE_DEPS) \
+	src/core/ngx_array.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_array.o \
+		src/core/ngx_array.c
+
+
+objs/src/core/ngx_list.o:	$(CORE_DEPS) \
+	src/core/ngx_list.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_list.o \
+		src/core/ngx_list.c
+
+
+objs/src/core/ngx_hash.o:	$(CORE_DEPS) \
+	src/core/ngx_hash.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_hash.o \
+		src/core/ngx_hash.c
+
+
+objs/src/core/ngx_buf.o:	$(CORE_DEPS) \
+	src/core/ngx_buf.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_buf.o \
+		src/core/ngx_buf.c
+
+
+objs/src/core/ngx_queue.o:	$(CORE_DEPS) \
+	src/core/ngx_queue.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_queue.o \
+		src/core/ngx_queue.c
+
+
+objs/src/core/ngx_output_chain.o:	$(CORE_DEPS) \
+	src/core/ngx_output_chain.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_output_chain.o \
+		src/core/ngx_output_chain.c
+
+
+objs/src/core/ngx_string.o:	$(CORE_DEPS) \
+	src/core/ngx_string.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_string.o \
+		src/core/ngx_string.c
+
+
+objs/src/core/ngx_parse.o:	$(CORE_DEPS) \
+	src/core/ngx_parse.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_parse.o \
+		src/core/ngx_parse.c
+
+
+objs/src/core/ngx_parse_time.o:	$(CORE_DEPS) \
+	src/core/ngx_parse_time.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_parse_time.o \
+		src/core/ngx_parse_time.c
+
+
+objs/src/core/ngx_inet.o:	$(CORE_DEPS) \
+	src/core/ngx_inet.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_inet.o \
+		src/core/ngx_inet.c
+
+
+objs/src/core/ngx_file.o:	$(CORE_DEPS) \
+	src/core/ngx_file.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_file.o \
+		src/core/ngx_file.c
+
+
+objs/src/core/ngx_crc32.o:	$(CORE_DEPS) \
+	src/core/ngx_crc32.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_crc32.o \
+		src/core/ngx_crc32.c
+
+
+objs/src/core/ngx_murmurhash.o:	$(CORE_DEPS) \
+	src/core/ngx_murmurhash.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_murmurhash.o \
+		src/core/ngx_murmurhash.c
+
+
+objs/src/core/ngx_md5.o:	$(CORE_DEPS) \
+	src/core/ngx_md5.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_md5.o \
+		src/core/ngx_md5.c
+
+
+objs/src/core/ngx_sha1.o:	$(CORE_DEPS) \
+	src/core/ngx_sha1.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_sha1.o \
+		src/core/ngx_sha1.c
+
+
+objs/src/core/ngx_rbtree.o:	$(CORE_DEPS) \
+	src/core/ngx_rbtree.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_rbtree.o \
+		src/core/ngx_rbtree.c
+
+
+objs/src/core/ngx_radix_tree.o:	$(CORE_DEPS) \
+	src/core/ngx_radix_tree.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_radix_tree.o \
+		src/core/ngx_radix_tree.c
+
+
+objs/src/core/ngx_slab.o:	$(CORE_DEPS) \
+	src/core/ngx_slab.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_slab.o \
+		src/core/ngx_slab.c
+
+
+objs/src/core/ngx_times.o:	$(CORE_DEPS) \
+	src/core/ngx_times.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_times.o \
+		src/core/ngx_times.c
+
+
+objs/src/core/ngx_shmtx.o:	$(CORE_DEPS) \
+	src/core/ngx_shmtx.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_shmtx.o \
+		src/core/ngx_shmtx.c
+
+
+objs/src/core/ngx_connection.o:	$(CORE_DEPS) \
+	src/core/ngx_connection.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_connection.o \
+		src/core/ngx_connection.c
+
+
+objs/src/core/ngx_cycle.o:	$(CORE_DEPS) \
+	src/core/ngx_cycle.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_cycle.o \
+		src/core/ngx_cycle.c
+
+
+objs/src/core/ngx_spinlock.o:	$(CORE_DEPS) \
+	src/core/ngx_spinlock.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_spinlock.o \
+		src/core/ngx_spinlock.c
+
+
+objs/src/core/ngx_rwlock.o:	$(CORE_DEPS) \
+	src/core/ngx_rwlock.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_rwlock.o \
+		src/core/ngx_rwlock.c
+
+
+objs/src/core/ngx_cpuinfo.o:	$(CORE_DEPS) \
+	src/core/ngx_cpuinfo.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_cpuinfo.o \
+		src/core/ngx_cpuinfo.c
+
+
+objs/src/core/ngx_conf_file.o:	$(CORE_DEPS) \
+	src/core/ngx_conf_file.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_conf_file.o \
+		src/core/ngx_conf_file.c
+
+
+objs/src/core/ngx_module.o:	$(CORE_DEPS) \
+	src/core/ngx_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_module.o \
+		src/core/ngx_module.c
+
+
+objs/src/core/ngx_resolver.o:	$(CORE_DEPS) \
+	src/core/ngx_resolver.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_resolver.o \
+		src/core/ngx_resolver.c
+
+
+objs/src/core/ngx_open_file_cache.o:	$(CORE_DEPS) \
+	src/core/ngx_open_file_cache.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_open_file_cache.o \
+		src/core/ngx_open_file_cache.c
+
+
+objs/src/core/ngx_crypt.o:	$(CORE_DEPS) \
+	src/core/ngx_crypt.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_crypt.o \
+		src/core/ngx_crypt.c
+
+
+objs/src/core/ngx_proxy_protocol.o:	$(CORE_DEPS) \
+	src/core/ngx_proxy_protocol.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_proxy_protocol.o \
+		src/core/ngx_proxy_protocol.c
+
+
+objs/src/core/ngx_syslog.o:	$(CORE_DEPS) \
+	src/core/ngx_syslog.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_syslog.o \
+		src/core/ngx_syslog.c
+
+
+objs/src/event/ngx_event.o:	$(CORE_DEPS) \
+	src/event/ngx_event.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/event/ngx_event.o \
+		src/event/ngx_event.c
+
+
+objs/src/event/ngx_event_timer.o:	$(CORE_DEPS) \
+	src/event/ngx_event_timer.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/event/ngx_event_timer.o \
+		src/event/ngx_event_timer.c
+
+
+objs/src/event/ngx_event_posted.o:	$(CORE_DEPS) \
+	src/event/ngx_event_posted.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/event/ngx_event_posted.o \
+		src/event/ngx_event_posted.c
+
+
+objs/src/event/ngx_event_accept.o:	$(CORE_DEPS) \
+	src/event/ngx_event_accept.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/event/ngx_event_accept.o \
+		src/event/ngx_event_accept.c
+
+
+objs/src/event/ngx_event_connect.o:	$(CORE_DEPS) \
+	src/event/ngx_event_connect.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/event/ngx_event_connect.o \
+		src/event/ngx_event_connect.c
+
+
+objs/src/event/ngx_event_pipe.o:	$(CORE_DEPS) \
+	src/event/ngx_event_pipe.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/event/ngx_event_pipe.o \
+		src/event/ngx_event_pipe.c
+
+
+objs/src/os/unix/ngx_time.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_time.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_time.o \
+		src/os/unix/ngx_time.c
+
+
+objs/src/os/unix/ngx_errno.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_errno.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_errno.o \
+		src/os/unix/ngx_errno.c
+
+
+objs/src/os/unix/ngx_alloc.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_alloc.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_alloc.o \
+		src/os/unix/ngx_alloc.c
+
+
+objs/src/os/unix/ngx_files.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_files.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_files.o \
+		src/os/unix/ngx_files.c
+
+
+objs/src/os/unix/ngx_socket.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_socket.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_socket.o \
+		src/os/unix/ngx_socket.c
+
+
+objs/src/os/unix/ngx_recv.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_recv.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_recv.o \
+		src/os/unix/ngx_recv.c
+
+
+objs/src/os/unix/ngx_readv_chain.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_readv_chain.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_readv_chain.o \
+		src/os/unix/ngx_readv_chain.c
+
+
+objs/src/os/unix/ngx_udp_recv.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_udp_recv.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_udp_recv.o \
+		src/os/unix/ngx_udp_recv.c
+
+
+objs/src/os/unix/ngx_send.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_send.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_send.o \
+		src/os/unix/ngx_send.c
+
+
+objs/src/os/unix/ngx_writev_chain.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_writev_chain.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_writev_chain.o \
+		src/os/unix/ngx_writev_chain.c
+
+
+objs/src/os/unix/ngx_udp_send.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_udp_send.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_udp_send.o \
+		src/os/unix/ngx_udp_send.c
+
+
+objs/src/os/unix/ngx_channel.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_channel.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_channel.o \
+		src/os/unix/ngx_channel.c
+
+
+objs/src/os/unix/ngx_shmem.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_shmem.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_shmem.o \
+		src/os/unix/ngx_shmem.c
+
+
+objs/src/os/unix/ngx_process.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_process.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_process.o \
+		src/os/unix/ngx_process.c
+
+
+objs/src/os/unix/ngx_daemon.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_daemon.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_daemon.o \
+		src/os/unix/ngx_daemon.c
+
+
+objs/src/os/unix/ngx_setaffinity.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_setaffinity.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_setaffinity.o \
+		src/os/unix/ngx_setaffinity.c
+
+
+objs/src/os/unix/ngx_setproctitle.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_setproctitle.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_setproctitle.o \
+		src/os/unix/ngx_setproctitle.c
+
+
+objs/src/os/unix/ngx_posix_init.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_posix_init.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_posix_init.o \
+		src/os/unix/ngx_posix_init.c
+
+
+objs/src/os/unix/ngx_user.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_user.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_user.o \
+		src/os/unix/ngx_user.c
+
+
+objs/src/os/unix/ngx_dlopen.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_dlopen.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_dlopen.o \
+		src/os/unix/ngx_dlopen.c
+
+
+objs/src/os/unix/ngx_process_cycle.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_process_cycle.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_process_cycle.o \
+		src/os/unix/ngx_process_cycle.c
+
+
+objs/src/os/unix/ngx_linux_init.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_linux_init.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_linux_init.o \
+		src/os/unix/ngx_linux_init.c
+
+
+objs/src/os/unix/ngx_linux_sendfile_chain.o:	$(CORE_DEPS) \
+	src/os/unix/ngx_linux_sendfile_chain.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/os/unix/ngx_linux_sendfile_chain.o \
+		src/os/unix/ngx_linux_sendfile_chain.c
+
+
+objs/src/core/ngx_regex.o:	$(CORE_DEPS) \
+	src/core/ngx_regex.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) \
+		-o objs/src/core/ngx_regex.o \
+		src/core/ngx_regex.c
+
+
+objs/src/http/ngx_http.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http.o \
+		src/http/ngx_http.c
+
+
+objs/src/http/ngx_http_core_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_core_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_core_module.o \
+		src/http/ngx_http_core_module.c
+
+
+objs/src/http/ngx_http_special_response.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_special_response.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_special_response.o \
+		src/http/ngx_http_special_response.c
+
+
+objs/src/http/ngx_http_request.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_request.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_request.o \
+		src/http/ngx_http_request.c
+
+
+objs/src/http/ngx_http_parse.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_parse.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_parse.o \
+		src/http/ngx_http_parse.c
+
+
+objs/src/http/modules/ngx_http_log_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_log_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_log_module.o \
+		src/http/modules/ngx_http_log_module.c
+
+
+objs/src/http/ngx_http_request_body.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_request_body.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_request_body.o \
+		src/http/ngx_http_request_body.c
+
+
+objs/src/http/ngx_http_variables.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_variables.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_variables.o \
+		src/http/ngx_http_variables.c
+
+
+objs/src/http/ngx_http_script.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_script.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_script.o \
+		src/http/ngx_http_script.c
+
+
+objs/src/http/ngx_http_upstream.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_upstream.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_upstream.o \
+		src/http/ngx_http_upstream.c
+
+
+objs/src/http/ngx_http_upstream_round_robin.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_upstream_round_robin.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_upstream_round_robin.o \
+		src/http/ngx_http_upstream_round_robin.c
+
+
+objs/src/http/ngx_http_file_cache.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_file_cache.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_file_cache.o \
+		src/http/ngx_http_file_cache.c
+
+
+objs/src/http/ngx_http_write_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_write_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_write_filter_module.o \
+		src/http/ngx_http_write_filter_module.c
+
+
+objs/src/http/ngx_http_header_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_header_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_header_filter_module.o \
+		src/http/ngx_http_header_filter_module.c
+
+
+objs/src/http/modules/ngx_http_chunked_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_chunked_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_chunked_filter_module.o \
+		src/http/modules/ngx_http_chunked_filter_module.c
+
+
+objs/src/http/modules/ngx_http_range_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_range_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_range_filter_module.o \
+		src/http/modules/ngx_http_range_filter_module.c
+
+
+objs/src/http/modules/ngx_http_gzip_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_gzip_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_gzip_filter_module.o \
+		src/http/modules/ngx_http_gzip_filter_module.c
+
+
+objs/src/http/ngx_http_postpone_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_postpone_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_postpone_filter_module.o \
+		src/http/ngx_http_postpone_filter_module.c
+
+
+objs/src/http/modules/ngx_http_ssi_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_ssi_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_ssi_filter_module.o \
+		src/http/modules/ngx_http_ssi_filter_module.c
+
+
+objs/src/http/modules/ngx_http_charset_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_charset_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_charset_filter_module.o \
+		src/http/modules/ngx_http_charset_filter_module.c
+
+
+objs/src/http/modules/ngx_http_userid_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_userid_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_userid_filter_module.o \
+		src/http/modules/ngx_http_userid_filter_module.c
+
+
+objs/src/http/modules/ngx_http_headers_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_headers_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_headers_filter_module.o \
+		src/http/modules/ngx_http_headers_filter_module.c
+
+
+objs/src/http/ngx_http_copy_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/ngx_http_copy_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/ngx_http_copy_filter_module.o \
+		src/http/ngx_http_copy_filter_module.c
+
+
+objs/src/http/modules/ngx_http_not_modified_filter_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_not_modified_filter_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_not_modified_filter_module.o \
+		src/http/modules/ngx_http_not_modified_filter_module.c
+
+
+objs/src/http/modules/ngx_http_static_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_static_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_static_module.o \
+		src/http/modules/ngx_http_static_module.c
+
+
+objs/src/http/modules/ngx_http_autoindex_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_autoindex_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_autoindex_module.o \
+		src/http/modules/ngx_http_autoindex_module.c
+
+
+objs/src/http/modules/ngx_http_index_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_index_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_index_module.o \
+		src/http/modules/ngx_http_index_module.c
+
+
+objs/src/http/modules/ngx_http_auth_basic_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_auth_basic_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_auth_basic_module.o \
+		src/http/modules/ngx_http_auth_basic_module.c
+
+
+objs/src/http/modules/ngx_http_access_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_access_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_access_module.o \
+		src/http/modules/ngx_http_access_module.c
+
+
+objs/src/http/modules/ngx_http_limit_conn_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_limit_conn_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_limit_conn_module.o \
+		src/http/modules/ngx_http_limit_conn_module.c
+
+
+objs/src/http/modules/ngx_http_limit_req_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_limit_req_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_limit_req_module.o \
+		src/http/modules/ngx_http_limit_req_module.c
+
+
+objs/src/http/modules/ngx_http_geo_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_geo_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_geo_module.o \
+		src/http/modules/ngx_http_geo_module.c
+
+
+objs/src/http/modules/ngx_http_map_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_map_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_map_module.o \
+		src/http/modules/ngx_http_map_module.c
+
+
+objs/src/http/modules/ngx_http_split_clients_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_split_clients_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_split_clients_module.o \
+		src/http/modules/ngx_http_split_clients_module.c
+
+
+objs/src/http/modules/ngx_http_referer_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_referer_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_referer_module.o \
+		src/http/modules/ngx_http_referer_module.c
+
+
+objs/src/http/modules/ngx_http_rewrite_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_rewrite_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_rewrite_module.o \
+		src/http/modules/ngx_http_rewrite_module.c
+
+
+objs/src/http/modules/ngx_http_proxy_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_proxy_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_proxy_module.o \
+		src/http/modules/ngx_http_proxy_module.c
+
+
+objs/src/http/modules/ngx_http_fastcgi_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_fastcgi_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_fastcgi_module.o \
+		src/http/modules/ngx_http_fastcgi_module.c
+
+
+objs/src/http/modules/ngx_http_uwsgi_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_uwsgi_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_uwsgi_module.o \
+		src/http/modules/ngx_http_uwsgi_module.c
+
+
+objs/src/http/modules/ngx_http_scgi_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_scgi_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_scgi_module.o \
+		src/http/modules/ngx_http_scgi_module.c
+
+
+objs/src/http/modules/ngx_http_memcached_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_memcached_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_memcached_module.o \
+		src/http/modules/ngx_http_memcached_module.c
+
+
+objs/src/http/modules/ngx_http_empty_gif_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_empty_gif_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_empty_gif_module.o \
+		src/http/modules/ngx_http_empty_gif_module.c
+
+
+objs/src/http/modules/ngx_http_browser_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_browser_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_browser_module.o \
+		src/http/modules/ngx_http_browser_module.c
+
+
+objs/src/http/modules/ngx_http_upstream_hash_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_upstream_hash_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_upstream_hash_module.o \
+		src/http/modules/ngx_http_upstream_hash_module.c
+
+
+objs/src/http/modules/ngx_http_upstream_ip_hash_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_upstream_ip_hash_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_upstream_ip_hash_module.o \
+		src/http/modules/ngx_http_upstream_ip_hash_module.c
+
+
+objs/src/http/modules/ngx_http_upstream_least_conn_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_upstream_least_conn_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_upstream_least_conn_module.o \
+		src/http/modules/ngx_http_upstream_least_conn_module.c
+
+
+objs/src/http/modules/ngx_http_upstream_keepalive_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_upstream_keepalive_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_upstream_keepalive_module.o \
+		src/http/modules/ngx_http_upstream_keepalive_module.c
+
+
+objs/src/http/modules/ngx_http_upstream_zone_module.o:	$(CORE_DEPS) $(HTTP_DEPS) \
+	src/http/modules/ngx_http_upstream_zone_module.c
+	$(CC) -c $(CFLAGS) $(CORE_INCS) $(HTTP_INCS) \
+		-o objs/src/http/modules/ngx_http_upstream_zone_module.o \
+		src/http/modules/ngx_http_upstream_zone_module.c
+
+
+manpage:	objs/nginx.8
+
+objs/nginx.8:	man/nginx.8 objs/ngx_auto_config.h
+	sed -e "s|%%PREFIX%%||" \
+		-e "s|%%PID_PATH%%|/logs/nginx.pid|" \
+		-e "s|%%CONF_PATH%%|/conf/nginx.conf|" \
+		-e "s|%%ERROR_LOG_PATH%%|/logs/error.log|" \
+		< man/nginx.8 > $@
+
+install:	build 
+	test -d '$(DESTDIR)' || mkdir -p '$(DESTDIR)'
+
+	test -d '$(DESTDIR)/sbin' \
+		|| mkdir -p '$(DESTDIR)/sbin'
+	test ! -f '$(DESTDIR)/sbin/nginx' \
+		|| mv '$(DESTDIR)/sbin/nginx' \
+			'$(DESTDIR)/sbin/nginx.old'
+	cp objs/nginx '$(DESTDIR)/sbin/nginx'
+
+	test -d '$(DESTDIR)/etc/nginx' \
+		|| mkdir -p '$(DESTDIR)/etc/nginx'
+
+	cp conf/koi-win '$(DESTDIR)/etc/nginx'
+	cp conf/koi-utf '$(DESTDIR)/etc/nginx'
+	cp conf/win-utf '$(DESTDIR)/etc/nginx'
+
+	test -f '$(DESTDIR)/etc/nginx/mime.types' \
+		|| cp conf/mime.types '$(DESTDIR)/etc/nginx'
+	cp conf/mime.types '$(DESTDIR)/etc/nginx/mime.types.default'
+
+	test -f '$(DESTDIR)/etc/nginx/fastcgi_params' \
+		|| cp conf/fastcgi_params '$(DESTDIR)/etc/nginx'
+	cp conf/fastcgi_params \
+		'$(DESTDIR)/etc/nginx/fastcgi_params.default'
+
+	test -f '$(DESTDIR)/etc/nginx/fastcgi.conf' \
+		|| cp conf/fastcgi.conf '$(DESTDIR)/etc/nginx'
+	cp conf/fastcgi.conf '$(DESTDIR)/etc/nginx/fastcgi.conf.default'
+
+	test -f '$(DESTDIR)/etc/nginx/uwsgi_params' \
+		|| cp conf/uwsgi_params '$(DESTDIR)/etc/nginx'
+	cp conf/uwsgi_params \
+		'$(DESTDIR)/etc/nginx/uwsgi_params.default'
+
+	test -f '$(DESTDIR)/etc/nginx/scgi_params' \
+		|| cp conf/scgi_params '$(DESTDIR)/etc/nginx'
+	cp conf/scgi_params \
+		'$(DESTDIR)/etc/nginx/scgi_params.default'
+
+	test -f '$(DESTDIR)/etc/nginx/nginx.conf' \
+		|| cp conf/nginx.conf '$(DESTDIR)/etc/nginx/nginx.conf'
+	cp conf/nginx.conf '$(DESTDIR)/etc/nginx/nginx.conf.default'
+
+	test -d '$(DESTDIR)/var/log/nginx' \
+		|| mkdir -p '$(DESTDIR)/var/log/nginx'
+
+	test -d '$(DESTDIR)/var/www' \
+		|| cp -R html '$(DESTDIR)/var/www'
