diff -ru pango-1.40.11.orig/configure pango-1.40.11/configure
--- pango-1.40.11.orig/configure	2017-08-19 15:28:44.000000000 +0200
+++ pango-1.40.11/configure	2017-08-25 21:29:22.922827837 +0200
@@ -19005,33 +19005,11 @@
 fi
 
 if $have_cairo ; then
-  pango_save_libs=$LIBS
-  LIBS="$LIBS $CAIRO_LIBS"
-  pango_save_ldflags=$LDFLAGS
-  LDFLAGS="$LDFLAGS $CAIRO_LIBS"
-
-  have_cairo=false
-  cairo_font_backends=""
-
-  if $have_freetype; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"cairo-ft >= \$cairo_required\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "cairo-ft >= $cairo_required") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  have_cairo_freetype=true
-else
-  :
-fi
-  fi
-  if $have_cairo_freetype; then
 
 $as_echo "#define HAVE_CAIRO_FREETYPE 1" >>confdefs.h
 
     cairo_font_backends="freetype $cairo_font_backends"
     have_cairo=true
-  fi
 
   if $have_win32; then
     if test -n "$PKG_CONFIG" && \

--- pango-1.40.11/pango/pangocairo-fontmap.c.orig	2016-10-22 07:00:37.000000000 +0200
+++ pango-1.40.11/pango/pangocairo-fontmap.c	2017-08-25 22:30:01.766561892 +0200
@@ -35,6 +35,10 @@
 #  include "pangocairo-fc.h"
 #endif
 
+PANGO_AVAILABLE_IN_ALL
+GType __attribute__((weak)) pango_cairo_fc_font_map_get_type (void) {
+	return NULL;
+}
 
 typedef PangoCairoFontMapIface PangoCairoFontMapInterface;
 G_DEFINE_INTERFACE (PangoCairoFontMap, pango_cairo_font_map, PANGO_TYPE_FONT_MAP)
