diff -ru perl-5.22.1.orig/installman perl-5.22.1/installman
--- perl-5.22.1.orig/installman	2017-11-17 23:35:34.386593179 +0100
+++ perl-5.22.1/installman	2017-11-17 23:03:41.388858972 +0100
@@ -146,9 +146,10 @@
 	    next;
 	}
 
-	if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'uwin' || $^O eq 'cygwin') {
+	#Fix me later?
+	#if ($^O eq 'os2' || $^O eq 'amigaos' || $^O eq 'uwin' || $^O eq 'cygwin' || $^O eq 'midipix') {
             $manpage =~ s#::#.#g;
-	}
+	#}
         my $tmp = "${mandir}/${manpage}.tmp";
 	$manpage = "${mandir}/${manpage}.${manext}";
 
diff -ru perl-5.22.1.orig/Makefile perl-5.22.1/Makefile
--- perl-5.22.1.orig/Makefile	2017-11-17 19:06:50.298908618 +0100
+++ perl-5.22.1/Makefile	2017-11-17 23:32:21.687695934 +0100
@@ -135,7 +135,7 @@
 perl$x: perlmain$o $(LIBPERL) $(static_tgt) static.list ext.libs
 	$(eval extlibs=$(shell cat ext.libs))
 	$(eval statars=$(shell cat static.list))
-	$(CC) $(LDFLAGS) -o $@ $(filter %$o,$^) $(LIBPERL) $(statars) $(LIBS) $(extlibs)
+	$(CC) $(LDFLAGS) -o $@ $(filter %$o,$^) -L. -lperl $(statars) $(LIBS) $(extlibs)
 
 %$o: %.c config.h
 	$(CC) $(CFLAGS) -c -o $@ $<
