diff -ru php-8.0.11.orig/build/ php-8.0.11/build/libtool.m4
--- php-8.0.11.orig/build/libtool.m4    2021-09-21 19:07:44.000000000 +0200
+++ php-8.0.11/build/libtool.m4 2021-09-29 22:21:23.610917784 +0200
@@ -56,10 +56,6 @@
 # This can be used to rebuild libtool when needed
 LIBTOOL_DEPS="$ac_aux_dir/ltmain.sh"

-# Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
-AC_SUBST(LIBTOOL)dnl
-
 # Prevent multiple expansion
 define([AC_PROG_LIBTOOL], [])
 ])# _AC_PROG_LIBTOOL
diff -ru php-8.0.8.orig/TSRM/TSRM.h php-8.0.8/TSRM/TSRM.h
--- php-8.0.8.orig/TSRM/TSRM.h  2021-06-29 09:41:19.000000000 +0200
+++ php-8.0.8/TSRM/TSRM.h       2021-07-17 21:51:11.409950944 +0200
@@ -147,7 +147,7 @@
 # define __has_attribute(x) 0
 #endif

-#if !__has_attribute(tls_model) || defined(__FreeBSD__) || defined(__MUSL__)
+#if !__has_attribute(tls_model) || defined(__FreeBSD__) || defined(__MUSL__) || defined(__midipix__)
 # define TSRM_TLS_MODEL_ATTR
 #elif __PIC__
 # define TSRM_TLS_MODEL_ATTR __attribute__((tls_model("initial-exec")))
diff -ru php-8.0.1.orig/configure php-8.0.1/configure
--- php-8.0.1.orig/configure	2021-01-05 14:54:54.000000000 +0100
+++ php-8.0.1/configure	2021-01-13 23:01:56.427279925 +0100
@@ -52030,8 +52030,11 @@
 $as_echo_n "checking for mmap() using shm_open() shared memory support... " >&6; }
   if test "$cross_compiling" = yes; then :
 
-    { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
+$as_echo "#define HAVE_SHM_MMAP_POSIX 1" >>confdefs.h
+
+    { $as_echo "$as_me:${as_lineno-$LINENO}: cross-compiling, assuming yes" >&5
+$as_echo "cross-compiling, assuming yes" >&6; }
+    have_shm_mmap_posix=yes
 
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
@@ -81688,7 +81691,7 @@
 enable_static=yes

 case $php_sapi_module in
-  shared)
+  *)
     if test "$PHP_CGI" = "no" && test "$PHP_CLI" = "no" && test "$PHP_FPM" = "no" && test "$PHP_LITESPEED" = "no" && test "$PHP_PHPDBG" = "no"; then
       enable_static=no
     fi
@@ -81702,12 +81705,6 @@
     esac
     EXTRA_LDFLAGS="$EXTRA_LDFLAGS -avoid-version -module"
     ;;
-  *)
-    standard_libtool_flag='-prefer-non-pic -static'
-    if test -z "$PHP_MODULES" && test -z "$PHP_ZEND_EX"; then
-      enable_shared=no
-    fi
-    ;;
 esac

 EXTRA_LIBS="$EXTRA_LIBS $DLIBS $LIBS"
@@ -85405,7 +85408,7 @@
       lt_prog_compiler_static='-Bstatic'
       ;;

-    linux* | k*bsd*-gnu)
+    linux* | k*bsd*-gnu | midipix*)
       case $cc_basename in
       # old Intel for x86_64 which still supported -KPIC.
       ecc*)
@@ -87098,6 +87101,17 @@
   dynamic_linker='GNU/Linux ld.so'
   ;;
 
+midipix*)
+  version_type=linux # correct to gnu/linux during the next big refactor
+  need_lib_prefix=no
+  need_version=no
+  library_names_spec='$libname$release$shared_ext$versuffix $libname$release$shared_ext$major $libname$shared_ext'
+  soname_spec='$libname$release$shared_ext$major'
+  finish_cmds='PATH="\$PATH:/sbin" ldconfig -n $libdir'
+  shlibpath_var=LD_LIBRARY_PATH
+  dynamic_linker='GNU/Linux ld.so'
+  ;;
+
 netbsd*)
   version_type=sunos
   need_lib_prefix=no
@@ -89172,7 +89185,7 @@
     hardcode_libdir_flag_spec_CXX='${wl}-rpath ${wl}$libdir'
     hardcode_libdir_separator_CXX=:
     ;;
-  linux* | k*bsd*-gnu)
+  linux* | k*bsd*-gnu | midipix*)
     case $cc_basename in
       KCC*)
        # Kuck and Associates, Inc. (KAI) C++ Compiler
@@ -91441,37 +91448,6 @@
 # This can be used to rebuild libtool when needed
 LIBTOOL_DEPS="$ac_aux_dir/ltmain.sh"

-# Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
-
-# Prevent multiple expansion
-
-
-
-
-
-
-
-
-
-
-
-
-  if test -z "$LIBTOOL"; then
-    LIBTOOL='$(SHELL) $(top_builddir)/libtool --silent'
-  else
-    LIBTOOL="$LIBTOOL --silent"
-  fi
-
-
-
-  if test -z "$LIBTOOL"; then
-    LIBTOOL='$(SHELL) $(top_builddir)/libtool --preserve-dup-deps'
-  else
-    LIBTOOL="$LIBTOOL --preserve-dup-deps"
-  fi
-
-
 test -z "$PHP_COMPILE" && PHP_COMPILE='$(LIBTOOL) --mode=compile $(COMPILE) -c $<'
 test -z "$CXX_PHP_COMPILE" && CXX_PHP_COMPILE='$(LIBTOOL) --mode=compile $(CXX_COMPILE) -c $<'
 SHARED_LIBTOOL='$(LIBTOOL)'
diff -ru php-8.0.1.orig/sapi/cgi/Makefile.frag php-8.0.1/sapi/cgi/Makefile.frag
--- php-8.0.1.orig/sapi/cgi/Makefile.frag	2021-01-05 14:54:54.000000000 +0100
+++ php-8.0.1/sapi/cgi/Makefile.frag	2021-01-14 00:20:44.769574312 +0100
@@ -6,7 +6,7 @@
 install-cgi: $(SAPI_CGI_PATH)
 	@echo "Installing PHP CGI binary:        $(INSTALL_ROOT)$(bindir)/"
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(bindir)
-	@$(INSTALL) -m 0755 $(SAPI_CGI_PATH) $(INSTALL_ROOT)$(bindir)/$(program_prefix)php-cgi$(program_suffix)$(EXEEXT)
+	@$(LIBTOOL) --mode=install cp $(SAPI_CGI_PATH) $(INSTALL_ROOT)$(bindir)/$(program_prefix)php-cgi$(program_suffix)$(EXEEXT)
 	@echo "Installing PHP CGI man page:      $(INSTALL_ROOT)$(mandir)/man1/"
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(mandir)/man1
 	@$(INSTALL_DATA) sapi/cgi/php-cgi.1 $(INSTALL_ROOT)$(mandir)/man1/$(program_prefix)php-cgi$(program_suffix).1
diff -ru php-8.0.1.orig/sapi/cli/Makefile.frag php-8.0.1/sapi/cli/Makefile.frag
--- php-8.0.1.orig/sapi/cli/Makefile.frag	2021-01-05 14:54:54.000000000 +0100
+++ php-8.0.1/sapi/cli/Makefile.frag	2021-01-14 00:48:00.415700667 +0100
@@ -6,7 +6,7 @@
 install-cli: $(SAPI_CLI_PATH)
 	@echo "Installing PHP CLI binary:        $(INSTALL_ROOT)$(bindir)/"
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(bindir)
-	@$(INSTALL) -m 0755 $(SAPI_CLI_PATH) $(INSTALL_ROOT)$(bindir)/$(program_prefix)php$(program_suffix)$(EXEEXT)
+	@$(LIBTOOL) --mode=install cp $(SAPI_CLI_PATH) $(INSTALL_ROOT)$(bindir)/$(program_prefix)php$(program_suffix)$(EXEEXT)
 	@echo "Installing PHP CLI man page:      $(INSTALL_ROOT)$(mandir)/man1/"
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(mandir)/man1
 	@$(INSTALL_DATA) sapi/cli/php.1 $(INSTALL_ROOT)$(mandir)/man1/$(program_prefix)php$(program_suffix).1
diff -ru php-8.0.1.orig/sapi/phpdbg/Makefile.frag php-8.0.1/sapi/phpdbg/Makefile.frag
--- php-8.0.1.orig/sapi/phpdbg/Makefile.frag	2021-01-05 14:54:54.000000000 +0100
+++ php-8.0.1/sapi/phpdbg/Makefile.frag	2021-01-14 00:35:57.023625462 +0100
@@ -25,7 +25,7 @@
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(bindir)
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(localstatedir)/log
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(localstatedir)/run
-	@$(INSTALL) -m 0755 $(BUILD_BINARY) $(INSTALL_ROOT)$(bindir)/$(program_prefix)phpdbg$(program_suffix)$(EXEEXT)
+	@$(LIBTOOL) --mode=install cp $(BUILD_BINARY) $(INSTALL_ROOT)$(bindir)/$(program_prefix)phpdbg$(program_suffix)$(EXEEXT)
 	@echo "Installing phpdbg man page:       $(INSTALL_ROOT)$(mandir)/man1/"
 	@$(mkinstalldirs) $(INSTALL_ROOT)$(mandir)/man1
 	@$(INSTALL_DATA) sapi/phpdbg/phpdbg.1 $(INSTALL_ROOT)$(mandir)/man1/$(program_prefix)phpdbg$(program_suffix).1
