--- pkg-cron.git/Makefile.orig	2016-07-11 19:06:33.212362869 +0000
+++ pkg-cron.git/Makefile	2016-09-11 20:37:42.714987721 +0000
@@ -79,7 +79,7 @@
 #DEFS		=	-D_BSD_SIGNALS -Dconst=
 #<<the name of the BSD-like install program>>
 #INSTALL = installbsd
-INSTALL = install -s
+INSTALL = install $(INSTALLFLAGS)
 #<<any special load flags>>
 # LDFLAGS		=	-s
 # Let install do the strip
@@ -118,8 +118,8 @@
 			$(CC) $(LDFLAGS) -o crontab $(CRONTAB_OBJ) $(LIBS)
 
 install		:	all
-			$(INSTALL) -c -m  755 -o root cron    $(DESTSBIN)/
-			$(INSTALL) -c -m 4755 -o root crontab $(DESTBIN)/
+			$(INSTALL) -c -m  755 cron    $(DESTSBIN)/
+			$(INSTALL) -c -m 4755 crontab $(DESTBIN)/
 			sh putman.sh crontab.1 $(DESTMAN)
 			sh putman.sh cron.8    $(DESTMAN)
 			sh putman.sh crontab.5 $(DESTMAN)
