diff -Nru posix_cc-1.4.orig/config.h.in posix_cc-1.4/config.h.in
--- posix_cc-1.4.orig/config.h.in	2001-12-12 13:45:12.000000000 +0100
+++ posix_cc-1.4/config.h.in	2024-04-09 09:22:44.846399546 +0200
@@ -1,46 +1,63 @@
-/* config.h.in.  Generated automatically from configure.in by autoheader.  */
+/* config.h.in.  Generated from configure.ac by autoheader.  */
 
-/* Define if you have the <inttypes.h> header file. */
+/* Define to 1 if you have the <inttypes.h> header file. */
 #undef HAVE_INTTYPES_H
 
-/* Define if your system has a working `malloc' function. */
-#undef HAVE_MALLOC
-
-/* Define if you have the <memory.h> header file. */
-#undef HAVE_MEMORY_H
-
-/* Define if you have the <stdint.h> header file. */
+/* Define to 1 if you have the <stdint.h> header file. */
 #undef HAVE_STDINT_H
 
-/* Define if you have the <stdlib.h> header file. */
+/* Define to 1 if you have the <stdio.h> header file. */
+#undef HAVE_STDIO_H
+
+/* Define to 1 if you have the <stdlib.h> header file. */
 #undef HAVE_STDLIB_H
 
-/* Define if you have the <strings.h> header file. */
+/* Define to 1 if you have the <strings.h> header file. */
 #undef HAVE_STRINGS_H
 
-/* Define if you have the <string.h> header file. */
+/* Define to 1 if you have the <string.h> header file. */
 #undef HAVE_STRING_H
 
-/* Define if you have the <sys/stat.h> header file. */
+/* Define to 1 if you have the <sys/stat.h> header file. */
 #undef HAVE_SYS_STAT_H
 
-/* Define if you have the <sys/types.h> header file. */
+/* Define to 1 if you have the <sys/types.h> header file. */
 #undef HAVE_SYS_TYPES_H
 
-/* Define if you have the <unistd.h> header file. */
+/* Define to 1 if you have the <unistd.h> header file. */
 #undef HAVE_UNISTD_H
 
 /* Name of package */
 #undef PACKAGE
 
-/* Define if you have the ANSI C header files. */
+/* Define to the address where bug reports for this package should be sent. */
+#undef PACKAGE_BUGREPORT
+
+/* Define to the full name of this package. */
+#undef PACKAGE_NAME
+
+/* Define to the full name and version of this package. */
+#undef PACKAGE_STRING
+
+/* Define to the one symbol short name of this package. */
+#undef PACKAGE_TARNAME
+
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
+/* Define to the version of this package. */
+#undef PACKAGE_VERSION
+
+/* Define to 1 if all of the C89 standard headers exist (not just the ones
+   required in a freestanding environment). This macro is provided for
+   backward compatibility; new code need not use it. */
 #undef STDC_HEADERS
 
 /* Version number of package */
 #undef VERSION
 
-/* Define to empty if `const' does not conform to ANSI C. */
+/* Define to empty if 'const' does not conform to ANSI C. */
 #undef const
 
-/* Define to `unsigned' if <sys/types.h> does not define. */
+/* Define as 'unsigned int' if <stddef.h> doesn't define. */
 #undef size_t
diff -Nru posix_cc-1.4.orig/configure.ac posix_cc-1.4/configure.ac
--- posix_cc-1.4.orig/configure.ac	1970-01-01 01:00:00.000000000 +0100
+++ posix_cc-1.4/configure.ac	2024-04-09 09:20:54.446401106 +0200
@@ -0,0 +1,15 @@
+# Process this file with autoconf to produce a configure script.
+AC_INIT(posix_cc, 1.4)
+AC_CONFIG_SRCDIR([c89.c])
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE
+
+# Checks for programs.
+AC_PROG_CC
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_C_CONST
+AC_TYPE_SIZE_T
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
diff -Nru posix_cc-1.4.orig/configure.in posix_cc-1.4/configure.in
--- posix_cc-1.4.orig/configure.in	2001-12-12 15:53:13.000000000 +0100
+++ posix_cc-1.4/configure.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,24 +0,0 @@
-# Process this file with autoconf to produce a configure script.
-AC_INIT(posix_cc, 1.4)
-AM_INIT_AUTOMAKE(posix_c, 1.4)
-AC_CONFIG_SRCDIR([c89.c])
-AM_CONFIG_HEADER([config.h])
-
-# Checks for programs.
-AC_PROG_CC
-
-# Checks for libraries.
-
-# Checks for header files.
-AC_HEADER_STDC
-AC_CHECK_HEADERS([stdlib.h string.h unistd.h])
-
-# Checks for typedefs, structures, and compiler characteristics.
-AC_C_CONST
-AC_TYPE_SIZE_T
-
-# Checks for library functions.
-AC_FUNC_MALLOC
-
-AC_CONFIG_FILES([Makefile])
-AC_OUTPUT
