diff -ru proxytunnel-1.10.20200507.orig/Makefile proxytunnel-1.10.20200507/Makefile
--- proxytunnel-1.10.20200507.orig/Makefile	2020-05-07 17:17:41.000000000 +0200
+++ proxytunnel-1.10.20200507/Makefile	2020-05-22 12:14:53.409128241 +0200
@@ -90,7 +90,6 @@
 install:
 	install -d $(DESTDIR)$(bindir)
 	install -p -m555 $(name) $(DESTDIR)$(bindir)
-	$(MAKE) -C docs install
 
 .c.o:
 	$(CC) $(CFLAGS) $(OPTFLAGS) -c -o $@ $<
