--- build/sbpython2/project/config/cfgdefs.sh.orig	2022-12-26 19:52:02.067193887 +0100
+++ build/sbpython2/project/config/cfgdefs.sh	2023-03-20 20:11:46.912982474 +0100
@@ -70,7 +70,7 @@
 		exit 2
 	fi
 
-	"$mb_native_cc" -E -dM "$mb_source_dir/Include/patchlevel.h" \
+	$mb_native_cc -E -dM "$mb_source_dir/Include/patchlevel.h" \
 		> "$mb_internal_verinfo"
 
 	python_major=$(grep '#define PY_MAJOR_VERSION ' "$mb_internal_verinfo" | cut -d' ' -f3)
