--- ./build/sbpython2/project/config/cfgdefs.sh.orig	2021-04-06 10:42:07.823432394 +0100
+++ ./build/sbpython2/project/config/cfgdefs.sh	2021-04-06 10:46:44.345427809 +0100
@@ -106,6 +106,7 @@
 	if [ -n "$pycfg_pymalloc" ]; then
 		cfgtest_cflags_append '-DWITH_PYMALLOC'
 	fi
+cfgtest_ldflags_append "-L$mb_prefix/lib"
 }
 
 
@@ -306,8 +307,8 @@
 	fi
 
 	# ncurses libs (common part)
-	mb_ncurses_libs='-lpanelw -lncursesw'
-	mb_ncurses_tinfo='-ltinfo'
+	mb_ncurses_libs='-lpanelw -lncursesw -ltinfow'
+	mb_ncurses_tinfo='-ltinfow'
 
 	if cfgtest_library_presence $mb_ncurses_tinfo; then
 		mb_ncurses_libs="$mb_ncurses_libs $mb_ncurses_tinfo"
--- ./build/sbpython2/project/pyexts.mk.orig	2021-04-06 10:42:07.827432250 +0100
+++ ./build/sbpython2/project/pyexts.mk	2021-04-06 10:49:55.946499708 +0100
@@ -1017,7 +1017,7 @@
 PYEXT_OSSAUDIODEV_LOBJS		= $(PYEXT_OSSAUDIODEV_SRCS:.c=.lo)
 PYEXT_OSSAUDIODEV_OBJS		= $(PYEXT_OSSAUDIODEV_SRCS:.c=.o)
 
-PYEXT_OSSAUDIODEV_SHARED	= pyext/ossaudiodev$(OS_LIB_SUFFIX)
+#PYEXT_OSSAUDIODEV_SHARED	= pyext/ossaudiodev$(OS_LIB_SUFFIX)
 PYEXT_OSSAUDIODEV_STATIC	= pyext/ossaudiodev$(OS_ARCHIVE_EXT)
 
 $(PYEXT_OSSAUDIODEV_SHARED):	$(PYEXT_OSSAUDIODEV_LOBJS)
