diff -ru rdesktop-1.8.3.orig/rdpsnd_alsa.c rdesktop-1.8.3/rdpsnd_alsa.c
--- rdesktop-1.8.3.orig/rdpsnd_alsa.c	2015-08-21 08:24:23.000000000 +0200
+++ rdesktop-1.8.3/rdpsnd_alsa.c	2018-11-01 20:53:55.216422142 +0100
@@ -218,7 +218,7 @@
 #endif
 
 	rate = pwfx->nSamplesPerSec;
-	if ((err = snd_pcm_hw_params_set_rate_near(pcm, hwparams, &rate, 0)) < 0)
+	if ((err = __snd_pcm_hw_params_set_rate_near(pcm, hwparams, &rate, 0)) < 0)
 	{
 		error("snd_pcm_hw_params_set_rate_near: %s\n", snd_strerror(err));
 		return False;
@@ -233,7 +233,7 @@
 
 
 	buffertime = 500000;	/* microseconds */
-	if ((err = snd_pcm_hw_params_set_buffer_time_near(pcm, hwparams, &buffertime, 0)) < 0)
+	if ((err = __snd_pcm_hw_params_set_buffer_time_near(pcm, hwparams, &buffertime, 0)) < 0)
 	{
 		error("snd_pcm_hw_params_set_buffer_time_near: %s\n", snd_strerror(err));
 		return False;
