diff -ru screen-4.6.2-854c367.orig/Makefile.in screen-4.6.2-854c367/Makefile.in
--- screen-4.6.2-854c367.orig/Makefile.in	2019-07-05 11:55:51.818882542 +0200
+++ screen-4.6.2-854c367/Makefile.in	2019-07-05 12:17:53.551869845 +0200
@@ -32,7 +32,7 @@
 SCREENENCODINGS = $(datadir)/screen/utf8encodings
 
 CC = @CC@
-CFLAGS = @CFLAGS@ -Wall -Wextra -std=c11
+CFLAGS = @CFLAGS@ -Wall -Wextra -std=c99
 CPPFLAGS = @CPPFLAGS@ -iquote. -DSCREENENCODINGS='"$(SCREENENCODINGS)"' -DBUILD_DATE='"$(BUILD_DATE)"'
 LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
@@ -80,7 +80,7 @@
 	-if [ -f $(DESTDIR)$(bindir)/$(SCREEN) ] && [ ! -f $(DESTDIR)$(bindir)/$(SCREEN).old ]; \
 		then mv $(DESTDIR)$(bindir)/$(SCREEN) $(DESTDIR)$(bindir)/$(SCREEN).old; fi
 	$(INSTALL_PROGRAM) screen $(DESTDIR)$(bindir)/$(SCREEN)
-	-chown root $(DESTDIR)$(bindir)/$(SCREEN) && chmod 4755 $(DESTDIR)$(bindir)/$(SCREEN)
+	-chmod 4755 $(DESTDIR)$(bindir)/$(SCREEN)
 # This doesn't work if $(bindir)/screen is a symlink
 	-if [ -f $(DESTDIR)$(bindir)/screen ] && [ ! -f $(DESTDIR)$(bindir)/screen.old ]; then mv $(DESTDIR)$(bindir)/screen $(DESTDIR)$(bindir)/screen.old; fi
 	rm -f $(DESTDIR)$(bindir)/screen
