--- SmallBASIC/configure.orig	2017-11-20 01:35:25.276000000 +0000
+++ SmallBASIC/configure	2017-11-20 01:37:03.332000000 +0000
@@ -5171,7 +5171,7 @@
 
 $as_echo "#define USE_PCRE 1" >>confdefs.h
 
-     PACKAGE_LIBS="${PACKAGE_LIBS} `pcre-config --libs`"
+     PACKAGE_LIBS="${PACKAGE_LIBS} ${PCRE_LIBS}"
    fi
 }
 
--- SmallBASIC/src/platform/console/main.c.orig	2017-12-19 08:54:21.692000000 +0000
+++ SmallBASIC/src/platform/console/main.c	2017-12-19 13:53:21.540000000 +0000
@@ -253,7 +253,7 @@
   return 1;
 }
 
-#if defined(__GNUC__)
+#if defined(__GLIBC__)
 // for analysing excessive malloc calls using kdbg
 extern void *__libc_malloc(size_t size);
 void* malloc (size_t size) {
