diff -ru a/unix/configure b/unix/configure
--- unix/configure	2017-09-07 08:54:11.000000000 -0400
+++ unix/configure	2019-08-20 12:55:27.989692823 -0400
@@ -3957,7 +3957,7 @@
 
 $as_echo "#define _REENTRANT 1" >>confdefs.h
 
-	if test "`uname -s`" = "SunOS" ; then
+	if test "${target}" = "SunOS" ; then
 
 $as_echo "#define _POSIX_PTHREAD_SEMANTICS 1" >>confdefs.h
 
@@ -4893,13 +4893,13 @@
 	if test "${TEA_PLATFORM}" = "windows" ; then
 	    tcl_cv_sys_version=windows
 	else
-	    tcl_cv_sys_version=`uname -s`-`uname -r`
+	    tcl_cv_sys_version=${target}-`uname -r`
 	    if test "$?" -ne 0 ; then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: can't find uname command" >&5
 $as_echo "$as_me: WARNING: can't find uname command" >&2;}
 		tcl_cv_sys_version=unknown
 	    else
-		if test "`uname -s`" = "AIX" ; then
+		if test "${target}" = "AIX" ; then
 		    tcl_cv_sys_version=AIX-`uname -v`.`uname -r`
 		fi
 	    fi
@@ -5718,6 +5718,29 @@
 		LD_SEARCH_FLAGS='-Wl,-rpath,${LIB_RUNTIME_DIR}'
 fi
 	    ;;
+	*-midipix-*)
+	    SHLIB_CFLAGS="-fPIC"
+	    SHLIB_SUFFIX=".so"
+
+	    CFLAGS_OPTIMIZE="-O2"
+
+	    SHLIB_LD='${CC} -shared ${CFLAGS} ${LDFLAGS}'
+	    DL_OBJS="tclLoadDl.o"
+	    DL_LIBS="-ldl"
+	    LD_SEARCH_FLAGS=${CC_SEARCH_FLAGS}
+	    if test $do64bit = yes; then :
+
+		{ $as_echo "$as_me:${as_lineno-$LINENO}: checking if compiler accepts -m64 flag" >&5
+$as_echo_n "checking if compiler accepts -m64 flag... " >&6; }
+if ${tcl_cv_cc_m64+:} false; then :
+  $as_echo_n "(cached) " >&6
+else
+  tcl_cv_cc_m64=no
+fi
+{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $tcl_cv_cc_m64" >&5
+$as_echo "$tcl_cv_cc_m64" >&6; }
+fi
+	    ;;
 	OpenBSD-*)
 	    arch=`arch -s`
 	    case "$arch" in
@@ -7429,7 +7452,7 @@
 fi
 
 
-if test "`uname -s`" = "Darwin" && test "${TCL_THREADS}" = 1 && \
+if test "${target}" = "Darwin" && test "${TCL_THREADS}" = 1 && \
 	test "`uname -r | awk -F. '{print $1}'`" -lt 7; then
     # prior to Darwin 7, realpath is not threadsafe, so don't
     # use it when threads are enabled, c.f. bug # 711232
@@ -7937,7 +7960,7 @@
 
 fi
 
-    if test "`uname -s`" = "Darwin" && \
+    if test "${target}" = "Darwin" && \
 	    test "`uname -r | awk -F. '{print $1}'`" -gt 5; then
 	# Starting with Darwin 6 (Mac OSX 10.2), gethostbyX
 	# are actually MT-safe as they always return pointers
@@ -7949,7 +7972,7 @@
 $as_echo "#define HAVE_MTSAFE_GETHOSTBYADDR 1" >>confdefs.h
 
 
-    elif test "`uname -s`" = "HP-UX" && \
+    elif test "${target}" = "HP-UX" && \
 	      test "`uname -r|sed -e 's|B\.||' -e 's|\..*$||'`" -gt 10; then
         # Starting with HPUX 11.00 (we believe), gethostbyX
         # are actually MT-safe as they always return pointers
@@ -8330,7 +8353,7 @@
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for advanced notifier support" >&5
 $as_echo_n "checking for advanced notifier support... " >&6; }
-case x`uname -s` in
+case x${target} in
   xLinux)
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: epoll(7)" >&5
 $as_echo "epoll(7)" >&6; }
@@ -9676,7 +9699,7 @@
 # Darwin specific API checks and defines
 #--------------------------------------------------------------------
 
-if test "`uname -s`" = "Darwin" ; then
+if test "${target}" = "Darwin" ; then
     for ac_func in getattrlist
 do :
   ac_fn_c_check_func "$LINENO" "getattrlist" "ac_cv_func_getattrlist"
@@ -9942,13 +9965,13 @@
 	if test "${TEA_PLATFORM}" = "windows" ; then
 	    tcl_cv_sys_version=windows
 	else
-	    tcl_cv_sys_version=`uname -s`-`uname -r`
+	    tcl_cv_sys_version=${target}-`uname -r`
 	    if test "$?" -ne 0 ; then
 		{ $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: can't find uname command" >&5
 $as_echo "$as_me: WARNING: can't find uname command" >&2;}
 		tcl_cv_sys_version=unknown
 	    else
-		if test "`uname -s`" = "AIX" ; then
+		if test "${target}" = "AIX" ; then
 		    tcl_cv_sys_version=AIX-`uname -v`.`uname -r`
 		fi
 	    fi
@@ -10133,9 +10156,9 @@
 
     DTRACE_SRC="\${DTRACE_SRC}"
     DTRACE_HDR="\${DTRACE_HDR}"
-    if test "`uname -s`" != "Darwin" ; then
+    if test "${target}" != "Darwin" ; then
 	DTRACE_OBJ="\${DTRACE_OBJ}"
-	if test "`uname -s`" = "SunOS" -a "$SHARED_BUILD" = "0" ; then
+	if test "${target}" = "SunOS" -a "$SHARED_BUILD" = "0" ; then
 	    # Need to create an intermediate object file to ensure tclDTrace.o
 	    # gets included when linking against the static tcl library.
 	    STLIB_LD='stlib_ld () { /usr/ccs/bin/ld -r -o $${1%.a}.o "$${@:2}" && '"${STLIB_LD}"' $${1} $${1%.a}.o ; } && stlib_ld'
@@ -10219,9 +10242,9 @@
 # AIX remembers this path and will attempt to use it at run-time to look
 # up the Tcl library.
 
-if test "`uname -s`" = "Darwin" ; then
+if test "${target}" = "Darwin" ; then
 
-    if test "`uname -s`" = "Darwin" ; then
+    if test "${target}" = "Darwin" ; then
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking how to package libraries" >&5
 $as_echo_n "checking how to package libraries... " >&6; }
 	# Check whether --enable-framework was given.
@@ -10319,7 +10342,7 @@
         TCL_LIB_FLAG="-ltcl`echo ${TCL_VERSION} | tr -d .`"
     fi
     TCL_BUILD_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_LIB_FLAG}"
-    TCL_LIB_SPEC="-L${libdir} ${TCL_LIB_FLAG}"
+    TCL_LIB_SPEC="${TCL_LIB_FLAG}"
 fi
 VERSION='${VERSION}'
 eval "CFG_TCL_SHARED_LIB_SUFFIX=${TCL_SHARED_LIB_SUFFIX}"
@@ -10363,12 +10386,12 @@
 fi
 
 TCL_BUILD_STUB_LIB_SPEC="-L`pwd | sed -e 's/ /\\\\ /g'` ${TCL_STUB_LIB_FLAG}"
-TCL_STUB_LIB_SPEC="-L${TCL_STUB_LIB_DIR} ${TCL_STUB_LIB_FLAG}"
+TCL_STUB_LIB_SPEC="${TCL_STUB_LIB_FLAG}"
 TCL_BUILD_STUB_LIB_PATH="`pwd`/${TCL_STUB_LIB_FILE}"
 TCL_STUB_LIB_PATH="${TCL_STUB_LIB_DIR}/${TCL_STUB_LIB_FILE}"
 
 # Install time header dir can be set via --includedir
-eval "TCL_INCLUDE_SPEC=\"-I${includedir}\""
+#eval "TCL_INCLUDE_SPEC=\"-I${includedir}\""
 
 #------------------------------------------------------------------------
 # tclConfig.sh refers to this by a different name
