diff -ru the_silver_searcher-2.1.0.orig/src/main.c the_silver_searcher-2.1.0/src/main.c
--- the_silver_searcher-2.1.0.orig/src/main.c	2017-06-27 05:33:25.000000000 +0200
+++ the_silver_searcher-2.1.0/src/main.c	2017-11-18 21:00:58.908603541 +0100
@@ -154,7 +154,7 @@
             }
 #if defined(HAVE_PTHREAD_SETAFFINITY_NP) && (defined(USE_CPU_SET) || defined(HAVE_SYS_CPUSET_H))
             if (opts.use_thread_affinity) {
-#ifdef __linux__
+#if defined(__linux__) || defined(__midipix__)
                 cpu_set_t cpu_set;
 #elif __FreeBSD__
                 cpuset_t cpu_set;
