diff -ru thttpd-2.29.orig/extras/Makefile.in thttpd-2.29/extras/Makefile.in
--- thttpd-2.29.orig/extras/Makefile.in	2005-05-20 21:10:25.000000000 +0200
+++ thttpd-2.29/extras/Makefile.in	2020-05-23 22:44:07.959407057 +0200
@@ -68,7 +68,6 @@
 install:	all
 	rm -f $(BINDIR)/makeweb $(BINDIR)/htpasswd $(BINDIR)/syslogtocern
 	cp makeweb $(BINDIR)/makeweb
-	chgrp $(WEBGROUP) $(BINDIR)/makeweb
 	chmod 2755 $(BINDIR)/makeweb
 	cp htpasswd $(BINDIR)/htpasswd
 	cp syslogtocern $(BINDIR)/syslogtocern
diff -ru thttpd-2.29.orig/Makefile.in thttpd-2.29/Makefile.in
--- thttpd-2.29.orig/Makefile.in	2005-05-20 21:10:24.000000000 +0200
+++ thttpd-2.29/Makefile.in	2020-05-23 22:45:10.811407438 +0200
@@ -110,11 +110,11 @@
 
 installthis:
 	-mkdir -p $(DESTDIR)$(BINDIR)
-	$(INSTALL) -m 555 -o bin -g bin thttpd $(DESTDIR)$(BINDIR)
+	$(INSTALL) -m 755 thttpd $(DESTDIR)$(BINDIR)
 
 install-man:
 	-mkdir -p $(DESTDIR)$(MANDIR)/man8
-	$(INSTALL) -m 444 -o bin -g bin thttpd.8 $(DESTDIR)$(MANDIR)/man8
+	$(INSTALL) -m 444 thttpd.8 $(DESTDIR)$(MANDIR)/man8
 
 installsubdirs:
 	for i in $(SUBDIRS) ; do ( \
