diff -ru tk8.7a1.orig/unix/configure tk8.7a1/unix/configure
--- tk8.7a1.orig/unix/configure	2023-03-30 14:25:29.062891597 +0200
+++ tk8.7a1/unix/configure	2023-03-30 14:25:08.830891475 +0200
@@ -7752,8 +7752,8 @@
   *)
     # Record where we found X for the cache.
     ac_cv_have_x="have_x=yes\
-	ac_x_includes='$ac_x_includes'\
-	ac_x_libraries='$ac_x_libraries'"
+	ac_x_includes=''\
+	ac_x_libraries=''"
 esac
 fi
 ;; #(
@@ -8054,13 +8054,20 @@
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $enable_xft" >&5
 $as_echo "$enable_xft" >&6; }
     else
+    if [ $cross_compiling = yes ]; then
+        XFT_CONFIG=${host}-xft-config
+        PKG_CONFIG=${host}-pkg-config
+    else
+        XFT_CONFIG=xft-config
+        PKG_CONFIG=pkg-config
+   fi
 	found_xft="yes"
-			XFT_CFLAGS=`xft-config --cflags 2>/dev/null` || found_xft="no"
-	XFT_LIBS=`xft-config --libs 2>/dev/null` || found_xft="no"
+	XFT_CFLAGS=`"${PKG_CONFIG}" --cflags 2>/dev/null` || found_xft="no"
+	XFT_LIBS=`"${PKG_CONFIG}" --libs 2>/dev/null` || found_xft="no"
 	if test "$found_xft" = "no" ; then
 	    found_xft=yes
-	    XFT_CFLAGS=`pkg-config --cflags xft 2>/dev/null` || found_xft="no"
-	    XFT_LIBS=`pkg-config --libs xft 2>/dev/null` || found_xft="no"
+	    XFT_CFLAGS=`"${PKG_CONFIG}" --cflags xft 2>/dev/null` || found_xft="no"
+	    XFT_LIBS=`"${PKG_CONFIG}" --libs xft 2>/dev/null` || found_xft="no"
 	fi
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $found_xft" >&5
 $as_echo "$found_xft" >&6; }
