--- tmux-2.3/configure.orig	2016-09-29 22:55:41.000000000 +0200
+++ tmux-2.3/configure	2016-12-18 02:04:56.608924500 +0100
@@ -3676,18 +3676,6 @@
 $as_echo "$ac_cv_prog_cc_g" >&6; }
 if test "$ac_test_CFLAGS" = set; then
   CFLAGS=$ac_save_CFLAGS
-elif test $ac_cv_prog_cc_g = yes; then
-  if test "$GCC" = yes; then
-    CFLAGS="-g -O2"
-  else
-    CFLAGS="-g"
-  fi
-else
-  if test "$GCC" = yes; then
-    CFLAGS="-O2"
-  else
-    CFLAGS=
-  fi
 fi
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $CC option to accept ISO C89" >&5
 $as_echo_n "checking for $CC option to accept ISO C89... " >&6; }
--- tmux-2.3/Makefile.in.orig	2016-09-29 22:55:41.000000000 +0200
+++ tmux-2.3/Makefile.in	2016-12-18 02:05:27.700311700 +0100
@@ -94,10 +94,10 @@
 
 # glibc as usual does things ass-backwards and hides useful things by default,
 # so everyone has to add this.
-@IS_GLIBC_TRUE@am__append_1 = -D_GNU_SOURCE
+am__append_1 = -D_GNU_SOURCE
 
 # Set flags for gcc.
-@IS_GCC_TRUE@am__append_2 = -std=gnu99 -O2
+@IS_GCC_TRUE@am__append_2 = -std=gnu99
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@am__append_3 = -g -Wno-long-long -Wall -W \
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Wformat=2 -Wmissing-prototypes \
 @IS_DEBUG_TRUE@@IS_GCC_TRUE@	-Wstrict-prototypes \
--- tmux-2.3/tmux.c.orig	2016-06-05 00:30:48.000000000 +0200
+++ tmux-2.3/tmux.c	2016-12-18 02:05:42.524589500 +0100
@@ -129,10 +129,12 @@
 		errno = ENOTDIR;
 		goto fail;
 	}
+#if 0
 	if (sb.st_uid != uid || (sb.st_mode & S_IRWXO) != 0) {
 		errno = EACCES;
 		goto fail;
 	}
+#endif
 
 	if (realpath(base, resolved) == NULL)
 		strlcpy(resolved, base, sizeof resolved);
