diff -ru toilet.orig/Makefile.in toilet/Makefile.in
--- toilet.orig/Makefile.in	2017-09-03 18:16:42.175886270 +0200
+++ toilet/Makefile.in	2017-09-03 18:29:29.600903132 +0200
@@ -310,7 +310,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src tools fonts doc test
+SUBDIRS = src tools fonts test
 DIST_SUBDIRS = $(SUBDIRS)
 EXTRA_DIST = bootstrap
 CACA2TLF = $(top_builddir)/tools/caca2tlf
