diff -ru toilet.orig/configure toilet/configure
--- toilet.orig/configure	2017-09-03 18:16:41.831887603 +0200
+++ toilet/configure	2017-09-03 18:22:18.302580756 +0200
@@ -4412,81 +4412,6 @@
 fi
 
 
-pkg_failed=no
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for CACA" >&5
-$as_echo_n "checking for CACA... " >&6; }
-
-if test -n "$CACA_CFLAGS"; then
-    pkg_cv_CACA_CFLAGS="$CACA_CFLAGS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"caca >= 0.99.beta18\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "caca >= 0.99.beta18") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_CACA_CFLAGS=`$PKG_CONFIG --cflags "caca >= 0.99.beta18" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-if test -n "$CACA_LIBS"; then
-    pkg_cv_CACA_LIBS="$CACA_LIBS"
- elif test -n "$PKG_CONFIG"; then
-    if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"caca >= 0.99.beta18\""; } >&5
-  ($PKG_CONFIG --exists --print-errors "caca >= 0.99.beta18") 2>&5
-  ac_status=$?
-  $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
-  test $ac_status = 0; }; then
-  pkg_cv_CACA_LIBS=`$PKG_CONFIG --libs "caca >= 0.99.beta18" 2>/dev/null`
-		      test "x$?" != "x0" && pkg_failed=yes
-else
-  pkg_failed=yes
-fi
- else
-    pkg_failed=untried
-fi
-
-
-
-if test $pkg_failed = yes; then
-   	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-
-if $PKG_CONFIG --atleast-pkgconfig-version 0.20; then
-        _pkg_short_errors_supported=yes
-else
-        _pkg_short_errors_supported=no
-fi
-        if test $_pkg_short_errors_supported = yes; then
-	        CACA_PKG_ERRORS=`$PKG_CONFIG --short-errors --print-errors --cflags --libs "caca >= 0.99.beta18" 2>&1`
-        else
-	        CACA_PKG_ERRORS=`$PKG_CONFIG --print-errors --cflags --libs "caca >= 0.99.beta18" 2>&1`
-        fi
-	# Put the nasty error message in config.log where it belongs
-	echo "$CACA_PKG_ERRORS" >&5
-
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-  as_fn_error $? "you need libcaca version 0.99.beta18 or later" "$LINENO" 5
-elif test $pkg_failed = untried; then
-     	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-  as_fn_error $? "you need libcaca version 0.99.beta18 or later" "$LINENO" 5
-else
-	CACA_CFLAGS=$pkg_cv_CACA_CFLAGS
-	CACA_LIBS=$pkg_cv_CACA_LIBS
-        { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-	:
-fi
-
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for release date" >&5
 $as_echo_n "checking for release date... " >&6; }
 tmp="${stamp%-*}"
diff -ru toilet.orig/Makefile.in toilet/Makefile.in
--- toilet.orig/Makefile.in	2017-09-03 18:16:42.175886270 +0200
+++ toilet/Makefile.in	2017-09-03 18:29:29.600903132 +0200
@@ -310,7 +310,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = src tools fonts doc test
+SUBDIRS = src tools fonts test
 DIST_SUBDIRS = $(SUBDIRS)
 EXTRA_DIST = bootstrap
 CACA2TLF = $(top_builddir)/tools/caca2tlf
