diff -ru tree-2.0.2.orig/Makefile tree-2.0.2/Makefile
--- tree-2.0.2.orig/Makefile	2022-02-16 22:45:43.000000000 +0100
+++ tree-2.0.2/Makefile	2022-02-17 22:49:24.340545322 +0100
@@ -26,15 +26,13 @@
 DESTDIR=${PREFIX}/bin
 MAN=tree.1
 # Probably needs to be ${PREFIX}/share/man for most systems now
-MANDIR=${PREFIX}/man
+MANDIR=${DESTDIR}${PREFIX}/share/man
 OBJS=tree.o list.o hash.o color.o file.o filter.o info.o unix.o xml.o json.o html.o strverscmp.o
 
 # Uncomment options below for your particular OS:
 
 # Linux defaults:
 #CFLAGS=-ggdb -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS=-O3 -pedantic -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-LDFLAGS=-s
 
 # Uncomment for FreeBSD:
 #CC=cc
@@ -105,7 +103,7 @@
 install: tree
 	$(INSTALL) -d $(DESTDIR)
 	$(INSTALL) -d $(MANDIR)/man1
-	$(INSTALL) $(TREE_DEST) $(DESTDIR)/$(TREE_DEST); \
+	$(INSTALL) $(TREE_DEST) $(DESTDIR)${PREFIX}/bin/$(TREE_DEST); \
 	$(INSTALL) -m 644 doc/$(MAN) $(MANDIR)/man1/$(MAN)
 
 distclean:
