diff -ru tree-2.0.3.orig/Makefile tree-2.0.3/Makefile
--- tree-2.0.3.orig/Makefile	2022-08-29 16:06:51.186529283 +0200
+++ tree-2.0.3/Makefile	2022-08-29 16:07:26.594528175 +0200
@@ -26,7 +26,7 @@
 DESTDIR=${PREFIX}/bin
 MAN=tree.1
 # Probably needs to be ${PREFIX}/share/man for most systems now
-MANDIR=${PREFIX}/man
+MANDIR=${DESTDIR}${PREFIX}/man
 OBJS=tree.o list.o hash.o color.o file.o filter.o info.o unix.o xml.o json.o html.o strverscmp.o
 
 # Uncomment options below for your particular OS:
