diff -ru weechat-3.8.orig/configure.ac weechat-3.8/configure.ac
--- weechat-3.8.orig/configure.ac	2023-01-08 10:07:07.000000000 +0100
+++ weechat-3.8/configure.ac	2024-02-17 14:04:48.583831305 +0100
@@ -184,18 +184,6 @@
 fi
 
 # ------------------------------------------------------------------------------
-#                                  pkg-config
-# ------------------------------------------------------------------------------
-
-PKGCONFIG=""
-AC_CHECK_PROGS(PKGCONFIG, pkg-config)
-if test "x$PKGCONFIG" = "x"; then
-    AC_MSG_ERROR([
-*** "pkg-config" couldn't be found on your system.
-*** Try to install it with your software package manager.])
-fi
-
-# ------------------------------------------------------------------------------
 #                                dynamic loader
 # ------------------------------------------------------------------------------
 
@@ -231,7 +219,7 @@
             NCURSES_LFLAGS="-lncurses"
         fi
     else
-        NCURSES_LFLAGS="-lncursesw"
+        NCURSES_LFLAGS="-lncursesw -ltinfow"
     fi
     AC_CHECK_HEADERS([ncurses.h ncursesw/curses.h])
     AC_SUBST(NCURSES_LFLAGS)
@@ -409,52 +397,6 @@
 
 # ---------------------------------- perl --------------------------------------
 
-PERL_VERSION=
-
-if test "x$enable_perl" = "xyes" ; then
-    AC_PATH_PROGS(PERL, perl perl5)
-    if test -z $PERL ; then
-        AC_MSG_WARN([
-*** Perl must be installed on your system but perl interpreter couldn't be found in path.
-*** Please check that perl is in path, or install it with your software package manager.
-*** WeeChat will be built without Perl support.])
-        enable_perl="no"
-        not_found="$not_found perl"
-    else
-        PERL_VERSION=`perl -V:version | sed "s/version='\(.*\)';/\1/"`
-        AC_MSG_CHECKING(for Perl headers files)
-
-        PERL_HEADER_TEST=`PT=perltest.c ; echo "#include <EXTERN.h>" > $PT;  echo "#include <perl.h>" >> $PT;  echo "#include <XSUB.h>" >> $PT ;  echo "int main() { return 0; }" >> $PT ; $CC -Wall $PT -o $PT.out $($PERL -MExtUtils::Embed -e ccopts -e ldopts) 1>/dev/null 2>&1; echo $?; rm -f $PT $PT.out 1>/dev/null 2>&1`
-
-        if test "x$PERL_HEADER_TEST" = "x0" ; then
-            PERL_CFLAGS=`$PERL -MExtUtils::Embed -e ccopts`
-            AC_MSG_RESULT(found)
-            AC_MSG_CHECKING(for Perl library)
-            PERL_LIB_TEST=`PT=perltest.c ; echo "int main() { return 0; }" > $PT ; $CC -Wall $PT -o $PT.out $($PERL -MExtUtils::Embed -e ldopts) 1>/dev/null 2>&1; echo $?; rm -f $PT $PT.out 1>/dev/null 2>&1`
-            if test "x$PERL_LIB_TEST" = "x0" ; then
-                PERL_LFLAGS=`$PERL -MExtUtils::Embed -e ldopts`
-                AC_MSG_RESULT(found)
-            else
-                AC_MSG_WARN([
-*** Perl library couldn't be found on your system.
-*** Try to install it with your software package manager.
-*** WeeChat will be built without Perl support.])
-                enable_perl="no"
-                not_found="$not_found perl"
-            fi
-        else
-            AC_MSG_WARN([
-*** Perl headers couldn't be found on your system.
-*** Try to install it with your software package manager.
-*** WeeChat will be built without Perl support.])
-            enable_perl="no"
-            not_found="$not_found perl"
-        fi
-    fi
-else
-    not_asked="$not_asked perl"
-fi
-
 if test "x$enable_perl" = "xyes" ; then
     AC_SUBST(PERL_CFLAGS)
     AC_SUBST(PERL_LFLAGS)
@@ -1053,8 +995,8 @@
 *** or try to install it with your software package manager.])
 else
     AC_MSG_RESULT(yes)
-    GCRYPT_CFLAGS=`libgcrypt-config --cflags`
-    GCRYPT_LFLAGS=`libgcrypt-config --libs`
+    GCRYPT_CFLAGS=`$LIBGCRYPT_CONFIG --cflags`
+    GCRYPT_LFLAGS=`$LIBGCRYPT_CONFIG --libs`
     AC_SUBST(GCRYPT_CFLAGS)
     AC_SUBST(GCRYPT_LFLAGS)
 fi
@@ -1074,8 +1016,8 @@
 *** or try to install it with your software package manager.])
 else
     AC_MSG_RESULT(yes)
-    GNUTLS_CFLAGS=`pkg-config gnutls --cflags`
-    GNUTLS_LFLAGS=`pkg-config gnutls --libs`
+    GNUTLS_CFLAGS=`$PKGCONFIG gnutls --cflags`
+    GNUTLS_LFLAGS=`$PKGCONFIG gnutls --libs`
     AC_SUBST(GNUTLS_CFLAGS)
     AC_SUBST(GNUTLS_LFLAGS)
 fi
@@ -1164,8 +1106,8 @@
 *** or try to install it with your software package manager.])
 else
     AC_MSG_RESULT(yes)
-    ZLIB_CFLAGS=`pkg-config zlib --cflags`
-    ZLIB_LFLAGS=`pkg-config zlib --libs`
+    ZLIB_CFLAGS=`$PKGCONFIG zlib --cflags`
+    ZLIB_LFLAGS=`$PKGCONFIG zlib --libs`
     AC_SUBST(ZLIB_CFLAGS)
     AC_SUBST(ZLIB_LFLAGS)
 fi
