--- weechat-1.6/configure.orig	2017-11-22 14:47:17.644000000 +0000
+++ weechat-1.6/configure	2017-11-22 15:05:58.224000000 +0000
@@ -24277,8 +24277,8 @@
 else
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-    GCRYPT_CFLAGS=`libgcrypt-config --cflags`
-    GCRYPT_LFLAGS=`libgcrypt-config --libs`
+    GCRYPT_CFLAGS=`${LIBGCRYPT_CONFIG} --cflags`
+    GCRYPT_LFLAGS=`${LIBGCRYPT_CONFIG} --libs`
 
 
 fi
@@ -24355,8 +24355,8 @@
     else
         { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-        GNUTLS_CFLAGS=`pkg-config gnutls --cflags`
-        GNUTLS_LFLAGS=`pkg-config gnutls --libs`
+        GNUTLS_CFLAGS=`$PKG_CONFIG gnutls --cflags`
+        GNUTLS_LFLAGS=`$PKG_CONFIG gnutls --libs`
 
 
         $as_echo "#define HAVE_GNUTLS 1" >>confdefs.h
@@ -24567,8 +24567,8 @@
 else
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
-    ZLIB_CFLAGS=`pkg-config zlib --cflags`
-    ZLIB_LFLAGS=`pkg-config zlib --libs`
+    ZLIB_CFLAGS=`$PKG_CONFIG zlib --cflags`
+    ZLIB_LFLAGS=`$PKG_CONFIG zlib --libs`
 
 
 fi
