diff -ru wget2-2.0.1.orig/libwget/net.c wget2-2.0.1/libwget/net.c
--- wget2-2.0.1.orig/libwget/net.c	2022-12-21 13:17:32.426227872 +0100
+++ wget2-2.0.1/libwget/net.c	2022-12-21 13:16:45.802227589 +0100
@@ -52,6 +52,7 @@
 # include <fcntl.h>
 #endif
 
+#if 0
 #if defined __APPLE__ && defined __MACH__ && defined CONNECT_DATA_IDEMPOTENT && defined CONNECT_RESUME_ON_READ_WRITE
 # define TCP_FASTOPEN_OSX
 #elif defined TCP_FASTOPEN_CONNECT // since Linux 4.11
@@ -59,6 +60,7 @@
 #elif defined TCP_FASTOPEN && defined MSG_FASTOPEN
 # define TCP_FASTOPEN_LINUX
 #endif
+#endif
 
 #include <wget.h>
 #include "private.h"
@@ -608,9 +610,11 @@
 	if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, (void *)&on, sizeof(on)) == -1)
 		error_printf(_("Failed to set socket option REUSEADDR\n"));
 
+#ifndef __midipix__
 	on = 1;
 	if (setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, (void *)&on, sizeof(on)) == -1)
 		error_printf(_("Failed to set socket option NODELAY\n"));
+#endif
 
 #ifdef SO_BINDTODEVICE
 	if (tcp->bind_interface) {
