--- whois-5.5.7/Makefile.orig	2020-10-03 16:43:15.000000000 +0100
+++ whois-5.5.7/Makefile	2021-01-23 07:55:24.889153830 +0000
@@ -42,12 +42,12 @@
 $(error Please fix your build system to stop defining HAVE_LIBIDN!)
 endif
 
-ifeq ($(shell $(PKG_CONFIG) --exists 'libidn2 >= 2.0.3' || echo NO),)
-whois_LDADD += $(shell $(PKG_CONFIG) --libs libidn2)
-DEFS += -DHAVE_LIBIDN2 $(shell $(PKG_CONFIG) --cflags libidn2)
-else ifeq ($(shell $(PKG_CONFIG) --exists 'libidn' || echo NO),)
-whois_LDADD += $(shell $(PKG_CONFIG) --libs libidn)
-DEFS += -DHAVE_LIBIDN $(shell $(PKG_CONFIG) --cflags libidn)
+ifeq ($(shell ${PKG_CONFIG} --exists 'libidn2 >= 2.0.3' || echo NO),)
+whois_LDADD += $(shell ${PKG_CONFIG} --libs libidn2)
+DEFS += -DHAVE_LIBIDN2 $(shell ${PKG_CONFIG} --cflags libidn2)
+else ifeq ($(shell ${PKG_CONFIG} --exists 'libidn' || echo NO),)
+whois_LDADD += $(shell ${PKG_CONFIG} --libs libidn)
+DEFS += -DHAVE_LIBIDN $(shell ${PKG_CONFIG} --cflags libidn)
 endif
 
 ifdef HAVE_ICONV
@@ -72,7 +72,7 @@
 CPPFLAGS += $(DEFS) $(INCLUDES)
 
 ##############################################################################
-all: Makefile.depend whois mkpasswd pos
+all: Makefile.depend whois mkpasswd
 
 ##############################################################################
 %.o: %.c
@@ -121,7 +121,7 @@
 	nice afl-fuzz -i ../afl_in -o ../afl_out -- ./whois
 
 ##############################################################################
-install: install-whois install-mkpasswd install-pos
+install: install-whois install-mkpasswd
 
 install-whois: whois
 	$(INSTALL) -d $(BASEDIR)$(prefix)/bin/
@@ -137,9 +137,6 @@
 	$(INSTALL) -m 0755 mkpasswd $(BASEDIR)$(prefix)/bin/
 	$(INSTALL) -m 0644 mkpasswd.1 $(BASEDIR)$(prefix)/share/man/man1/
 
-install-pos:
-	cd po && $(MAKE) install
-
 distclean: clean
 	rm -f po/whois.pot
 
@@ -148,9 +145,6 @@
 		new_gtlds.h tld_serv.h servers_charset.h *.o whois mkpasswd
 	rm -f po/*.mo
 
-pos:
-	cd po && $(MAKE)
-
 depend: Makefile.depend
 Makefile.depend:
 	$(CC) $(CPPFLAGS) $(CFLAGS) -MM -MG *.c > $@
