diff -ru wolfssl-4.4.0-stable.orig/configure.ac wolfssl-4.4.0-stable/configure.ac
--- wolfssl-4.4.0-stable.orig/configure.ac	2020-04-22 19:43:44.000000000 +0200
+++ wolfssl-4.4.0-stable/configure.ac	2020-06-18 10:23:08.173591237 +0200
@@ -179,7 +179,6 @@
     enable_certservice=yes
     enable_jni=yes
     enable_lighty=yes
-    enable_haproxy=yes
     enable_stunnel=yes
     enable_nginx=yes
     enable_openvpn=yes
