diff -ru yabasic-2.83.0.orig/configure yabasic-2.83.0/configure
--- yabasic-2.83.0.orig/configure	2019-06-15 16:58:01.081213787 +0200
+++ yabasic-2.83.0/configure	2019-06-15 16:58:27.697212954 +0200
@@ -13304,7 +13304,7 @@
   return 0;
 }
 _ACEOF
-for ac_lib in '' curses; do
+for ac_lib in '' ncurses; do
   if test -z "$ac_lib"; then
     ac_res="none required"
   else
Only in yabasic-2.83.0: configure.orig
diff -ru yabasic-2.83.0.orig/graphic.c yabasic-2.83.0/graphic.c
--- yabasic-2.83.0.orig/graphic.c	2019-05-16 06:01:06.000000000 +0200
+++ yabasic-2.83.0/graphic.c	2019-06-15 17:03:51.116903177 +0200
@@ -19,7 +19,11 @@
 #include "yabasic.h"		/* all prototypes and structures */
 #endif
 #ifdef UNIX
+#ifdef HAVE_NCURSES_H
+#include <ncurses.h>
+#elif HAVE_CURSES_H
 #include <curses.h>
+#endif
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
diff -ru yabasic-2.83.0.orig/whereami.c yabasic-2.83.0/whereami.c
--- yabasic-2.83.0.orig/whereami.c	2019-03-10 16:36:25.000000000 +0100
+++ yabasic-2.83.0/whereami.c	2019-06-15 17:08:39.723539493 +0200
@@ -164,7 +164,7 @@
   return length;
 }
 
-#elif defined(__linux__) || defined(__CYGWIN__) || defined(__sun) || (defined(__NetBSD__) && !defined(KERN_PROC_PATHNAME))
+#elif defined(__linux__) || defined(__midipix__) || defined(__CYGWIN__) || defined(__sun) || (defined(__NetBSD__) && !defined(KERN_PROC_PATHNAME))
 
 #include <stdio.h>
 #include <stdlib.h>
