diff -ru zip30.orig/unix/Makefile zip30/unix/Makefile
--- zip30.orig/unix/Makefile	2008-05-07 08:33:56.000000000 +0200
+++ zip30/unix/Makefile	2016-10-17 01:03:49.734470317 +0200
@@ -196,6 +196,12 @@
 # NO_UNICODE_SUPPORT    - do not enable Unicode support even if available.
 # NO_BZIP2_SUPPORT      - do not compile in bzip2 code even if available.
 
+# Midipix target
+midipix:
+	$(MAKE) $(MAKEF) zips LDFLAGS1="" CC="x86_64-nt64-midipix-gcc" CPP="x86_64-nt64-midipix-gcc -E" \
+	 CFLAGS="-g3 -O0 -DUNIX -I. -DHAVE_TERMIOS_H -DNO_BZIP2_SUPPORT -DLARGE_FILE_SUPPORT -DUNICODE_SUPPORT \
+	 -DHAVE_DIRENT_H -DUIDGID_NOT_16BIT" OBJA="" OCRCU8="crc32_.o " OCRCTB="" LN="ln -s" LFLAGS2=""
+
 #               Generic targets:
 
 generic: flags
