diff -ru a/programs/Makefile b/programs/Makefile
--- a/programs/Makefile	2018-03-26 22:19:34.000000000 +0000
+++ b/programs/Makefile	2018-04-30 14:55:39.468000000 +0000
@@ -38,7 +38,6 @@
 CPPFLAGS+= -I$(ZSTDDIR) -I$(ZSTDDIR)/common -I$(ZSTDDIR)/compress \
            -I$(ZSTDDIR)/dictBuilder \
            -DXXH_NAMESPACE=ZSTD_
-CFLAGS  ?= -O3
 DEBUGFLAGS+=-Wall -Wextra -Wcast-qual -Wcast-align -Wshadow \
             -Wstrict-aliasing=1 -Wswitch-enum -Wdeclaration-after-statement \
             -Wstrict-prototypes -Wundef -Wpointer-arith -Wformat-security \
diff -ru a/lib/libzstd.pc.in b/lib/libzstd.pc.in
--- a/lib/libzstd.pc.in	2020-03-27 11:36:13.368590549 -0400
+++ b/lib/libzstd.pc.in	2020-03-27 11:36:28.362034466 -0400
@@ -10,5 +10,5 @@
 Description: fast lossless compression algorithm library
 URL: http://www.zstd.net/
 Version: @VERSION@
-Libs: -L${libdir} -lzstd
+Libs: -lzstd
 Cflags: -I${includedir}
