diff -ru a/configfsf.sub b/configfsf.sub
--- a/configfsf.sub	2013-09-30 06:18:28.000000000 -0400
+++ b/configfsf.sub	2015-05-28 00:18:19.772345286 -0400
@@ -794,6 +794,9 @@
 	microblaze*)
 		basic_machine=microblaze-xilinx
 		;;
+	midipix)
+		os=-midipix
+		;;
 	mingw64)
 		basic_machine=x86_64-pc
 		os=-mingw64
@@ -1370,7 +1373,7 @@
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -mingw64* | -linux-gnu* | -linux-android* \
+	      | -midipix* | -mingw32* | -mingw64* | -linux-gnu* | -linux-android* \
 	      | -linux-newlib* | -linux-musl* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
diff -ru a/configure b/configure
--- a/configure	2013-09-30 06:18:33.000000000 -0400
+++ b/configure	2015-05-28 00:18:20.160345295 -0400
@@ -28684,7 +28684,7 @@
 
 echo "include_mpn(\`x86_64/darwin.m4')" >> $gmp_tmpconfigm4i
  ;;
-	    *-*-mingw* | *-*-cygwin)
+	    *-*-mingw* | *-*-cygwin | *-*-midipix)
 
 echo "include_mpn(\`x86_64/dos64.m4')" >> $gmp_tmpconfigm4i
  ;;
@@ -32457,3 +32457,15 @@
      ;;
    esac
 fi
+
+# a transitional workaround...
+case $host_os in
+		midipix*)
+			cp $ac_confdir/libtool.midipix libtool || exit 2
+
+			sed -i	-e 's/@cbb_host_os@/'$host'/g' \
+				-e 's/@cbb_build_alias@/'$build_alias'/g' \
+				-e 's/@cbb_build_os@/'$build_os'/g' \
+				-e 's/@cbb_build@/'$build'/g' libtool || exit 2
+			;;
+esac
