diff -ru a/configure b/configure
--- a/configure	2012-09-06 06:06:27.000000000 -0400
+++ b/configure	2015-03-10 19:12:26.580180368 -0400
@@ -16378,3 +16378,15 @@
 $as_echo "$as_me: WARNING: unrecognized options: $ac_unrecognized_opts" >&2;}
 fi
 
+# a transitional workaround...
+case $host_os in
+		midipix*)
+			cp $ac_confdir/libtool.midipix libtool || exit 2
+
+			sed -i	-e 's/@cbb_host_os@/'$host'/g' \
+				-e 's/@cbb_build_alias@/'$build_alias'/g' \
+				-e 's/@cbb_build_os@/'$build_os'/g' \
+				-e 's/@cbb_build@/'$build'/g' libtool || exit 2
+			;;
+esac
+
