--- musl-1.1.12/src/signal/nt32/sigsetjmp.s.orig	1970-01-01 00:00:00.000000000 +0000
+++ musl-1.1.12/src/signal/nt32/sigsetjmp.s	2016-05-30 20:27:49.741925447 +0000
@@ -0,0 +1,17 @@
+.text
+.globl	___sigsetjmp
+.globl	_sigsetjmp
+
+.def ___sigsetjmp; .scl 2; .type 32; .endef
+.def _sigsetjmp; .scl 2; .type 32; .endef
+
+___sigsetjmp:
+_sigsetjmp:
+	jmp ___sigsetjmp_tail	# defer to ___sigsetjmp_tail
+
+
+	.section .got$_sigsetjmp,"r"
+	.global __imp__sigsetjmp
+__imp__sigsetjmp:
+	.long	_sigsetjmp
+	.linkonce discard
