--- nc110/Makefile.orig	1996-03-20 03:16:06.000000000 +0000
+++ nc110/Makefile	2016-06-28 17:26:50.985424240 +0000
@@ -9,12 +9,12 @@
 # pick gcc if you'd rather , and/or do -g instead of -O if debugging
 # debugging
 # DFLAGS = -DTEST -DDEBUG
-CFLAGS = -O
+CFLAGS = -O $(CFLAGS_EXTRA)
 XFLAGS = 	# xtra cflags, set by systype targets
 XLIBS =		# xtra libs if necessary?
 # -Bstatic for sunos,  -static for gcc, etc.  You want this, trust me.
 STATIC =
-CC = cc $(CFLAGS)
+CC = cc ${CFLAGS}
 LD = $(CC) -s	# linker; defaults to stripped executables
 o = o		# object extension
 
@@ -28,7 +28,7 @@
 ### HARD TARGETS
 
 nc:	netcat.c
-	$(LD) $(DFLAGS) $(XFLAGS) $(STATIC) -o nc netcat.c $(XLIBS)
+	$(LD) $(CFLAGS) $(LDFLAGS) $(DFLAGS) $(XFLAGS) $(STATIC) -o nc netcat.c $(XLIBS)
 
 nc-dos:
 	@echo "DOS?!  Maybe someday, but not now"
@@ -109,6 +109,10 @@
 next:
 	make -e $(ALL) $(MFLAGS) XFLAGS='-DNEXT' STATIC=-Bstatic
 
+# midipix
+midipix:
+	make -e $(ALL) $(MFLAGS) XFLAGS='-DLINUX' STATIC=
+
 # start with this for a new architecture, and see what breaks.
 generic:
 	make -e $(ALL) $(MFLAGS) XFLAGS='-DGENERIC' STATIC=
--- nc110/generic.h.orig	1996-02-18 01:36:43.000000000 +0000
+++ nc110/generic.h	2016-06-15 18:44:06.379382216 +0000
@@ -360,6 +360,14 @@
 #undef HAVE_SELECT_X
 #endif /* NeXTSTEP 3.2 motorola */
 
+/* midipix */
+#ifdef MIDIPIX
+#undef UTMPX
+#undef HAVE_SYSINFO
+#undef HAVE_SELECT_H
+#undef HAVE_TTYENT_H
+#endif /* midipix */
+
 /* Make some "generic" assumptions if all else fails */
 #ifdef GENERIC
 #undef HAVE_FLOCK
