--- tmate-2.2.1/etc/config.sub.orig	2016-06-24 10:09:12.860128500 +0200
+++ tmate-2.2.1/etc/config.sub	2016-06-24 10:10:02.677362500 +0200
@@ -796,6 +796,10 @@
 	microblaze*)
 		basic_machine=microblaze-xilinx
 		;;
+	midipix*)
+		basic_machine=x86_64-pc
+		os=-midipix
+		;;
 	mingw64)
 		basic_machine=x86_64-pc
 		os=-mingw64
@@ -1376,7 +1380,7 @@
 	      | -udi* | -eabi* | -lites* | -ieee* | -go32* | -aux* \
 	      | -chorusos* | -chorusrdb* | -cegcc* \
 	      | -cygwin* | -msys* | -pe* | -psos* | -moss* | -proelf* | -rtems* \
-	      | -mingw32* | -mingw64* | -linux-gnu* | -linux-android* \
+	      | -midipix* | -mingw32* | -mingw64* | -linux-gnu* | -linux-android* \
 	      | -linux-newlib* | -linux-musl* | -linux-uclibc* \
 	      | -uxpv* | -beos* | -mpeix* | -udk* | -moxiebox* \
 	      | -interix* | -uwin* | -mks* | -rhapsody* | -darwin* | -opened* \
