--- vim-7.4.1952/src/osdef2.h.in.orig	2016-06-21 23:42:20.000000000 +0200
+++ vim-7.4.1952/src/osdef2.h.in	2016-09-06 18:33:43.655518796 +0200
@@ -92,5 +92,5 @@
 extern int	_Xmblen(char const *, size_t);
 #else
 		/* This is different from the header but matches mblen() */
-extern int	_Xmblen(char *, size_t);
+extern int	_Xmblen(char *, int);
 #endif
--- vim-7.4.1952/src/Makefile.orig	2016-06-21 23:42:20.000000000 +0200
+++ vim-7.4.1952/src/Makefile	2016-09-07 06:29:14.483862359 +0200
@@ -2692,7 +2692,7 @@
 	    $(PERLLIB)/ExtUtils/typemap if_perl.xs >> $@
 
 auto/osdef.h: auto/config.h osdef.sh osdef1.h.in osdef2.h.in
-	CC="$(CC) $(OSDEF_CFLAGS)" srcdir=$(srcdir) sh $(srcdir)/osdef.sh
+	CC="$(CC) $(OSDEF_CFLAGS) -I$(PKG_PREFIX)/include" srcdir=$(srcdir) sh $(srcdir)/osdef.sh
 
 auto/pathdef.c: Makefile auto/config.mk
 	-@echo creating $@
--- vim-7.4.1952/src/auto/configure.orig	2016-06-21 23:42:20.000000000 +0200
+++ vim-7.4.1952/src/auto/configure	2016-09-16 16:37:28.147266000 +0200
@@ -2603,7 +2603,7 @@
 ## -------------------- ##
 
 ac_ext=c
-ac_cpp='$CPP $CPPFLAGS'
+ac_cpp='$CPP $CFLAGS $CPPFLAGS'
 ac_compile='$CC -c $CFLAGS $CPPFLAGS conftest.$ac_ext >&5'
 ac_link='$CC -o conftest$ac_exeext $CFLAGS $CPPFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&5'
 ac_compiler_gnu=$ac_cv_c_compiler_gnu
@@ -7607,6 +7607,7 @@
 
 # Standard set of common directories for X headers.
 # Check X11 before X11Rn because it is often a symlink to the current release.
+if test -z "$ac_x_header_dirs"; then
 ac_x_header_dirs='
 /usr/X11/include
 /usr/X11R7/include
@@ -7645,6 +7646,7 @@
 
 /usr/openwin/include
 /usr/openwin/share/include'
+fi
 
 if test "$ac_x_includes" = no; then
   # Guess where to find include files, by looking for Xlib.h.
