--- xorg-server-1.20.1.orig/hw/xwin/winwndproc.c	2018-08-07 18:31:04.000000000 +0200
+++ xorg-server-1.20.1/hw/xwin/winwndproc.c	2018-08-26 03:03:49.505186369 +0200
@@ -120,7 +120,7 @@
         s_pScreenInfo = s_pScreenPriv->pScreenInfo;
         s_pScreen = s_pScreenInfo->pScreen;
         s_hwndLastPrivates = hwnd;
-        s_uTaskbarRestart = RegisterWindowMessage(TEXT("TaskbarCreated"));
+        s_uTaskbarRestart = RegisterWindowMessage("TaskbarCreated");
         SetProp(hwnd, WIN_SCR_PROP, s_pScreenPriv);
 
         /* Setup tray icon */
