--- xorg-server-1.20.1.orig/hw/xwin/winshadgdi.c	2018-08-07 18:31:04.000000000 +0200
+++ xorg-server-1.20.1/hw/xwin/winshadgdi.c	2018-08-20 22:51:52.902466150 +0200
@@ -332,7 +332,7 @@
     pScreenPriv->hbmpShadow = CreateDIBSection(pScreenPriv->hdcScreen,
                                                (BITMAPINFO *) pScreenPriv->
                                                pbmih, DIB_RGB_COLORS,
-                                               (VOID **) &pScreenInfo->pfb,
+                                               (void **) &pScreenInfo->pfb,
                                                NULL, 0);
     if (pScreenPriv->hbmpShadow == NULL || pScreenInfo->pfb == NULL) {
         winW32Error(2, "winAllocateFBShadowGDI - CreateDIBSection failed:");
