--- classpath-0.99.orig/native/jni/java-math/gnu_java_math_GMP.c	2008-06-22 04:05:39.000000000 +0200
+++ classpath-0.99/native/jni/java-math/gnu_java_math_GMP.c	2018-06-07 16:26:12.407213673 +0200
@@ -1131,7 +1131,7 @@
         res = 0;
         break;
       case 1:
-        res = mpz_popcount (_this);
+        res = mpz_popcount (_this); /* Falls through. */
       default:
         JCL_ThrowException (env, "java/lang/Error",
                             "Unexpected sign value for a native MPI");
