--- classpath-0.99.orig/configure.ac	2012-03-08 19:03:15.000000000 +0100
+++ classpath-0.99/configure.ac	2018-06-07 16:14:32.962168731 +0200
@@ -553,7 +553,7 @@
     dnl Check if we can link against the XTest library and set
     dnl HAVE_XTEST accordingly.
     AC_CHECK_LIB([Xtst], [XTestQueryExtension],
-                 [AC_DEFINE(HAVE_XTEST, 1, [Define to 1 if you have libXtst.])[XTEST_LIBS="$XTEST_LIBS -X11 -lXtst"]],
+                 [AC_DEFINE(HAVE_XTEST, 1, [Define to 1 if you have libXtst.])[XTEST_LIBS="$XTEST_LIBS -lX11 -lXtst"]],
                  [true],
                  [${X_LIBS}])
 
