diff -ru SDL2-2.0.12.orig/src/stdlib/SDL_iconv.c SDL2-2.0.12/src/stdlib/SDL_iconv.c
--- SDL2-2.0.12.orig/src/stdlib/SDL_iconv.c	2020-03-11 02:36:18.000000000 +0100
+++ SDL2-2.0.12/src/stdlib/SDL_iconv.c	2020-10-07 12:52:35.217963940 +0200
@@ -37,7 +37,7 @@
    iconv() may or may not use const char ** for the inbuf param.
    If we get this wrong, it's just a warning, so no big deal.
 */
-#if defined(_XGP6) || defined(__APPLE__) || defined(__RISCOS__) || \
+#if defined(_XGP6) || defined(__APPLE__) || defined(__RISCOS__) || defined(__midipix__) || \
     defined(__EMSCRIPTEN__) || \
     (defined(__GLIBC__) && ((__GLIBC__ > 2) || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 2)) || \
     (defined(_NEWLIB_VERSION)))
