diff -ru SmallBASIC.orig/src/common/extlib.c SmallBASIC/src/common/extlib.c
--- SmallBASIC.orig/src/common/extlib.c	2018-08-18 16:06:03.103804486 +0200
+++ SmallBASIC/src/common/extlib.c	2018-08-18 16:13:04.390441908 +0200
@@ -22,7 +22,7 @@
  #define WIN_EXTLIB
  #define LIB_EXT ".dll"
  #define DEFAULT_PATH "c:/sbasic/lib"
-#elif defined(__linux__) && defined(_UnixOS)
+#elif defined(__linux__) || defined(__midipix__) && defined(_UnixOS)
  #include <dlfcn.h>
  #define LNX_EXTLIB
  #define LIB_EXT ".so"
